{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c1dafe59",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package punkt to /homes/stirunag/nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "# !pip install scikit-learn\n",
    "import ast\n",
    "import pandas as pd\n",
    "from sklearn.model_selection import train_test_split\n",
    "import sys\n",
    "\n",
    "# Add the directory to sys.path\n",
    "sys.path.append('../../biohackathon2024/src/bh24_literature_mining')\n",
    "\n",
    "# Now import ml_tools\n",
    "import ml_tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "acb2ad01",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Task for</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(120, 129, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(72, 81, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(54, 63, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC11458576</td>\n",
       "      <td>Ana</td>\n",
       "      <td>Putative nucleases were identified by searchin...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(48, 57, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC11458576</td>\n",
       "      <td>Ana</td>\n",
       "      <td>Functional Enrichment Analysis (FEA) was perfo...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(57, 66, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2314</th>\n",
       "      <td>PMC10767879</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The authors wish to thank members of the VEuPa...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(41, 50, 'VEuPathDB', 'veupathdb')</td>\n",
       "      <td>Model organisms, Bioinformatics, Workflows, RN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2315</th>\n",
       "      <td>PMC10767879</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VEuPathDB has implemented an improved tool for...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(0, 9, 'VEuPathDB', 'veupathdb')</td>\n",
       "      <td>Model organisms, Bioinformatics, Workflows, RN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2316</th>\n",
       "      <td>PMC11223846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The web server provides an easy-to-use pipelin...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(128, 137, 'VEuPathDB', 'veupathdb')</td>\n",
       "      <td>Model organisms, Bioinformatics, Workflows, RN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2317</th>\n",
       "      <td>PMC11223846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>We ran a series of jobs with matching input ge...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(159, 168, 'VEuPathDB', 'veupathdb')</td>\n",
       "      <td>Model organisms, Bioinformatics, Workflows, RN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2318</th>\n",
       "      <td>PMC11223846</td>\n",
       "      <td>NaN</td>\n",
       "      <td>This could due to the fact that the VEuPathDB ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(36, 45, 'VEuPathDB', 'veupathdb')</td>\n",
       "      <td>Model organisms, Bioinformatics, Workflows, RN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2319 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID Task for                                           Sentence  \\\n",
       "0     PMC11286849      Ana  The identified proteins also included 52 (70%)...   \n",
       "1     PMC11286849      Ana  The localization of 52 of these 445 proteins i...   \n",
       "2     PMC11286849      Ana  The 445 identified proteins were searched agai...   \n",
       "3     PMC11458576      Ana  Putative nucleases were identified by searchin...   \n",
       "4     PMC11458576      Ana  Functional Enrichment Analysis (FEA) was perfo...   \n",
       "...           ...      ...                                                ...   \n",
       "2314  PMC10767879      NaN  The authors wish to thank members of the VEuPa...   \n",
       "2315  PMC10767879      NaN  VEuPathDB has implemented an improved tool for...   \n",
       "2316  PMC11223846      NaN  The web server provides an easy-to-use pipelin...   \n",
       "2317  PMC11223846      NaN  We ran a series of jobs with matching input ge...   \n",
       "2318  PMC11223846      NaN  This could due to the fact that the VEuPathDB ...   \n",
       "\n",
       "     True? False?                              NER_Tags  \\\n",
       "0     True  False  (120, 129, 'SubtiWiki', 'subtiwiki')   \n",
       "1     True  False    (72, 81, 'SubtiWiki', 'subtiwiki')   \n",
       "2     True  False    (54, 63, 'SubtiWiki', 'subtiwiki')   \n",
       "3     True  False    (48, 57, 'SubtiWiki', 'subtiwiki')   \n",
       "4     True  False    (57, 66, 'SubtiWiki', 'subtiwiki')   \n",
       "...    ...    ...                                   ...   \n",
       "2314   NaN    NaN    (41, 50, 'VEuPathDB', 'veupathdb')   \n",
       "2315   NaN    NaN      (0, 9, 'VEuPathDB', 'veupathdb')   \n",
       "2316   NaN    NaN  (128, 137, 'VEuPathDB', 'veupathdb')   \n",
       "2317   NaN    NaN  (159, 168, 'VEuPathDB', 'veupathdb')   \n",
       "2318   NaN    NaN    (36, 45, 'VEuPathDB', 'veupathdb')   \n",
       "\n",
       "                                                 Topics  \n",
       "0     Molecular interactions, pathways and networks,...  \n",
       "1     Molecular interactions, pathways and networks,...  \n",
       "2     Molecular interactions, pathways and networks,...  \n",
       "3     Molecular interactions, pathways and networks,...  \n",
       "4     Molecular interactions, pathways and networks,...  \n",
       "...                                                 ...  \n",
       "2314  Model organisms, Bioinformatics, Workflows, RN...  \n",
       "2315  Model organisms, Bioinformatics, Workflows, RN...  \n",
       "2316  Model organisms, Bioinformatics, Workflows, RN...  \n",
       "2317  Model organisms, Bioinformatics, Workflows, RN...  \n",
       "2318  Model organisms, Bioinformatics, Workflows, RN...  \n",
       "\n",
       "[2319 rows x 7 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# sheet_url = 'https://docs.google.com/spreadsheets/d/13BnWa9_r-8aVkbWSnuInvyhI41_WEALMBfT9X3frekI/export?format=csv&gid=79076201'\n",
    "new_sheet_url =  'https://docs.google.com/spreadsheets/d/13BnWa9_r-8aVkbWSnuInvyhI41_WEALMBfT9X3frekI/export?format=csv&gid=425276317'\n",
    "# Read the sheet into a DataFrame\n",
    "df = pd.read_csv(new_sheet_url)\n",
    "\n",
    "# Optionally, convert 'TRUE' and 'FALSE' to boolean\n",
    "df['True?'] = df['True?'].apply(lambda x: True if x == 'TRUE' else (False if x == 'FALSE' else x))\n",
    "df['False?'] = df['False?'].apply(lambda x: True if x == 'TRUE' else (False if x == 'FALSE' else x))\n",
    "\n",
    "\n",
    "# Display the DataFrame\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "010bee4e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Task for</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(120, 129, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(72, 81, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(54, 63, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC11458576</td>\n",
       "      <td>Ana</td>\n",
       "      <td>Putative nucleases were identified by searchin...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(48, 57, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC11458576</td>\n",
       "      <td>Ana</td>\n",
       "      <td>Functional Enrichment Analysis (FEA) was perfo...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(57, 66, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1681</th>\n",
       "      <td>PMC10761765</td>\n",
       "      <td>Rob</td>\n",
       "      <td>Actions such as rworkflows need only be update...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(210, 220, 'rworkflows', 'rworkflows')</td>\n",
       "      <td>Workflows, Computer science, Omics</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1682</th>\n",
       "      <td>PMC10761765</td>\n",
       "      <td>Rob</td>\n",
       "      <td>Sums of each of these metrics across all were ...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(99, 109, 'rworkflows', 'rworkflows')</td>\n",
       "      <td>Workflows, Computer science, Omics</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1683</th>\n",
       "      <td>PMC10761765</td>\n",
       "      <td>Rob</td>\n",
       "      <td>That said, if users wish to create a more cust...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(234, 244, 'rworkflows', 'rworkflows')</td>\n",
       "      <td>Workflows, Computer science, Omics</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1684</th>\n",
       "      <td>PMC10761765</td>\n",
       "      <td>Rob</td>\n",
       "      <td>That said, if users wish to create a more cust...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(85, 95, 'rworkflows', 'rworkflows')</td>\n",
       "      <td>Workflows, Computer science, Omics</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1685</th>\n",
       "      <td>PMC10761765</td>\n",
       "      <td>Rob</td>\n",
       "      <td>That said, if users wish to create a more cust...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(203, 213, 'rworkflows', 'rworkflows')</td>\n",
       "      <td>Workflows, Computer science, Omics</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>868 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID Task for                                           Sentence  \\\n",
       "0     PMC11286849      Ana  The identified proteins also included 52 (70%)...   \n",
       "1     PMC11286849      Ana  The localization of 52 of these 445 proteins i...   \n",
       "2     PMC11286849      Ana  The 445 identified proteins were searched agai...   \n",
       "3     PMC11458576      Ana  Putative nucleases were identified by searchin...   \n",
       "4     PMC11458576      Ana  Functional Enrichment Analysis (FEA) was perfo...   \n",
       "...           ...      ...                                                ...   \n",
       "1681  PMC10761765      Rob  Actions such as rworkflows need only be update...   \n",
       "1682  PMC10761765      Rob  Sums of each of these metrics across all were ...   \n",
       "1683  PMC10761765      Rob  That said, if users wish to create a more cust...   \n",
       "1684  PMC10761765      Rob  That said, if users wish to create a more cust...   \n",
       "1685  PMC10761765      Rob  That said, if users wish to create a more cust...   \n",
       "\n",
       "     True? False?                                NER_Tags  \\\n",
       "0     True  False    (120, 129, 'SubtiWiki', 'subtiwiki')   \n",
       "1     True  False      (72, 81, 'SubtiWiki', 'subtiwiki')   \n",
       "2     True  False      (54, 63, 'SubtiWiki', 'subtiwiki')   \n",
       "3     True  False      (48, 57, 'SubtiWiki', 'subtiwiki')   \n",
       "4     True  False      (57, 66, 'SubtiWiki', 'subtiwiki')   \n",
       "...    ...    ...                                     ...   \n",
       "1681  True  False  (210, 220, 'rworkflows', 'rworkflows')   \n",
       "1682  True  False   (99, 109, 'rworkflows', 'rworkflows')   \n",
       "1683  True  False  (234, 244, 'rworkflows', 'rworkflows')   \n",
       "1684  True  False    (85, 95, 'rworkflows', 'rworkflows')   \n",
       "1685  True  False  (203, 213, 'rworkflows', 'rworkflows')   \n",
       "\n",
       "                                                 Topics  \n",
       "0     Molecular interactions, pathways and networks,...  \n",
       "1     Molecular interactions, pathways and networks,...  \n",
       "2     Molecular interactions, pathways and networks,...  \n",
       "3     Molecular interactions, pathways and networks,...  \n",
       "4     Molecular interactions, pathways and networks,...  \n",
       "...                                                 ...  \n",
       "1681                 Workflows, Computer science, Omics  \n",
       "1682                 Workflows, Computer science, Omics  \n",
       "1683                 Workflows, Computer science, Omics  \n",
       "1684                 Workflows, Computer science, Omics  \n",
       "1685                 Workflows, Computer science, Omics  \n",
       "\n",
       "[868 rows x 7 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['False?'] == True, 'NER_Tags'] = None\n",
    "\n",
    "true_checked_df = df[(df['True?'] == True) | (df['False?'] == True)]\n",
    "\n",
    "# true_checked_df = df[(df['True?']| (df['False?']]\n",
    "\n",
    "true_checked_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1bac9831",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Task for</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>PMC11473683</td>\n",
       "      <td>Ana</td>\n",
       "      <td>Equally, 30/82 (37%) of SLiM-positive patients...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PMC11473683</td>\n",
       "      <td>Ana</td>\n",
       "      <td>SLiM-CRAB-positive patients presented with eit...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>PMC11473683</td>\n",
       "      <td>Ana</td>\n",
       "      <td>43 patients had more than one MDE with 3 (7%) ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>PMC11473683</td>\n",
       "      <td>Ana</td>\n",
       "      <td>43 patients had more than one MDE with 3 (7%) ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>PMC11491431</td>\n",
       "      <td>Ana</td>\n",
       "      <td>KM curves of groups of OS from OS-all cohort (...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1626</th>\n",
       "      <td>PMC11369124</td>\n",
       "      <td>Rob</td>\n",
       "      <td>Table 1Ten cases of accidental intra-arterial ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Metabolomics, Endocrinology and metabolism, Mo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1627</th>\n",
       "      <td>PMC11369124</td>\n",
       "      <td>Rob</td>\n",
       "      <td>Table 1Ten cases of accidental intra-arterial ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Metabolomics, Endocrinology and metabolism, Mo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1638</th>\n",
       "      <td>PMC9971431</td>\n",
       "      <td>Rob</td>\n",
       "      <td>Another non-microscopy-based method is called ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Metagenomics, Machine learning, Metagenomic se...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1639</th>\n",
       "      <td>PMC9971431</td>\n",
       "      <td>Rob</td>\n",
       "      <td>(B) NanoDeep (Ambrosetti et al., 2021), which ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Metagenomics, Machine learning, Metagenomic se...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1666</th>\n",
       "      <td>PMC10792193</td>\n",
       "      <td>Rob</td>\n",
       "      <td>REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Transcriptomics, Cell biology, Machine learnin...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>345 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID Task for                                           Sentence  \\\n",
       "37    PMC11473683      Ana  Equally, 30/82 (37%) of SLiM-positive patients...   \n",
       "38    PMC11473683      Ana  SLiM-CRAB-positive patients presented with eit...   \n",
       "39    PMC11473683      Ana  43 patients had more than one MDE with 3 (7%) ...   \n",
       "40    PMC11473683      Ana  43 patients had more than one MDE with 3 (7%) ...   \n",
       "41    PMC11491431      Ana  KM curves of groups of OS from OS-all cohort (...   \n",
       "...           ...      ...                                                ...   \n",
       "1626  PMC11369124      Rob  Table 1Ten cases of accidental intra-arterial ...   \n",
       "1627  PMC11369124      Rob  Table 1Ten cases of accidental intra-arterial ...   \n",
       "1638   PMC9971431      Rob  Another non-microscopy-based method is called ...   \n",
       "1639   PMC9971431      Rob  (B) NanoDeep (Ambrosetti et al., 2021), which ...   \n",
       "1666  PMC10792193      Rob   REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...   \n",
       "\n",
       "      True? False? NER_Tags                                             Topics  \n",
       "37    False   True     None  Ecology, Molecular interactions, pathways and ...  \n",
       "38    False   True     None  Ecology, Molecular interactions, pathways and ...  \n",
       "39    False   True     None  Ecology, Molecular interactions, pathways and ...  \n",
       "40    False   True     None  Ecology, Molecular interactions, pathways and ...  \n",
       "41    False   True     None  Ecology, Molecular interactions, pathways and ...  \n",
       "...     ...    ...      ...                                                ...  \n",
       "1626  False   True     None  Metabolomics, Endocrinology and metabolism, Mo...  \n",
       "1627  False   True     None  Metabolomics, Endocrinology and metabolism, Mo...  \n",
       "1638  False   True     None  Metagenomics, Machine learning, Metagenomic se...  \n",
       "1639  False   True     None  Metagenomics, Machine learning, Metagenomic se...  \n",
       "1666  False   True     None  Transcriptomics, Cell biology, Machine learnin...  \n",
       "\n",
       "[345 rows x 7 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_checked_df[true_checked_df['False?'] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "2fa9c575",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>(120, 129, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>(72, 81, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>(54, 63, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Putative nucleases were identified by searchin...</td>\n",
       "      <td>(48, 57, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Functional Enrichment Analysis (FEA) was perfo...</td>\n",
       "      <td>(57, 66, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1681</th>\n",
       "      <td>Actions such as rworkflows need only be update...</td>\n",
       "      <td>(210, 220, rworkflows, rworkflows)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1682</th>\n",
       "      <td>Sums of each of these metrics across all were ...</td>\n",
       "      <td>(99, 109, rworkflows, rworkflows)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1683</th>\n",
       "      <td>That said, if users wish to create a more cust...</td>\n",
       "      <td>(234, 244, rworkflows, rworkflows)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1684</th>\n",
       "      <td>That said, if users wish to create a more cust...</td>\n",
       "      <td>(85, 95, rworkflows, rworkflows)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1685</th>\n",
       "      <td>That said, if users wish to create a more cust...</td>\n",
       "      <td>(203, 213, rworkflows, rworkflows)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>868 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sentence  \\\n",
       "0     The identified proteins also included 52 (70%)...   \n",
       "1     The localization of 52 of these 445 proteins i...   \n",
       "2     The 445 identified proteins were searched agai...   \n",
       "3     Putative nucleases were identified by searchin...   \n",
       "4     Functional Enrichment Analysis (FEA) was perfo...   \n",
       "...                                                 ...   \n",
       "1681  Actions such as rworkflows need only be update...   \n",
       "1682  Sums of each of these metrics across all were ...   \n",
       "1683  That said, if users wish to create a more cust...   \n",
       "1684  That said, if users wish to create a more cust...   \n",
       "1685  That said, if users wish to create a more cust...   \n",
       "\n",
       "                                ner_ines  \n",
       "0       (120, 129, SubtiWiki, subtiwiki)  \n",
       "1         (72, 81, SubtiWiki, subtiwiki)  \n",
       "2         (54, 63, SubtiWiki, subtiwiki)  \n",
       "3         (48, 57, SubtiWiki, subtiwiki)  \n",
       "4         (57, 66, SubtiWiki, subtiwiki)  \n",
       "...                                  ...  \n",
       "1681  (210, 220, rworkflows, rworkflows)  \n",
       "1682   (99, 109, rworkflows, rworkflows)  \n",
       "1683  (234, 244, rworkflows, rworkflows)  \n",
       "1684    (85, 95, rworkflows, rworkflows)  \n",
       "1685  (203, 213, rworkflows, rworkflows)  \n",
       "\n",
       "[868 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_checked_df = true_checked_df[['Sentence','NER_Tags']]\n",
    "\n",
    "transformed_df = true_checked_df.rename(columns={\n",
    "    'Sentence': 'sentence',\n",
    "    'NER_Tags': 'ner_ines'\n",
    "})\n",
    "\n",
    "transformed_df['ner_ines'] = transformed_df['ner_ines'].apply(lambda x: ast.literal_eval(x) if isinstance(x, str) else x)\n",
    "\n",
    "transformed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0c6da881",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(120, 129, 'SubtiWiki', 'subtiwiki')"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transformed_df['ner_ines'].iloc[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "75effbd4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Group by 'sentence' and aggregate 'ner_ines'\n",
    "grouped_df = transformed_df.groupby('sentence')['ner_ines'].apply(lambda x: [i for i in x if i is not None]).reset_index()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "12394c19",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Algorithm 1A single iteration of the MaBoSS s...</td>\n",
       "      <td>[(38, 44, MaBoSS, maboss)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Core Tip: This manuscript is to correct the c...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REAGENT or RESOURCESOURCEIDENTIFIERBiological...</td>\n",
       "      <td>[(1638, 1649, CLIPPER 2.0, clipper_2.0)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>subtilis secretome as annotated in the Subtiwi...</td>\n",
       "      <td>[(39, 48, SubtiWiki, subtiwiki)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>tRigon session report in html-format for a k-m...</td>\n",
       "      <td>[(0, 6, tRigon, trigon)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>trimal -in mafft_out.fa -out mafftout_0.13.fas...</td>\n",
       "      <td>[(0, 6, trimAl, trimal)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>“Problems paying medical bills” is based on a ...</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>653</th>\n",
       "      <td>•Trans-PCO outperforms existing methods by fin...</td>\n",
       "      <td>[(81, 90, trans-PCO, trans-pco), (1, 10, trans...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>654 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sentence  \\\n",
       "0     Algorithm 1A single iteration of the MaBoSS s...   \n",
       "1     Core Tip: This manuscript is to correct the c...   \n",
       "2     REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...   \n",
       "3     REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...   \n",
       "4     REAGENT or RESOURCESOURCEIDENTIFIERBiological...   \n",
       "..                                                 ...   \n",
       "649  subtilis secretome as annotated in the Subtiwi...   \n",
       "650  tRigon session report in html-format for a k-m...   \n",
       "651  trimal -in mafft_out.fa -out mafftout_0.13.fas...   \n",
       "652  “Problems paying medical bills” is based on a ...   \n",
       "653  •Trans-PCO outperforms existing methods by fin...   \n",
       "\n",
       "                                              ner_ines  \n",
       "0                           [(38, 44, MaBoSS, maboss)]  \n",
       "1                                                   []  \n",
       "2                                                   []  \n",
       "3                                                   []  \n",
       "4             [(1638, 1649, CLIPPER 2.0, clipper_2.0)]  \n",
       "..                                                 ...  \n",
       "649                   [(39, 48, SubtiWiki, subtiwiki)]  \n",
       "650                           [(0, 6, tRigon, trigon)]  \n",
       "651                           [(0, 6, trimAl, trimal)]  \n",
       "652                                                 []  \n",
       "653  [(81, 90, trans-PCO, trans-pco), (1, 10, trans...  \n",
       "\n",
       "[654 rows x 2 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e5f70ddf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Algorithm 1A single iteration of the MaBoSS s...</td>\n",
       "      <td>[[38, 44, MaBoSS, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Core Tip: This manuscript is to correct the c...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>REAGENT or RESOURCESOURCEIDENTIFIERBiological...</td>\n",
       "      <td>[[1638, 1649, CLIPPER 2.0, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>subtilis secretome as annotated in the Subtiwi...</td>\n",
       "      <td>[[39, 48, SubtiWiki, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>tRigon session report in html-format for a k-m...</td>\n",
       "      <td>[[0, 6, tRigon, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>trimal -in mafft_out.fa -out mafftout_0.13.fas...</td>\n",
       "      <td>[[0, 6, trimAl, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>“Problems paying medical bills” is based on a ...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>653</th>\n",
       "      <td>•Trans-PCO outperforms existing methods by fin...</td>\n",
       "      <td>[[81, 90, trans-PCO, BT], [1, 10, trans-PCO, BT]]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>654 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sentence  \\\n",
       "0     Algorithm 1A single iteration of the MaBoSS s...   \n",
       "1     Core Tip: This manuscript is to correct the c...   \n",
       "2     REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...   \n",
       "3     REAGENT or RESOURCESOURCEIDENTIFIERAntibodies...   \n",
       "4     REAGENT or RESOURCESOURCEIDENTIFIERBiological...   \n",
       "..                                                 ...   \n",
       "649  subtilis secretome as annotated in the Subtiwi...   \n",
       "650  tRigon session report in html-format for a k-m...   \n",
       "651  trimal -in mafft_out.fa -out mafftout_0.13.fas...   \n",
       "652  “Problems paying medical bills” is based on a ...   \n",
       "653  •Trans-PCO outperforms existing methods by fin...   \n",
       "\n",
       "                                              ner_ines  \n",
       "0                               [[38, 44, MaBoSS, BT]]  \n",
       "1                                                 None  \n",
       "2                                                 None  \n",
       "3                                                 None  \n",
       "4                      [[1638, 1649, CLIPPER 2.0, BT]]  \n",
       "..                                                 ...  \n",
       "649                          [[39, 48, SubtiWiki, BT]]  \n",
       "650                               [[0, 6, tRigon, BT]]  \n",
       "651                               [[0, 6, trimAl, BT]]  \n",
       "652                                               None  \n",
       "653  [[81, 90, trans-PCO, BT], [1, 10, trans-PCO, BT]]  \n",
       "\n",
       "[654 rows x 2 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_df['ner_ines'] = grouped_df['ner_ines'].apply(\n",
    "    lambda x: [[item[0], item[1], item[2], 'BT'] for item in x] if x else None\n",
    ")\n",
    "grouped_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a9fc967b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sentence    Actions such as rworkflows need only be update...\n",
       "ner_ines    [[166, 176, rworkflows, BT], [16, 26, rworkflo...\n",
       "Name: 52, dtype: object"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_df.iloc[52]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "869b7c99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>Note：Pinelliae Rhizoma Praeparatum Cum Alumine...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>456</th>\n",
       "      <td>The extracted patterns were classified as “bre...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>PHARE utilized two different pipelines for scR...</td>\n",
       "      <td>[[0, 5, PHARE, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>2a Illustration of a typical pyHiM analysis on...</td>\n",
       "      <td>[[29, 34, pyHiM, BT], [130, 135, pyHiM, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>570</th>\n",
       "      <td>Troubleshooting 2....# Load sam/bedtoolsmodule...</td>\n",
       "      <td>[[307, 315, SAMtools, BT], [79, 87, SAMtools, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sentence  \\\n",
       "321  Note：Pinelliae Rhizoma Praeparatum Cum Alumine...   \n",
       "456  The extracted patterns were classified as “bre...   \n",
       "340  PHARE utilized two different pipelines for scR...   \n",
       "29   2a Illustration of a typical pyHiM analysis on...   \n",
       "570  Troubleshooting 2....# Load sam/bedtoolsmodule...   \n",
       "\n",
       "                                              ner_ines  \n",
       "321                                               None  \n",
       "456                                               None  \n",
       "340                                [[0, 5, PHARE, BT]]  \n",
       "29        [[29, 34, pyHiM, BT], [130, 135, pyHiM, BT]]  \n",
       "570  [[307, 315, SAMtools, BT], [79, 87, SAMtools, ...  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_df, test_df = train_test_split(grouped_df, test_size=0.2, random_state=42)\n",
    "\n",
    "train_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ac80e692",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing batches: 100%|███████████████████████████████████████████████████████████████| 2/2 [00:00<00:00, 23.79it/s]\n",
      "Processing batches: 100%|██████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 126.58it/s]\n"
     ]
    }
   ],
   "source": [
    "# Example usage:\n",
    "output_folder = '../data/IOB/'\n",
    "\n",
    "# Convert train, dev, and test dataframes to IOB format\n",
    "ml_tools.convert_to_IOB_format_from_df(train_df, output_folder, 'train_IOB.tsv')\n",
    "ml_tools.convert_to_IOB_format_from_df(test_df, output_folder, 'dev_IOB.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3c335960",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Checking Dataset 1:\n",
      "Train set 1 is valid.\n",
      "Dev set 1 is valid.\n",
      "Test set 1 is valid.\n"
     ]
    }
   ],
   "source": [
    "# Example usage:\n",
    "train_files = [output_folder+'train_IOB.tsv']\n",
    "dev_files = [output_folder+'dev_IOB.tsv']\n",
    "\n",
    "ml_tools.check_integrity_of_files(train_files, dev_files,dev_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "372e935a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "BioHackathon 2024",
   "language": "python",
   "name": "biohackathon-2024"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
