{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1a32b65f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/nfs/production/literature/santosh_tirunagari/ENVS/biohackathon_env/bin/pip\r\n"
     ]
    }
   ],
   "source": [
    "!which pip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f8de185e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Using this technique, we identified 445 protei...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(233, 242, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(54, 63, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(120, 129, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(72, 81, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC10832249</td>\n",
       "      <td>MetaPathways v2.5 [37] was used to create the ...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(0, 12, 'MetaPathways', 'metapathways')</td>\n",
       "      <td>Metagenomics, Microbial ecology, Population ge...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         PMCID                                           Sentence True?  \\\n",
       "0  PMC11286849  Using this technique, we identified 445 protei...  TRUE   \n",
       "1  PMC11286849  The 445 identified proteins were searched agai...  TRUE   \n",
       "2  PMC11286849  The identified proteins also included 52 (70%)...  TRUE   \n",
       "3  PMC11286849  The localization of 52 of these 445 proteins i...  TRUE   \n",
       "4  PMC10832249  MetaPathways v2.5 [37] was used to create the ...  TRUE   \n",
       "\n",
       "   False?                                 NER_Tags  \\\n",
       "0   False     (233, 242, 'SubtiWiki', 'subtiwiki')   \n",
       "1   False       (54, 63, 'SubtiWiki', 'subtiwiki')   \n",
       "2   False     (120, 129, 'SubtiWiki', 'subtiwiki')   \n",
       "3   False       (72, 81, 'SubtiWiki', 'subtiwiki')   \n",
       "4   False  (0, 12, 'MetaPathways', 'metapathways')   \n",
       "\n",
       "                                              Topics  \n",
       "0  Molecular interactions, pathways and networks,...  \n",
       "1  Molecular interactions, pathways and networks,...  \n",
       "2  Molecular interactions, pathways and networks,...  \n",
       "3  Molecular interactions, pathways and networks,...  \n",
       "4  Metagenomics, Microbial ecology, Population ge...  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "sheet_url = 'https://docs.google.com/spreadsheets/d/13BnWa9_r-8aVkbWSnuInvyhI41_WEALMBfT9X3frekI/export?format=csv&gid=79076201'\n",
    "\n",
    "# Read the sheet into a DataFrame\n",
    "df = pd.read_csv(sheet_url)\n",
    "\n",
    "# Display the DataFrame\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "47696715",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Using this technique, we identified 445 protei...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(233, 242, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(54, 63, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(120, 129, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(72, 81, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC10832249</td>\n",
       "      <td>MetaPathways v2.5 [37] was used to create the ...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(0, 12, 'MetaPathways', 'metapathways')</td>\n",
       "      <td>Metagenomics, Microbial ecology, Population ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>PMC10988918</td>\n",
       "      <td>The CITEViz documentation website can be acces...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(4, 11, 'CITEViz', 'citeviz')</td>\n",
       "      <td>Cytometry, Transcriptomics, Workflows, RNA-Seq...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2022</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The central objective behind the development o...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(48, 59, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2023</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The steps to contributing to the project are d...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(103, 114, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2024</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>scalepopgen is hosted on GitHub under the popg...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(0, 11, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2025</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The results of the case study described in the...</td>\n",
       "      <td>TRUE</td>\n",
       "      <td>False</td>\n",
       "      <td>(97, 108, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>87 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID                                           Sentence True?  \\\n",
       "0     PMC11286849  Using this technique, we identified 445 protei...  TRUE   \n",
       "1     PMC11286849  The 445 identified proteins were searched agai...  TRUE   \n",
       "2     PMC11286849  The identified proteins also included 52 (70%)...  TRUE   \n",
       "3     PMC11286849  The localization of 52 of these 445 proteins i...  TRUE   \n",
       "4     PMC10832249  MetaPathways v2.5 [37] was used to create the ...  TRUE   \n",
       "...           ...                                                ...   ...   \n",
       "1088  PMC10988918  The CITEViz documentation website can be acces...  TRUE   \n",
       "2022  PMC10994858  The central objective behind the development o...  TRUE   \n",
       "2023  PMC10994858  The steps to contributing to the project are d...  TRUE   \n",
       "2024  PMC10994858  scalepopgen is hosted on GitHub under the popg...  TRUE   \n",
       "2025  PMC10994858  The results of the case study described in the...  TRUE   \n",
       "\n",
       "      False?                                  NER_Tags  \\\n",
       "0      False      (233, 242, 'SubtiWiki', 'subtiwiki')   \n",
       "1      False        (54, 63, 'SubtiWiki', 'subtiwiki')   \n",
       "2      False      (120, 129, 'SubtiWiki', 'subtiwiki')   \n",
       "3      False        (72, 81, 'SubtiWiki', 'subtiwiki')   \n",
       "4      False   (0, 12, 'MetaPathways', 'metapathways')   \n",
       "...      ...                                       ...   \n",
       "1088   False             (4, 11, 'CITEViz', 'citeviz')   \n",
       "2022   False    (48, 59, 'scalepopgen', 'scalepopgen')   \n",
       "2023   False  (103, 114, 'scalepopgen', 'scalepopgen')   \n",
       "2024   False     (0, 11, 'scalepopgen', 'scalepopgen')   \n",
       "2025   False   (97, 108, 'scalepopgen', 'scalepopgen')   \n",
       "\n",
       "                                                 Topics  \n",
       "0     Molecular interactions, pathways and networks,...  \n",
       "1     Molecular interactions, pathways and networks,...  \n",
       "2     Molecular interactions, pathways and networks,...  \n",
       "3     Molecular interactions, pathways and networks,...  \n",
       "4     Metagenomics, Microbial ecology, Population ge...  \n",
       "...                                                 ...  \n",
       "1088  Cytometry, Transcriptomics, Workflows, RNA-Seq...  \n",
       "2022  Workflows, Population genomics, Metagenomics, ...  \n",
       "2023  Workflows, Population genomics, Metagenomics, ...  \n",
       "2024  Workflows, Population genomics, Metagenomics, ...  \n",
       "2025  Workflows, Population genomics, Metagenomics, ...  \n",
       "\n",
       "[87 rows x 6 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter rows where 'True?' is checked\n",
    "true_checked_df = df[df['True?'] == 'TRUE']\n",
    "\n",
    "true_checked_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2496437c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Example usage:\n",
    "output_folder = '../data/DICT2ML/'\n",
    "\n",
    "# Convert train, dev, and test dataframes to IOB format\n",
    "convert_to_IOB_format_from_df(train_df, output_folder, 'train_IOB.tsv')\n",
    "convert_to_IOB_format_from_df(dev_df, output_folder, 'dev_IOB.tsv')\n",
    "convert_to_IOB_format_from_df(test_df, output_folder, 'test_IOB.tsv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "BioHackathon 2024",
   "language": "python",
   "name": "biohackathon-2024"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
