{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "c1dafe59",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install scikit-learn\n",
    "import ast\n",
    "import pandas as pd\n",
    "from sklearn.model_selection import train_test_split\n",
    "import sys\n",
    "from pathlib import Path\n",
    "\n",
    "# Add the directory to sys.path\n",
    "sys.path.append('../../biohackathon2024/src/bh24_literature_mining')\n",
    "\n",
    "# Now import ml_tools\n",
    "import ml_tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "acb2ad01",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load the data\n",
    "p = Path().cwd().parent\n",
    "path_sheet_mentions = p / 'data/annotated/250227mentions.csv'\n",
    "path_sheet_mentions_extra = p / 'data/annotated/250227mentions_extra.csv'\n",
    "\n",
    "# Read the sheet into a DataFrame\n",
    "sheet_mentions = pd.read_csv(path_sheet_mentions)\n",
    "sheet_mentions_extra = pd.read_csv(path_sheet_mentions_extra)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "cc7a603f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2097, 1320)"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(sheet_mentions), len(sheet_mentions_extra)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "f8849560",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3417"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Combine the two sheets\n",
    "df = pd.concat([sheet_mentions, sheet_mentions_extra], ignore_index=True)\n",
    "len(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "7bdf868b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 3417 entries, 0 to 3416\n",
      "Data columns (total 7 columns):\n",
      " #   Column    Non-Null Count  Dtype \n",
      "---  ------    --------------  ----- \n",
      " 0   PMCID     3417 non-null   object\n",
      " 1   Task for  2425 non-null   object\n",
      " 2   Sentence  3417 non-null   object\n",
      " 3   True?     3417 non-null   bool  \n",
      " 4   False?    3417 non-null   bool  \n",
      " 5   NER_Tags  3417 non-null   object\n",
      " 6   Topics    3314 non-null   object\n",
      "dtypes: bool(2), object(5)\n",
      "memory usage: 140.3+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "1fc892c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Task for</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(120, 129, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(72, 81, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Ana</td>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(54, 63, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC11458576</td>\n",
       "      <td>Ana</td>\n",
       "      <td>Putative nucleases were identified by searchin...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(48, 57, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC11458576</td>\n",
       "      <td>Ana</td>\n",
       "      <td>Functional Enrichment Analysis (FEA) was perfo...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(57, 66, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         PMCID Task for                                           Sentence  \\\n",
       "0  PMC11286849      Ana  The identified proteins also included 52 (70%)...   \n",
       "1  PMC11286849      Ana  The localization of 52 of these 445 proteins i...   \n",
       "2  PMC11286849      Ana  The 445 identified proteins were searched agai...   \n",
       "3  PMC11458576      Ana  Putative nucleases were identified by searchin...   \n",
       "4  PMC11458576      Ana  Functional Enrichment Analysis (FEA) was perfo...   \n",
       "\n",
       "   True?  False?                              NER_Tags  \\\n",
       "0   True   False  (120, 129, 'SubtiWiki', 'subtiwiki')   \n",
       "1   True   False    (72, 81, 'SubtiWiki', 'subtiwiki')   \n",
       "2   True   False    (54, 63, 'SubtiWiki', 'subtiwiki')   \n",
       "3   True   False    (48, 57, 'SubtiWiki', 'subtiwiki')   \n",
       "4   True   False    (57, 66, 'SubtiWiki', 'subtiwiki')   \n",
       "\n",
       "                                              Topics  \n",
       "0  Molecular interactions, pathways and networks,...  \n",
       "1  Molecular interactions, pathways and networks,...  \n",
       "2  Molecular interactions, pathways and networks,...  \n",
       "3  Molecular interactions, pathways and networks,...  \n",
       "4  Molecular interactions, pathways and networks,...  "
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Display the DataFrame\n",
    "df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "010bee4e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3376"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['False?'] == True, 'NER_Tags'] = None\n",
    "true_checked_df = df[(df['True?'] == True) | (df['False?'] == True)]\n",
    "\n",
    "len(true_checked_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "1bac9831",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Task for</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>PMC11473683</td>\n",
       "      <td>Ana</td>\n",
       "      <td>Equally, 30/82 (37%) of SLiM-positive patients...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PMC11473683</td>\n",
       "      <td>Ana</td>\n",
       "      <td>SLiM-CRAB-positive patients presented with eit...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>PMC11473683</td>\n",
       "      <td>Ana</td>\n",
       "      <td>43 patients had more than one MDE with 3 (7%) ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>PMC11473683</td>\n",
       "      <td>Ana</td>\n",
       "      <td>43 patients had more than one MDE with 3 (7%) ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>PMC11491431</td>\n",
       "      <td>Ana</td>\n",
       "      <td>KM curves of groups of OS from OS-all cohort (...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Ecology, Molecular interactions, pathways and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3412</th>\n",
       "      <td>PMC11377344</td>\n",
       "      <td>NaN</td>\n",
       "      <td>The proposed BLR model follows a Variational A...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Workflows, Sequencing, Sequence assembly, DNA ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3413</th>\n",
       "      <td>PMC11377344</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Additional details of the BLR model and VAE ap...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Workflows, Sequencing, Sequence assembly, DNA ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3414</th>\n",
       "      <td>PMC11299494</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS on this panel identified 13 QTLs signific...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Workflows, Sequencing, Sequence assembly, DNA ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3415</th>\n",
       "      <td>PMC11299494</td>\n",
       "      <td>NaN</td>\n",
       "      <td>(2018) assembled an international barley panel...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Workflows, Sequencing, Sequence assembly, DNA ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3416</th>\n",
       "      <td>PMC11299494</td>\n",
       "      <td>NaN</td>\n",
       "      <td>We hypothesised that this panel, which carries...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Workflows, Sequencing, Sequence assembly, DNA ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>783 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID Task for                                           Sentence  \\\n",
       "37    PMC11473683      Ana  Equally, 30/82 (37%) of SLiM-positive patients...   \n",
       "38    PMC11473683      Ana  SLiM-CRAB-positive patients presented with eit...   \n",
       "39    PMC11473683      Ana  43 patients had more than one MDE with 3 (7%) ...   \n",
       "40    PMC11473683      Ana  43 patients had more than one MDE with 3 (7%) ...   \n",
       "41    PMC11491431      Ana  KM curves of groups of OS from OS-all cohort (...   \n",
       "...           ...      ...                                                ...   \n",
       "3412  PMC11377344      NaN  The proposed BLR model follows a Variational A...   \n",
       "3413  PMC11377344      NaN  Additional details of the BLR model and VAE ap...   \n",
       "3414  PMC11299494      NaN  GWAS on this panel identified 13 QTLs signific...   \n",
       "3415  PMC11299494      NaN  (2018) assembled an international barley panel...   \n",
       "3416  PMC11299494      NaN  We hypothesised that this panel, which carries...   \n",
       "\n",
       "      True?  False? NER_Tags  \\\n",
       "37    False    True     None   \n",
       "38    False    True     None   \n",
       "39    False    True     None   \n",
       "40    False    True     None   \n",
       "41    False    True     None   \n",
       "...     ...     ...      ...   \n",
       "3412  False    True     None   \n",
       "3413  False    True     None   \n",
       "3414  False    True     None   \n",
       "3415  False    True     None   \n",
       "3416  False    True     None   \n",
       "\n",
       "                                                 Topics  \n",
       "37    Ecology, Molecular interactions, pathways and ...  \n",
       "38    Ecology, Molecular interactions, pathways and ...  \n",
       "39    Ecology, Molecular interactions, pathways and ...  \n",
       "40    Ecology, Molecular interactions, pathways and ...  \n",
       "41    Ecology, Molecular interactions, pathways and ...  \n",
       "...                                                 ...  \n",
       "3412  Workflows, Sequencing, Sequence assembly, DNA ...  \n",
       "3413  Workflows, Sequencing, Sequence assembly, DNA ...  \n",
       "3414  Workflows, Sequencing, Sequence assembly, DNA ...  \n",
       "3415  Workflows, Sequencing, Sequence assembly, DNA ...  \n",
       "3416  Workflows, Sequencing, Sequence assembly, DNA ...  \n",
       "\n",
       "[783 rows x 7 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_checked_df[true_checked_df['False?'] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "2fa9c575",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>(120, 129, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>(72, 81, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>(54, 63, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC11458576</td>\n",
       "      <td>Putative nucleases were identified by searchin...</td>\n",
       "      <td>(48, 57, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC11458576</td>\n",
       "      <td>Functional Enrichment Analysis (FEA) was perfo...</td>\n",
       "      <td>(57, 66, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3412</th>\n",
       "      <td>PMC11377344</td>\n",
       "      <td>The proposed BLR model follows a Variational A...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3413</th>\n",
       "      <td>PMC11377344</td>\n",
       "      <td>Additional details of the BLR model and VAE ap...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3414</th>\n",
       "      <td>PMC11299494</td>\n",
       "      <td>GWAS on this panel identified 13 QTLs signific...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3415</th>\n",
       "      <td>PMC11299494</td>\n",
       "      <td>(2018) assembled an international barley panel...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3416</th>\n",
       "      <td>PMC11299494</td>\n",
       "      <td>We hypothesised that this panel, which carries...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3376 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID                                           sentence  \\\n",
       "0     PMC11286849  The identified proteins also included 52 (70%)...   \n",
       "1     PMC11286849  The localization of 52 of these 445 proteins i...   \n",
       "2     PMC11286849  The 445 identified proteins were searched agai...   \n",
       "3     PMC11458576  Putative nucleases were identified by searchin...   \n",
       "4     PMC11458576  Functional Enrichment Analysis (FEA) was perfo...   \n",
       "...           ...                                                ...   \n",
       "3412  PMC11377344  The proposed BLR model follows a Variational A...   \n",
       "3413  PMC11377344  Additional details of the BLR model and VAE ap...   \n",
       "3414  PMC11299494  GWAS on this panel identified 13 QTLs signific...   \n",
       "3415  PMC11299494  (2018) assembled an international barley panel...   \n",
       "3416  PMC11299494  We hypothesised that this panel, which carries...   \n",
       "\n",
       "                              ner_ines  \n",
       "0     (120, 129, SubtiWiki, subtiwiki)  \n",
       "1       (72, 81, SubtiWiki, subtiwiki)  \n",
       "2       (54, 63, SubtiWiki, subtiwiki)  \n",
       "3       (48, 57, SubtiWiki, subtiwiki)  \n",
       "4       (57, 66, SubtiWiki, subtiwiki)  \n",
       "...                                ...  \n",
       "3412                              None  \n",
       "3413                              None  \n",
       "3414                              None  \n",
       "3415                              None  \n",
       "3416                              None  \n",
       "\n",
       "[3376 rows x 3 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_checked_df = true_checked_df[['PMCID', 'Sentence','NER_Tags']]\n",
    "\n",
    "transformed_df = true_checked_df.rename(columns={'PMCID': 'PMCID',\n",
    "    'Sentence': 'sentence',\n",
    "    'NER_Tags': 'ner_ines'\n",
    "})\n",
    "\n",
    "transformed_df['ner_ines'] = transformed_df['ner_ines'].apply(lambda x: ast.literal_eval(x) if isinstance(x, str) else x)\n",
    "\n",
    "transformed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "75effbd4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>PMCID</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Prana Jagannatha GN ,  Mendel B ,  Labi NPT...</td>\n",
       "      <td>PMC11317698</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>(A) Intersection of MAGMA, TWAS, PWAS in this...</td>\n",
       "      <td>PMC11443877</td>\n",
       "      <td>[(21, 26, MAGMA, magma-pipeline)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>- \"We identify some limitations of MarkerScan...</td>\n",
       "      <td>PMC11016177</td>\n",
       "      <td>[(36, 46, MarkerScan, markerscan), (128, 138, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Algorithm 1A single iteration of the MaBoSS s...</td>\n",
       "      <td>PMC11127412</td>\n",
       "      <td>[(38, 44, MaBoSS, maboss)]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Author contributions: Conceptualization: Shuc...</td>\n",
       "      <td>PMC11340858</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sentence        PMCID  \\\n",
       "0     Prana Jagannatha GN ,  Mendel B ,  Labi NPT...  PMC11317698   \n",
       "1   (A) Intersection of MAGMA, TWAS, PWAS in this...  PMC11443877   \n",
       "2   - \"We identify some limitations of MarkerScan...  PMC11016177   \n",
       "3   Algorithm 1A single iteration of the MaBoSS s...  PMC11127412   \n",
       "4   Author contributions: Conceptualization: Shuc...  PMC11340858   \n",
       "\n",
       "                                            ner_ines  \n",
       "0                                                 []  \n",
       "1                  [(21, 26, MAGMA, magma-pipeline)]  \n",
       "2  [(36, 46, MarkerScan, markerscan), (128, 138, ...  \n",
       "3                         [(38, 44, MaBoSS, maboss)]  \n",
       "4                                                 []  "
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Group by 'sentence' and aggregate 'ner_ines'\n",
    "grouped_df = transformed_df.groupby(['sentence', 'PMCID'])['ner_ines'].apply(lambda x: [i for i in x if i is not None]).reset_index()\n",
    "grouped_df.head(5)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5f70ddf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>PMCID</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Prana Jagannatha GN ,  Mendel B ,  Labi NPT...</td>\n",
       "      <td>PMC11317698</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>(A) Intersection of MAGMA, TWAS, PWAS in this...</td>\n",
       "      <td>PMC11443877</td>\n",
       "      <td>[[21, 26, MAGMA, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>- \"We identify some limitations of MarkerScan...</td>\n",
       "      <td>PMC11016177</td>\n",
       "      <td>[[36, 46, MarkerScan, BT], [128, 138, MarkerSc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Algorithm 1A single iteration of the MaBoSS s...</td>\n",
       "      <td>PMC11127412</td>\n",
       "      <td>[[38, 44, MaBoSS, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Author contributions: Conceptualization: Shuc...</td>\n",
       "      <td>PMC11340858</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            sentence        PMCID  \\\n",
       "0     Prana Jagannatha GN ,  Mendel B ,  Labi NPT...  PMC11317698   \n",
       "1   (A) Intersection of MAGMA, TWAS, PWAS in this...  PMC11443877   \n",
       "2   - \"We identify some limitations of MarkerScan...  PMC11016177   \n",
       "3   Algorithm 1A single iteration of the MaBoSS s...  PMC11127412   \n",
       "4   Author contributions: Conceptualization: Shuc...  PMC11340858   \n",
       "\n",
       "                                            ner_ines  \n",
       "0                                               None  \n",
       "1                              [[21, 26, MAGMA, BT]]  \n",
       "2  [[36, 46, MarkerScan, BT], [128, 138, MarkerSc...  \n",
       "3                             [[38, 44, MaBoSS, BT]]  \n",
       "4                                               None  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_df['ner_ines'] = grouped_df['ner_ines'].apply(\n",
    "    lambda x: [[item[0], item[1], item[2], 'BT'] for item in x] if x else None\n",
    ")\n",
    "grouped_df.reset_index(drop=True, inplace=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "869b7c99",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "561\n",
      "560\n"
     ]
    }
   ],
   "source": [
    "# Avoid PMCDID leak\n",
    "grouped_df.sort_values(by='PMCID', inplace=True)\n",
    "\n",
    "# Split the data into train and test sets\n",
    "train_df, test_df = train_test_split(grouped_df, test_size=0.2, random_state=42, shuffle=False)\n",
    "print(len(test_df))\n",
    "\n",
    "# Check if PMCID from train is in test, then drop those rows from test\n",
    "test_df = test_df[~test_df['PMCID'].isin(train_df['PMCID'])]\n",
    "print(len(test_df))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "700fd744",
   "metadata": {},
   "outputs": [],
   "source": [
    "train_df = train_df.drop(columns=['PMCID']).sample(frac=1, random_state=42).reset_index(drop=True)\n",
    "test_df = test_df.drop(columns=['PMCID']).sample(frac=1, random_state=42).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "efebfe21",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2239</th>\n",
       "      <td>The CadA protein 3D structure model quality wa...</td>\n",
       "      <td>[[63, 68, PyMOL, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2240</th>\n",
       "      <td>We performed simulations to evaluate power and...</td>\n",
       "      <td>[[63, 72, trans-PCO, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2241</th>\n",
       "      <td>The exclusion of indels by bcftools also leads...</td>\n",
       "      <td>[[27, 35, BCFtools, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2242</th>\n",
       "      <td>The five genomes for benchmarking are availabl...</td>\n",
       "      <td>[[78, 85, MerCat2, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2243</th>\n",
       "      <td>Response: We appreciate the reviewer’s insight...</td>\n",
       "      <td>[[73, 80, KinCytE, BT]]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sentence  \\\n",
       "2239  The CadA protein 3D structure model quality wa...   \n",
       "2240  We performed simulations to evaluate power and...   \n",
       "2241  The exclusion of indels by bcftools also leads...   \n",
       "2242  The five genomes for benchmarking are availabl...   \n",
       "2243  Response: We appreciate the reviewer’s insight...   \n",
       "\n",
       "                       ner_ines  \n",
       "2239      [[63, 68, PyMOL, BT]]  \n",
       "2240  [[63, 72, trans-PCO, BT]]  \n",
       "2241   [[27, 35, BCFtools, BT]]  \n",
       "2242    [[78, 85, MerCat2, BT]]  \n",
       "2243    [[73, 80, KinCytE, BT]]  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_df.tail(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "77e98ffe",
   "metadata": {},
   "outputs": [],
   "source": [
    "test_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac80e692",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Example usage:\n",
    "output_folder = '../data/IOB/'\n",
    "\n",
    "# Convert train, dev, and test dataframes to IOB format\n",
    "ml_tools.convert_to_IOB_format_from_df(train_df, output_folder, 'train_IOB.tsv')\n",
    "ml_tools.convert_to_IOB_format_from_df(test_df, output_folder, 'dev_IOB.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c335960",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Example usage:\n",
    "train_files = [output_folder+'train_IOB.tsv']\n",
    "dev_files = [output_folder+'dev_IOB.tsv']\n",
    "\n",
    "ml_tools.check_integrity_of_files(train_files, dev_files,dev_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "372e935a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bh24-literature-mining-Vwh2BCYY-py3.12",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
