{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c030b4b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/nfs/production/literature/santosh_tirunagari/ENVS/biohackathon_env/bin/pip\r\n"
     ]
    }
   ],
   "source": [
    "!which pip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "913c1497",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c1dafe59",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package punkt to /homes/stirunag/nltk_data...\n",
      "[nltk_data]   Package punkt is already up-to-date!\n"
     ]
    }
   ],
   "source": [
    "# !pip install scikit-learn\n",
    "import ast\n",
    "import pandas as pd\n",
    "from sklearn.model_selection import train_test_split\n",
    "import sys\n",
    "\n",
    "# Add the directory to sys.path\n",
    "sys.path.append('../../biohackathon2024/src/bh24_literature_mining')\n",
    "\n",
    "# Now import ml_tools\n",
    "import ml_tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "acb2ad01",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Using this technique, we identified 445 protei...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(233, 242, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(54, 63, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(120, 129, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(72, 81, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC10832249</td>\n",
       "      <td>MetaPathways v2.5 [37] was used to create the ...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(0, 12, 'MetaPathways', 'metapathways')</td>\n",
       "      <td>Metagenomics, Microbial ecology, Population ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2021</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>A complete interactive example can be assessed...</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>(82, 93, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2022</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The central objective behind the development o...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(48, 59, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2023</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The steps to contributing to the project are d...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(103, 114, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2024</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>scalepopgen is hosted on GitHub under the popg...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(0, 11, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2025</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The results of the case study described in the...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(97, 108, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2026 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID                                           Sentence  True?  \\\n",
       "0     PMC11286849  Using this technique, we identified 445 protei...   True   \n",
       "1     PMC11286849  The 445 identified proteins were searched agai...   True   \n",
       "2     PMC11286849  The identified proteins also included 52 (70%)...   True   \n",
       "3     PMC11286849  The localization of 52 of these 445 proteins i...   True   \n",
       "4     PMC10832249  MetaPathways v2.5 [37] was used to create the ...   True   \n",
       "...           ...                                                ...    ...   \n",
       "2021  PMC10994858  A complete interactive example can be assessed...  False   \n",
       "2022  PMC10994858  The central objective behind the development o...   True   \n",
       "2023  PMC10994858  The steps to contributing to the project are d...   True   \n",
       "2024  PMC10994858  scalepopgen is hosted on GitHub under the popg...   True   \n",
       "2025  PMC10994858  The results of the case study described in the...   True   \n",
       "\n",
       "      False?                                  NER_Tags  \\\n",
       "0      False      (233, 242, 'SubtiWiki', 'subtiwiki')   \n",
       "1      False        (54, 63, 'SubtiWiki', 'subtiwiki')   \n",
       "2      False      (120, 129, 'SubtiWiki', 'subtiwiki')   \n",
       "3      False        (72, 81, 'SubtiWiki', 'subtiwiki')   \n",
       "4      False   (0, 12, 'MetaPathways', 'metapathways')   \n",
       "...      ...                                       ...   \n",
       "2021   False    (82, 93, 'scalepopgen', 'scalepopgen')   \n",
       "2022   False    (48, 59, 'scalepopgen', 'scalepopgen')   \n",
       "2023   False  (103, 114, 'scalepopgen', 'scalepopgen')   \n",
       "2024   False     (0, 11, 'scalepopgen', 'scalepopgen')   \n",
       "2025   False   (97, 108, 'scalepopgen', 'scalepopgen')   \n",
       "\n",
       "                                                 Topics  \n",
       "0     Molecular interactions, pathways and networks,...  \n",
       "1     Molecular interactions, pathways and networks,...  \n",
       "2     Molecular interactions, pathways and networks,...  \n",
       "3     Molecular interactions, pathways and networks,...  \n",
       "4     Metagenomics, Microbial ecology, Population ge...  \n",
       "...                                                 ...  \n",
       "2021  Workflows, Population genomics, Metagenomics, ...  \n",
       "2022  Workflows, Population genomics, Metagenomics, ...  \n",
       "2023  Workflows, Population genomics, Metagenomics, ...  \n",
       "2024  Workflows, Population genomics, Metagenomics, ...  \n",
       "2025  Workflows, Population genomics, Metagenomics, ...  \n",
       "\n",
       "[2026 rows x 6 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "sheet_url = 'https://docs.google.com/spreadsheets/d/13BnWa9_r-8aVkbWSnuInvyhI41_WEALMBfT9X3frekI/export?format=csv&gid=79076201'\n",
    "\n",
    "# Read the sheet into a DataFrame\n",
    "df = pd.read_csv(sheet_url)\n",
    "\n",
    "# Optionally, convert 'TRUE' and 'FALSE' to boolean\n",
    "df['True?'] = df['True?'].apply(lambda x: True if x == 'TRUE' else (False if x == 'FALSE' else x))\n",
    "df['False?'] = df['False?'].apply(lambda x: True if x == 'TRUE' else (False if x == 'FALSE' else x))\n",
    "\n",
    "\n",
    "# Display the DataFrame\n",
    "df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "010bee4e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>Using this technique, we identified 445 protei...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(233, 242, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(54, 63, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(120, 129, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PMC11286849</td>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(72, 81, 'SubtiWiki', 'subtiwiki')</td>\n",
       "      <td>Molecular interactions, pathways and networks,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PMC10832249</td>\n",
       "      <td>MetaPathways v2.5 [37] was used to create the ...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(0, 12, 'MetaPathways', 'metapathways')</td>\n",
       "      <td>Metagenomics, Microbial ecology, Population ge...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>PMC10988918</td>\n",
       "      <td>The CITEViz documentation website can be acces...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(4, 11, 'CITEViz', 'citeviz')</td>\n",
       "      <td>Cytometry, Transcriptomics, Workflows, RNA-Seq...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2022</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The central objective behind the development o...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(48, 59, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2023</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The steps to contributing to the project are d...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(103, 114, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2024</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>scalepopgen is hosted on GitHub under the popg...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(0, 11, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2025</th>\n",
       "      <td>PMC10994858</td>\n",
       "      <td>The results of the case study described in the...</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>(97, 108, 'scalepopgen', 'scalepopgen')</td>\n",
       "      <td>Workflows, Population genomics, Metagenomics, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>243 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID                                           Sentence True?  \\\n",
       "0     PMC11286849  Using this technique, we identified 445 protei...  True   \n",
       "1     PMC11286849  The 445 identified proteins were searched agai...  True   \n",
       "2     PMC11286849  The identified proteins also included 52 (70%)...  True   \n",
       "3     PMC11286849  The localization of 52 of these 445 proteins i...  True   \n",
       "4     PMC10832249  MetaPathways v2.5 [37] was used to create the ...  True   \n",
       "...           ...                                                ...   ...   \n",
       "1088  PMC10988918  The CITEViz documentation website can be acces...  True   \n",
       "2022  PMC10994858  The central objective behind the development o...  True   \n",
       "2023  PMC10994858  The steps to contributing to the project are d...  True   \n",
       "2024  PMC10994858  scalepopgen is hosted on GitHub under the popg...  True   \n",
       "2025  PMC10994858  The results of the case study described in the...  True   \n",
       "\n",
       "      False?                                  NER_Tags  \\\n",
       "0      False      (233, 242, 'SubtiWiki', 'subtiwiki')   \n",
       "1      False        (54, 63, 'SubtiWiki', 'subtiwiki')   \n",
       "2      False      (120, 129, 'SubtiWiki', 'subtiwiki')   \n",
       "3      False        (72, 81, 'SubtiWiki', 'subtiwiki')   \n",
       "4      False   (0, 12, 'MetaPathways', 'metapathways')   \n",
       "...      ...                                       ...   \n",
       "1088   False             (4, 11, 'CITEViz', 'citeviz')   \n",
       "2022   False    (48, 59, 'scalepopgen', 'scalepopgen')   \n",
       "2023   False  (103, 114, 'scalepopgen', 'scalepopgen')   \n",
       "2024   False     (0, 11, 'scalepopgen', 'scalepopgen')   \n",
       "2025   False   (97, 108, 'scalepopgen', 'scalepopgen')   \n",
       "\n",
       "                                                 Topics  \n",
       "0     Molecular interactions, pathways and networks,...  \n",
       "1     Molecular interactions, pathways and networks,...  \n",
       "2     Molecular interactions, pathways and networks,...  \n",
       "3     Molecular interactions, pathways and networks,...  \n",
       "4     Metagenomics, Microbial ecology, Population ge...  \n",
       "...                                                 ...  \n",
       "1088  Cytometry, Transcriptomics, Workflows, RNA-Seq...  \n",
       "2022  Workflows, Population genomics, Metagenomics, ...  \n",
       "2023  Workflows, Population genomics, Metagenomics, ...  \n",
       "2024  Workflows, Population genomics, Metagenomics, ...  \n",
       "2025  Workflows, Population genomics, Metagenomics, ...  \n",
       "\n",
       "[243 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df['False?'] == True, 'NER_Tags'] = None\n",
    "\n",
    "true_checked_df = df[(df['True?'] == True) | (df['False?'] == True)]\n",
    "\n",
    "# true_checked_df = df[(df['True?']| (df['False?']]\n",
    "\n",
    "true_checked_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1bac9831",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>True?</th>\n",
       "      <th>False?</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>PMC11466383</td>\n",
       "      <td>The Big Skin Health Intervention Fiji Trial (“...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Cell biology, Imaging</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>PMC11466383</td>\n",
       "      <td>We extrapolated the total annual number of cas...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Cell biology, Imaging</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>PMC11466383</td>\n",
       "      <td>The estimated annual healthcare costs of scabi...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Cell biology, Imaging</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>PMC11466383</td>\n",
       "      <td>Scabies and related SSTIs lead to a heavy econ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Cell biology, Imaging</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>PMC11466383</td>\n",
       "      <td>This data is the same as supplement to Hospita...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Cell biology, Imaging</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>848</th>\n",
       "      <td>PMC11530086</td>\n",
       "      <td>Regarding our LOS analyses, we found Lowest MA...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Phylogenetics, Sequence analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>849</th>\n",
       "      <td>PMC11530086</td>\n",
       "      <td>While IOH had less influence than length of su...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Phylogenetics, Sequence analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>850</th>\n",
       "      <td>PMC11530086</td>\n",
       "      <td>Multivariable regression analyses revealed a s...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Phylogenetics, Sequence analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>851</th>\n",
       "      <td>PMC11530086</td>\n",
       "      <td>In this study, we found a three-phase associat...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Phylogenetics, Sequence analysis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>852</th>\n",
       "      <td>PMC11530086</td>\n",
       "      <td>In this retrospective trial, “Cumulative time ...</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>Phylogenetics, Sequence analysis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>156 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           PMCID                                           Sentence  True?  \\\n",
       "49   PMC11466383  The Big Skin Health Intervention Fiji Trial (“...  False   \n",
       "50   PMC11466383  We extrapolated the total annual number of cas...  False   \n",
       "51   PMC11466383  The estimated annual healthcare costs of scabi...  False   \n",
       "52   PMC11466383  Scabies and related SSTIs lead to a heavy econ...  False   \n",
       "53   PMC11466383  This data is the same as supplement to Hospita...  False   \n",
       "..           ...                                                ...    ...   \n",
       "848  PMC11530086  Regarding our LOS analyses, we found Lowest MA...  False   \n",
       "849  PMC11530086  While IOH had less influence than length of su...  False   \n",
       "850  PMC11530086  Multivariable regression analyses revealed a s...  False   \n",
       "851  PMC11530086  In this study, we found a three-phase associat...  False   \n",
       "852  PMC11530086  In this retrospective trial, “Cumulative time ...  False   \n",
       "\n",
       "     False? NER_Tags                            Topics  \n",
       "49     True     None             Cell biology, Imaging  \n",
       "50     True     None             Cell biology, Imaging  \n",
       "51     True     None             Cell biology, Imaging  \n",
       "52     True     None             Cell biology, Imaging  \n",
       "53     True     None             Cell biology, Imaging  \n",
       "..      ...      ...                               ...  \n",
       "848    True     None  Phylogenetics, Sequence analysis  \n",
       "849    True     None  Phylogenetics, Sequence analysis  \n",
       "850    True     None  Phylogenetics, Sequence analysis  \n",
       "851    True     None  Phylogenetics, Sequence analysis  \n",
       "852    True     None  Phylogenetics, Sequence analysis  \n",
       "\n",
       "[156 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_checked_df[true_checked_df['False?'] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2fa9c575",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Using this technique, we identified 445 protei...</td>\n",
       "      <td>(233, 242, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>(54, 63, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>(120, 129, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>(72, 81, SubtiWiki, subtiwiki)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MetaPathways v2.5 [37] was used to create the ...</td>\n",
       "      <td>(0, 12, MetaPathways, metapathways)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>The CITEViz documentation website can be acces...</td>\n",
       "      <td>(4, 11, CITEViz, citeviz)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2022</th>\n",
       "      <td>The central objective behind the development o...</td>\n",
       "      <td>(48, 59, scalepopgen, scalepopgen)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2023</th>\n",
       "      <td>The steps to contributing to the project are d...</td>\n",
       "      <td>(103, 114, scalepopgen, scalepopgen)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2024</th>\n",
       "      <td>scalepopgen is hosted on GitHub under the popg...</td>\n",
       "      <td>(0, 11, scalepopgen, scalepopgen)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2025</th>\n",
       "      <td>The results of the case study described in the...</td>\n",
       "      <td>(97, 108, scalepopgen, scalepopgen)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>243 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sentence  \\\n",
       "0     Using this technique, we identified 445 protei...   \n",
       "1     The 445 identified proteins were searched agai...   \n",
       "2     The identified proteins also included 52 (70%)...   \n",
       "3     The localization of 52 of these 445 proteins i...   \n",
       "4     MetaPathways v2.5 [37] was used to create the ...   \n",
       "...                                                 ...   \n",
       "1088  The CITEViz documentation website can be acces...   \n",
       "2022  The central objective behind the development o...   \n",
       "2023  The steps to contributing to the project are d...   \n",
       "2024  scalepopgen is hosted on GitHub under the popg...   \n",
       "2025  The results of the case study described in the...   \n",
       "\n",
       "                                  ner_ines  \n",
       "0         (233, 242, SubtiWiki, subtiwiki)  \n",
       "1           (54, 63, SubtiWiki, subtiwiki)  \n",
       "2         (120, 129, SubtiWiki, subtiwiki)  \n",
       "3           (72, 81, SubtiWiki, subtiwiki)  \n",
       "4      (0, 12, MetaPathways, metapathways)  \n",
       "...                                    ...  \n",
       "1088             (4, 11, CITEViz, citeviz)  \n",
       "2022    (48, 59, scalepopgen, scalepopgen)  \n",
       "2023  (103, 114, scalepopgen, scalepopgen)  \n",
       "2024     (0, 11, scalepopgen, scalepopgen)  \n",
       "2025   (97, 108, scalepopgen, scalepopgen)  \n",
       "\n",
       "[243 rows x 2 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "true_checked_df = true_checked_df[['Sentence','NER_Tags']]\n",
    "\n",
    "transformed_df = true_checked_df.rename(columns={\n",
    "    'Sentence': 'sentence',\n",
    "    'NER_Tags': 'ner_ines'\n",
    "})\n",
    "\n",
    "transformed_df['ner_ines'] = transformed_df['ner_ines'].apply(lambda x: ast.literal_eval(x) if isinstance(x, str) else x)\n",
    "\n",
    "transformed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0c6da881",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(233, 242, 'SubtiWiki', 'subtiwiki')"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transformed_df['ner_ines'].iloc[0]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e5f70ddf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Using this technique, we identified 445 protei...</td>\n",
       "      <td>[[233, 242, SubtiWiki, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>The 445 identified proteins were searched agai...</td>\n",
       "      <td>[[54, 63, SubtiWiki, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>The identified proteins also included 52 (70%)...</td>\n",
       "      <td>[[120, 129, SubtiWiki, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>The localization of 52 of these 445 proteins i...</td>\n",
       "      <td>[[72, 81, SubtiWiki, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MetaPathways v2.5 [37] was used to create the ...</td>\n",
       "      <td>[[0, 12, MetaPathways, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1088</th>\n",
       "      <td>The CITEViz documentation website can be acces...</td>\n",
       "      <td>[[4, 11, CITEViz, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2022</th>\n",
       "      <td>The central objective behind the development o...</td>\n",
       "      <td>[[48, 59, scalepopgen, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2023</th>\n",
       "      <td>The steps to contributing to the project are d...</td>\n",
       "      <td>[[103, 114, scalepopgen, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2024</th>\n",
       "      <td>scalepopgen is hosted on GitHub under the popg...</td>\n",
       "      <td>[[0, 11, scalepopgen, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2025</th>\n",
       "      <td>The results of the case study described in the...</td>\n",
       "      <td>[[97, 108, scalepopgen, BT]]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>243 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sentence  \\\n",
       "0     Using this technique, we identified 445 protei...   \n",
       "1     The 445 identified proteins were searched agai...   \n",
       "2     The identified proteins also included 52 (70%)...   \n",
       "3     The localization of 52 of these 445 proteins i...   \n",
       "4     MetaPathways v2.5 [37] was used to create the ...   \n",
       "...                                                 ...   \n",
       "1088  The CITEViz documentation website can be acces...   \n",
       "2022  The central objective behind the development o...   \n",
       "2023  The steps to contributing to the project are d...   \n",
       "2024  scalepopgen is hosted on GitHub under the popg...   \n",
       "2025  The results of the case study described in the...   \n",
       "\n",
       "                           ner_ines  \n",
       "0       [[233, 242, SubtiWiki, BT]]  \n",
       "1         [[54, 63, SubtiWiki, BT]]  \n",
       "2       [[120, 129, SubtiWiki, BT]]  \n",
       "3         [[72, 81, SubtiWiki, BT]]  \n",
       "4       [[0, 12, MetaPathways, BT]]  \n",
       "...                             ...  \n",
       "1088         [[4, 11, CITEViz, BT]]  \n",
       "2022    [[48, 59, scalepopgen, BT]]  \n",
       "2023  [[103, 114, scalepopgen, BT]]  \n",
       "2024     [[0, 11, scalepopgen, BT]]  \n",
       "2025   [[97, 108, scalepopgen, BT]]  \n",
       "\n",
       "[243 rows x 2 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transformed_df['ner_ines'] = transformed_df['ner_ines'].apply(lambda x: [[x[0], x[1], x[2], 'BT']] if x else None)\n",
    "transformed_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "a9fc967b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sentence    Lastly, we summed up the number of cases for t...\n",
       "ner_ines                                                 None\n",
       "Name: 81, dtype: object"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transformed_df.iloc[52]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "869b7c99",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sentence</th>\n",
       "      <th>ner_ines</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>535</th>\n",
       "      <td>There are studies analysing the iron content i...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>2CITEViz analysis of PBMC CITE-Seq data.</td>\n",
       "      <td>[[1, 8, CITEViz, BT]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>LOS: length of stay, PACU: postoperative care ...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>552</th>\n",
       "      <td>Continuous detection of the HUVEC monolayer fo...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504</th>\n",
       "      <td>Two days after cell seeding and continuous cul...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sentence               ner_ines\n",
       "535   There are studies analysing the iron content i...                   None\n",
       "1057           2CITEViz analysis of PBMC CITE-Seq data.  [[1, 8, CITEViz, BT]]\n",
       "835   LOS: length of stay, PACU: postoperative care ...                   None\n",
       "552   Continuous detection of the HUVEC monolayer fo...                   None\n",
       "504   Two days after cell seeding and continuous cul...                   None"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_df, test_df = train_test_split(transformed_df, test_size=0.2, random_state=42)\n",
    "\n",
    "train_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ac80e692",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Processing batches: 100%|██████████████████| 1/1 [00:00<00:00, 130.06it/s]\n",
      "Processing batches: 100%|██████████████████| 1/1 [00:00<00:00, 581.57it/s]\n"
     ]
    }
   ],
   "source": [
    "# Example usage:\n",
    "output_folder = '../data/IOB/'\n",
    "\n",
    "# Convert train, dev, and test dataframes to IOB format\n",
    "ml_tools.convert_to_IOB_format_from_df(train_df, output_folder, 'train_IOB.tsv')\n",
    "ml_tools.convert_to_IOB_format_from_df(test_df, output_folder, 'dev_IOB.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "3c335960",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Checking Dataset 1:\n",
      "Train set 1 is valid.\n",
      "Dev set 1 is valid.\n",
      "Test set 1 is valid.\n"
     ]
    }
   ],
   "source": [
    "# Example usage:\n",
    "train_files = [output_folder+'train_IOB.tsv']\n",
    "dev_files = [output_folder+'dev_IOB.tsv']\n",
    "\n",
    "ml_tools.check_integrity_of_files(train_files, dev_files,dev_files)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "372e935a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "BioHackathon 2024",
   "language": "python",
   "name": "biohackathon-2024"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
