{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Example notebook\n",
    "\n",
    "Here we get the Article classes for all given bio.tools tools."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15005\n"
     ]
    }
   ],
   "source": [
    "from bh24_literature_mining.utils import load_biotools_pub\n",
    "from bh24_literature_mining.biotools import Tool_entry\n",
    "\n",
    "tools = load_biotools_pub(\"../biotoolspub/biotoolspub.tsv\")\n",
    "\n",
    "# Initialize set to track unique names and a list to store Biotool objects\n",
    "tools_lower = set()\n",
    "unique_biotools = []\n",
    "\n",
    "# Iterate over the rows in the DataFrame\n",
    "for _, row in tools.iterrows():\n",
    "    name = row[\"name\"]\n",
    "    biotools_id = row[\"biotoolsID\"]\n",
    "    name_lower = name.lower()\n",
    "\n",
    "    # Only add unique tools based on name\n",
    "    if name_lower not in tools_lower:\n",
    "        tools_lower.add(name_lower)  # Add to the set to track uniqueness\n",
    "        # Create a Biotool object and add it to the list\n",
    "        biotool = Tool_entry(biotool_id=biotools_id, name=name)\n",
    "        unique_biotools.append(biotool)\n",
    "\n",
    "\n",
    "print(len(unique_biotools))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example on how to use europepmc_api"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No relevant paragraphs found iPHoP\n",
      "No relevant paragraphs found SLiM\n",
      "No relevant paragraphs found BCFtools\n",
      "No relevant paragraphs found AluMine\n",
      "No relevant paragraphs found ARTEM\n",
      "No relevant paragraphs found DataDiscovery\n",
      "No relevant paragraphs found ADAPTIVE\n",
      "No relevant paragraphs found Marine Metagenomics Portal\n",
      "No relevant paragraphs found mi-CNN\n",
      "No relevant paragraphs found SpaceM\n",
      "No relevant paragraphs found SPUTNIK\n",
      "No relevant paragraphs found PatchDock\n",
      "No relevant paragraphs found Reactome\n",
      "No relevant paragraphs found MAGMA\n",
      "No relevant paragraphs found DISGENET\n",
      "No relevant paragraphs found Phobius (EBI)\n",
      "No relevant paragraphs found hmrbase\n",
      "No relevant paragraphs found CDpred\n",
      "No relevant paragraphs found CancerLivER\n",
      "No relevant paragraphs found Stitch\n",
      "No relevant paragraphs found JCat\n",
      "No relevant paragraphs found Expasy\n",
      "No relevant paragraphs found ENDscript 2.0\n",
      "No relevant paragraphs found Smoother\n",
      "No relevant paragraphs found iVar\n"
     ]
    }
   ],
   "source": [
    "from pathlib import Path\n",
    "from bh24_literature_mining.europepmc_api import EuropePMCClient, write_tool_mentions_to_file\n",
    "from bh24_literature_mining.europepmc_api import identify_tool_mentions_in_sentences\n",
    "import pandas as pd\n",
    "\n",
    "results_list = []\n",
    "\n",
    "client = EuropePMCClient()\n",
    "\n",
    "for tool in unique_biotools[:100]:\n",
    "    # Call bio.tools query and get a list of Article objects\n",
    "    \n",
    "    biotools_articles = client.search_mentions(tool.name)\n",
    "\n",
    "    if len(biotools_articles) == 0:\n",
    "        continue\n",
    "    first_article = biotools_articles[0]\n",
    "\n",
    "    relevant_parahraphs = client.get_relevant_paragraphs(first_article.pmcid, tool.name)\n",
    "    if len(relevant_parahraphs) == 0:\n",
    "        print(\"No relevant paragraphs found\", tool.name)\n",
    "        continue\n",
    "    \n",
    "    tool.get_topics()\n",
    "\n",
    "\n",
    "    result = identify_tool_mentions_in_sentences(first_article.pmcid, tool, relevant_parahraphs)\n",
    "    results_list.extend(result)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['PMC11286849',\n",
       "  'Using this technique, we identified 445 proteins with high confidence from trace amounts of highly pure spore preparations, including 52 of the 79 proteins (approximately 70%) previously demonstrated to be localized in spores in the SubtiWiki database and detected through direct protein analysis.',\n",
       "  [(233, 242, 'SubtiWiki', 'subtiwiki')],\n",
       "  'Molecular interactions, pathways and networks, Gene expression, Model organisms, Endocrinology and metabolism, Phylogenetics'],\n",
       " ['PMC11286849',\n",
       "  'The 445 identified proteins were searched against the SubtiWiki and UniProt B.',\n",
       "  [(54, 63, 'SubtiWiki', 'subtiwiki')],\n",
       "  'Molecular interactions, pathways and networks, Gene expression, Model organisms, Endocrinology and metabolism, Phylogenetics'],\n",
       " ['PMC11286849',\n",
       "  'The identified proteins also included 52 (70%) of the 79 proteins previously annotated as \"sporulation proteins\" in the SubtiWiki database and found in direct protein analysis (marked in yellow in the Supplementary Table S4).',\n",
       "  [(120, 129, 'SubtiWiki', 'subtiwiki')],\n",
       "  'Molecular interactions, pathways and networks, Gene expression, Model organisms, Endocrinology and metabolism, Phylogenetics'],\n",
       " ['PMC11286849',\n",
       "  'The localization of 52 of these 445 proteins in spores is listed in the SubtiWiki database (Supplementary Table S5).',\n",
       "  [(72, 81, 'SubtiWiki', 'subtiwiki')],\n",
       "  'Molecular interactions, pathways and networks, Gene expression, Model organisms, Endocrinology and metabolism, Phylogenetics'],\n",
       " ['PMC10832249',\n",
       "  'MetaPathways v2.5 [37] was used to create the benchmark CAMI environment PGDB (ePGDB) which are PGDBs for microbial communities [38].',\n",
       "  [(0, 12, 'MetaPathways', 'metapathways')],\n",
       "  'Metagenomics, Microbial ecology, Population genomics'],\n",
       " ['PMC10832249',\n",
       "  'MetaPathways utilizes a modified version of PathoLogic for pathway prediction.',\n",
       "  [(0, 12, 'MetaPathways', 'metapathways')],\n",
       "  'Metagenomics, Microbial ecology, Population genomics'],\n",
       " ['PMC10832249',\n",
       "  'PathoLogic was not included in the comparison since MetaPathways uses it to create the ePGDB.',\n",
       "  [(52, 64, 'MetaPathways', 'metapathways')],\n",
       "  'Metagenomics, Microbial ecology, Population genomics'],\n",
       " ['PMC10832249',\n",
       "  'One limitation of the CAMI ePGDB as a benchmark is that it is automatically generated using MetaPathways but the predictions have not been curated so it can be said that the results demonstrate more how similar the other prediction methods are with MetaPathways and PathoLogic than their actual prediction performance.',\n",
       "  [(92, 104, 'MetaPathways', 'metapathways')],\n",
       "  'Metagenomics, Microbial ecology, Population genomics'],\n",
       " ['PMC11045623',\n",
       "  'We used Orbitrap HRMS data of surface water extracts to compare the Automated Target Screening (ATS) workflow with data evaluations performed with the vendor software TraceFinder and the established semi-automated analysis workflow in the MZmine software.',\n",
       "  [(239, 245, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'The ATS approach increased the overall evaluation performance of the peak annotation compared to the established MZmine module without the need for any post-hoc corrections.',\n",
       "  [(113, 119, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'The need for higher throughput data processing in NTS and SS led to the creation of software such as MZmine, which has been used for partially or fully automatic data processing including smoothing, spectra deconvolution, peak detections, and gap filling with sufficient throughput [12, 13].',\n",
       "  [(101, 107, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'In addition to NTS and SS, MZmine can also be used to perform targeted analysis by incorporating user-defined lists of target features and retention times to annotate peaks.',\n",
       "  [(27, 33, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'In this study, we introduce an open-source unsupervised workflow for Automated Target Screening (ATS) of both GC-HRMS and LC-HRMS data and validate its performance against a manual evaluation in TraceFinder (TF) and batch-mode evaluation in MZmine using a real-life data set.',\n",
       "  [(241, 247, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'MZmine (version 2.51) was used for comparing the peak annotation performance of ATS to a sophisticated (semi-)automated analysis workflow.',\n",
       "  [(0, 6, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'MZmine 2.51 was used in this comparison, although a newer version MZmine 3 has been launched during the preparation of the present study.',\n",
       "  [(0, 6, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'As MZmine 3 still uses the same peak detection algorithms, we consider that the results using the older version are comparable [15].',\n",
       "  [(3, 9, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'The details regarding sample extraction and settings of TraceFinder and MZmine can be accessed from Text S1 to Text S3 in Supporting Information S1.',\n",
       "  [(72, 78, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'These “CHECK” cases are similar to the peak status “ESTIMATED” in MZmine.',\n",
       "  [(66, 72, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'Compounds were selected for peak annotation performance comparison if they had at least one annotated response in one of the three evaluation workflows TraceFinder, ATS, or MZmine.',\n",
       "  [(173, 179, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'The peak annotation was compared by calculating the fractions of false positives (FP), false negatives (FN), true positives (TP), and true negatives (TN) of all observations n for MZmine and ATS compared to the manual evaluation in TraceFinder.',\n",
       "  [(180, 186, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'Only features with the “DETECTED” peak status were included in MZmine.',\n",
       "  [(63, 69, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'The highest number of chemicals was identified in TF while ATS and MZmine have a lower number but similar patterns of detected chemicals.',\n",
       "  [(67, 73, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'The comparison of the peak annotation (see Supporting Information S2, Table S2-4) via ATS_raw, ATS_mzML, and MZmine with selected key figures is listed in Table 1.',\n",
       "  [(109, 115, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'For the MZmine analysis of the GC dataset, two sets of retention times were used, (i) a list with retention times as used by ATS with shifts in the range of −1.1 to −0.9 min simulating the case of shifted and not optimized analysis conditions and (ii) the same list of analytes but with exact, shift-corrected retention times as used with TF.',\n",
       "  [(8, 14, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  ' Table 1Results of the peak annotation comparison of ATS and MZmineInstrument modeWorkflowTPTNFPFNSensitivitySpecificityAccuracyLC-ESIposATS_raw61110291421300.820.880.86ATS_mzML5161086872230.700.930.84MZmine49710271312570.660.890.80LC-ESInegATS_raw20333483280.880.800.83ATS_mzML17235665580.740.850.81MZmine18431775720.720.810.77GC-EIATS_raw429165691210.780.710.76ATS_mzML382171631680.690.730.71MZmine362091723670.090.540.31MZmine_sc*365163711850.660.700.67*sc shift-corrected, TP true positives, TN true negatives, FP false positives, FN false negatives.',\n",
       "  [(61, 67, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'Results of the peak annotation comparison of ATS and MZmine',\n",
       "  [(53, 59, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'The greatest improvement was observed for the sensitivity, which increased around 1.3-fold if ATS_raw is used compared to MZmine.',\n",
       "  [(122, 128, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'However, this could only be achieved for the shift-corrected dataset for MZmine, where the retention times were adjusted a priori to exactly match the correct times as used in TF.',\n",
       "  [(73, 79, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'ATS increased overall performance and accuracy compared to MZmine (see Table 1).',\n",
       "  [(59, 65, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'While some key numbers like sensitivity and specificity or even accuracy are in a similar range for ATS and MZmine (see Table 1), one has to consider that ATS was using a target feature list which was randomly shifted in the range of −1.1 and −0.9 min while MZmine was only performing well if retention times were exactly as observed for the datasets.',\n",
       "  [(108, 114, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'For MZmine annotation, retention times of the compound list have to be adjusted prior to annotation in another software (e.g., R or MS Excel), and in TF all expected retention times can be adjusted after visual inspection globally in the method or in relation to internal standards serving as retention time reference.',\n",
       "  [(4, 10, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'This was clearly demonstrated for the GC-EI dataset, which performed similarly for the shift-corrected MZmine dataset and the artificially shifted ATS datasets (see Table 1).',\n",
       "  [(103, 109, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'In this case, MZmine lists and assigns both peaks to both compounds in the final feature list, while ATS was able to split the peaks and assign the chemicals correctly (see Supporting Information S1, Figure S1-3A).',\n",
       "  [(14, 20, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'One strategy would be to assign three different features (peaks 1–3) and evaluate them independently in MZmine or manually integrate the three peaks together in TF.',\n",
       "  [(104, 110, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'ATS correctly identified and assigned the respective peak of the EIC, whereas MZmine was only able to do so if the retention times had been accurately shift-corrected and small RT windows in advance.',\n",
       "  [(78, 84, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'Otherwise, the peak finder of MZmine detects two peaks which have a similar peak pattern but the wrong retention time.',\n",
       "  [(30, 36, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11045623',\n",
       "  'ATS is able to increase the overall peak annotation accuracy and outperforms sophisticated workflows like MZmine for challenging scenarios like shifted retention times or co-eluting analytes.',\n",
       "  [(106, 112, 'MZmine', 'mzmine')],\n",
       "  'Proteomics, Metabolomics, Proteomics experiment, Small molecules'],\n",
       " ['PMC11449484',\n",
       "  'coli (strain s_6558), Klebsiella pneumoniae (strain 4300STDY7045912), and Salmonella enterica (strain 350627) were compared with Jalview.',\n",
       "  [(129, 136, 'Jalview', 'Jalview')],\n",
       "  'Sequence analysis, Data visualisation'],\n",
       " ['PMC11449484',\n",
       "  'coli were compared using Jalview.',\n",
       "  [(25, 32, 'Jalview', 'Jalview')],\n",
       "  'Sequence analysis, Data visualisation'],\n",
       " ['PMC11449484',\n",
       "  'Jalview was used for editing and viewing sequence alignments (Waterhouse et al., 2009).',\n",
       "  [(0, 7, 'Jalview', 'Jalview')],\n",
       "  'Sequence analysis, Data visualisation'],\n",
       " ['PMC11373567',\n",
       "  'To address the current digital divide amongst LMICs, and between LMICs and HICs, GGN recommended the Leiden Open Variation Database (LOVD) and the ITHANET Portal as platforms for data collection and data sharing.',\n",
       "  [(147, 154, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11373567',\n",
       "  'By contrast, ITHANET (https://www.ithanet.eu/; accessed on 17 April 2024) is a web portal initially conceived specifically for hemoglobinopathies, which, as of this writing, holds in its databases sequence and functional annotation for 3462 variants in 538 genes.',\n",
       "  [(13, 20, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11373567',\n",
       "  'More recently, ITHANET launched new databases, namely IthaPhen [71] for genotype-phenotype correlation and IthaCNVs [72] for improved diagnosis of copy number variants in hemoglobinopathies.',\n",
       "  [(15, 22, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11373567',\n",
       "  'The enormous diversity of causative β-globin mutations and modifier polymorphisms may be illustrated by using quick filters on ITHANET (www.ithanet.eu, accessed 17 April 2024).',\n",
       "  [(127, 134, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11373567',\n",
       "  'Accordingly, for the quick filter “Haemoglobinopathy/β-thalassaemia” on ITHANET, 491 variants were found that affect the HBB gene and cause β-thalassemia (https://www.ithanet.eu/db/ithagenes?action=list&hem=2), whereas a quick filter for “Functionality/Disease modifying mutations” revealed 886 entries, many of which are associated with changes in β-thalassemia disease severity (https://www.ithanet.eu/db/ithagenes?action=list&func=2).',\n",
       "  [(72, 79, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11373567',\n",
       "  'These databases are ClinVar, Global Variome shared LOVD, the Globin Gene Server/HbVar, the Malaysian Node of the Human Variome Project Database (MyHVPDb), and ITHANET.',\n",
       "  [(159, 166, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11373567',\n",
       "  'The IthaGenes database (https://www.ithanet.eu/db/ithagenes) [78] provides a graphical user interface with live and advanced search functionality and integrates and links gene and variant entries with other data on the ITHANET Portal and elsewhere.',\n",
       "  [(219, 226, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11373567',\n",
       "  'ITHANET with IthaGenes is the presently most comprehensive resource for hemoglobinopathies, which, beyond variant information for causative genes, also includes modifiers, detailed information on phenotypic features, detailed epidemiological information, national health policy and management information, and advanced tools for genotype-phenotype prediction for genetic counseling.',\n",
       "  [(0, 7, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11373567',\n",
       "  'The establishment of individual LOVD-based variant databases in GGN member countries and their integration with the ITHANET Portal as a central resource for hemoglobinopathies as part of this effort may be instrumental in initiating essential data-guided disease management in LMICs and in capturing the diversity of hemoglobinopathies as an international resource.',\n",
       "  [(116, 123, 'ITHANET', 'the_ithanet_portal')],\n",
       "  'Genetic variation, Genotype and phenotype, Public health and epidemiology, Haematology, Computational biology, Human genetics, Molecular genetics, Population genetics, Genetics, Molecular biology, Biomarkers, Data visualisation, Database management, Rare diseases'],\n",
       " ['PMC11466383',\n",
       "  'The Big Skin Health Intervention Fiji Trial (“Big SHIFT”) carried out surveillance for scabies and SSTIs from July 2018 to June 2019 in the Northern Division of Fiji, an area with high prevalence of scabies, prior to a division-wide ivermectin-based mass drug administration (MDA) campaign.',\n",
       "  [(33, 37, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'We extrapolated the total annual number of cases and direct healthcare costs for all divisions in Fiji based upon previous scabies and impetigo prevalence data across all divisions.',\n",
       "  [(98, 102, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The estimated annual healthcare costs of scabies and related SSTIs in Fiji was US$3.0 million, with cost per capita of $3.3.',\n",
       "  [(70, 74, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Scabies and related SSTIs lead to a heavy economic burden in Fiji and prevention would reduce these healthcare costs.',\n",
       "  [(61, 65, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'This data is the same as supplement to Hospital admissions for skin and soft tissue infections in a population with endemic scabies: A prospective study in Fiji, 2018–2019 - available at https://figshare.com/articles/dataset/Full_data_set_/13355845 The analysis code for replicating the results is available at https://github.com/EdifofonA/ScabiesSSTI-Costs-Fiji.',\n",
       "  [(156, 160, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'There is a high burden of scabies and related SSTIs at the population level in Fiji, an island country located in the South Pacific Ocean with a census population of 884,887 people in 2017 [6].',\n",
       "  [(79, 83, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'In the Skin Health Intervention Fiji Trial (SHIFT) conducted between 2012–2013, scabies prevalence was measured at 36.4% and impetigo prevalence at 23.4% among residents of three islands of Fiji [7].',\n",
       "  [(32, 36, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'In Fiji, scabies is often initially treated with traditional medicines, and many individuals only seek medical care for prolonged illness or when a secondary skin infection develops [9].',\n",
       "  [(3, 7, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'In 2016, the Fiji Government annual Health Status Report reported that SSTIs caused 4.3% of mortality within the country [10].',\n",
       "  [(13, 17, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'In Fiji, two pilot studies were conducted to provide evidence on MDA [7,11].',\n",
       "  [(3, 7, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The Skin Health Intervention Fiji Trial (SHIFT) study demonstrated that ivermectin-based MDA was safe and effective in controlling scabies and impetigo in a study population of approximately 2000 people in Fiji [7].',\n",
       "  [(29, 33, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Big SHIFT was delivered to the whole population of the Northern Division of Fiji, one of four primary administrative units of Fiji (2017 census population, 131,914) [6].',\n",
       "  [(76, 80, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The Northern division has the highest prevalence of scabies and impetigo among all Fiji divisions–with 1.3 higher odds of scabies and impetigo compared with those in other divisions [8].',\n",
       "  [(83, 87, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Healthcare in Fiji comprises a government public healthcare system and a smaller private healthcare sector [12].',\n",
       "  [(14, 18, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'In 2019, about US$144 million from the Fiji government’s revenue was allocated to health [13].',\n",
       "  [(39, 43, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Therefore, we sought to estimate the annual direct healthcare costs of scabies and related SSTIs in Fiji, a middle-income Pacific country with high burden of scabies.',\n",
       "  [(100, 104, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'This study uses prospectively collected data from the Northern Division of Fiji to estimate the costs of healthcare presentations and hospital admissions among patients with scabies and scabies-related SSTIs prior to the delivery of MDA through Big SHIFT.',\n",
       "  [(75, 79, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Costs are calculated using an ingredients-based approach and then extrapolated to the national level using epidemiological data from Fiji.',\n",
       "  [(133, 137, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'We estimated the mean costs as well as the total annual costs, using extrapolated number of cases in Fiji, to estimate the economic burden at a national level.',\n",
       "  [(101, 105, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'We used this perspective because public provision of most outpatient and inpatient care is free in Fiji and user fees are very low compared with the overall health expenditure in government facilities [12].',\n",
       "  [(99, 103, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'To estimate the quantities of medicines, we supplemented the utilization data collected in the trial with relevant antibiotics and treatment guidelines for scabies and SSTIs in Fiji [16,17].',\n",
       "  [(177, 181, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'We used the average household size in Fiji, 4.2 in 2021 [18], to quantify the number of tubes that would be sufficient to treat the whole family as recommended in the guidelines.',\n",
       "  [(38, 42, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'For those prescribed an injection, we used benzathine penicillin (for impetigo) or cloxacillin (for cellulitis, abscess, or severe SSTI) as recommended by Fiji antibiotics guidelines [16].',\n",
       "  [(155, 159, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The unit costs per diagnostic test, outpatient visit, admission to PHC facilities, and hospital admission were obtained from a Fiji costing study [20].',\n",
       "  [(127, 131, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The outpatient visits and admission unit cost estimates from the Fiji costing study includes both recurrent costs (staff time and overheads) and capital costs (building, equipment, beds, furniture, and vehicles).',\n",
       "  [(65, 69, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The Fiji costing study estimated the costs per bed day for the two hospitals (LTK and CWM), so we used the lower of the values to obtain conservative estimates.',\n",
       "  [(4, 8, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The cost of surgical procedures was not estimated separately because it was captured in the unit cost of admissions in the Fiji costing study [20].',\n",
       "  [(123, 127, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Costs obtained in other years and/or different currencies converted to Fijian dollar using the average exchange rate in that year [21], adjusted to 2020 prices using GDP deflator [22], and converted back to US$ using 2020 exchange rates.',\n",
       "  [(71, 75, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'We extrapolated case numbers to all Fiji divisions using the expected annual number of cases in the Northern Division, and the risk ratio (RR) of scabies infestation between the Northern Division and other divisions.',\n",
       "  [(36, 40, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'A study by Romani [8] indicated that scabies and impetigo cases were prevalent in the Northern division compared with other Fiji divisions, with an adjusted odds ratio (OR) of 1.3 for each disease.',\n",
       "  [(124, 128, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'We converted the OR to risk ratio (RR) using the formula in Zhang and Yu [24], and multiplied this RR WITH the risk in the Northern Division to derive the risk for other Fiji divisions.',\n",
       "  [(170, 174, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'To obtain the expected annual cases in other Fiji divisions, we multiplied the annual risk of cases above with the population size for all other Fiji divisions (752,973 based on 2017 census) [6].',\n",
       "  [(45, 49, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Lastly, we summed up the number of cases for the Northern and the number for other divisions to get the total annual number of cases in all Fiji divisions.',\n",
       "  [(140, 144, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'These extrapolated number of cases in Fiji were multiplied by the mean costs per case to yield the annual costs.',\n",
       "  [(38, 42, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'For ICU, used an average of ICU bed day estimates from two hospitals in the Fiji costing study [20] in the base case, so these two estimates served as our low/high value for sensitivity analysis.',\n",
       "  [(76, 80, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Ethical approval for Big SHIFT was granted by the Fiji National Health Research Ethics Review Committee (reference: 2018.38.NOR) and the Royal Children’s Hospital Human Research Ethics Committee in Melbourne, Australia (reference: 38020).',\n",
       "  [(50, 54, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Extrapolating the data from the Northern Division to the rest of Fiji, the estimated annual number of PHC presentations for scabies and related SSTIs was 82,183 (Table 4).',\n",
       "  [(65, 69, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'From a health system perspective, the estimated annual healthcare costs of scabies and related SSTIs in Fiji was US$3.0 million.',\n",
       "  [(104, 108, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'One-way sensitivity analysis indicated that variations in the cost per outpatient visit had the biggest influence on the total cost of scabies and SSTIs in Fiji (Fig 2).',\n",
       "  [(156, 160, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'We used before-intervention data from the Big SHIFT trial in Fiji and extrapolated these costs to the entire country.',\n",
       "  [(61, 65, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The trial provided rich information on PHC presentations (including visits and admissions) and hospital admissions for scabies and related SSTIs in the Northern Division of Fiji.',\n",
       "  [(173, 177, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The estimated annual direct healthcare costs of scabies and related SSTIs in Fiji was estimated to be $3.0 million, equivalent to 2.4% of government revenues allocated to health in 2019.',\n",
       "  [(77, 81, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Scabies and related SSTIs therefore lead to a heavy economic burden in Fiji, raising the potential benefit of prevention programs such as MDA.',\n",
       "  [(71, 75, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'The estimated 23,000 bed days in our study represents 7.1% of admissions for all divisional hospitals in Fiji in 2017 (3.6% of divisional and specialist hospital admissions).',\n",
       "  [(105, 109, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'In countries endemic for scabies like Fiji, patients may not seek treatment of scabies unless it creates a significant disturbance to their quality of life [27].',\n",
       "  [(38, 42, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'This amount is more than double the estimated cost of medicines for treatment of outpatient pneumonia in Fiji [29]; in that study, the average cost of medicines ranged from $1.3 for Nausori PHC to $2.6 for CWMH (in 2020 values).',\n",
       "  [(105, 109, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'While there are few data to guide the attribution of bacterial SSTIs to scabies in highly endemic settings, two lines of evidence suggest that impetigo is highly associated with scabies in Fiji and other Pacific Island countries.',\n",
       "  [(189, 193, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'First, the population attributable risk of impetigo to scabies has ranged from 41 to 93% in studies in Fiji and the Solomon Islands [8,30].',\n",
       "  [(103, 107, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'A strength of our study is that we analyzed PHC data obtained from a trial that was conducted among the entire population of the Northern Division of Fiji, including children and the elderly.',\n",
       "  [(150, 154, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'We only had data to calculate the odds ratio of impetigo in other divisions of Fiji compared to the Northern Division, so we used impetigo as a proxy for all potentially scabies-related SSTIs.',\n",
       "  [(79, 83, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Our approach also assumes that healthcare utilization patterns in other divisions of Fiji were consistent with those observed in the Northern Division.',\n",
       "  [(85, 89, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11466383',\n",
       "  'Our findings are likely to be relevant to other countries in the Pacific, where the burden of scabies and the costs of treatment may be similar to that of Fiji.',\n",
       "  [(155, 159, 'Fiji', 'Fiji')],\n",
       "  'Cell biology, Imaging'],\n",
       " ['PMC11385718',\n",
       "  'The strains (PGRG2 and PGRG5) were initially mapped to the reference genomes using Burrows-Wheeler aligner “mem” algorithm (BWA-mem) v.0.7.17 (5) and converted and sorted into BAM file using SAMtools v.0.1.18 (6).',\n",
       "  [(191, 199, 'SAMtools', 'samtools')],\n",
       "  'Mapping, Sequence analysis, Sequencing, Rare diseases'],\n",
       " ['PMC11385718',\n",
       "  'A gapped consensus sequence was produced using SAMtools v.0.1.18 and BEDTools (7).',\n",
       "  [(47, 55, 'SAMtools', 'samtools')],\n",
       "  'Mapping, Sequence analysis, Sequencing, Rare diseases'],\n",
       " ['PMC11385718',\n",
       "  'The coverage of consensus sequence was determined using SAMtools coverage with an overall coverage of 94.29% (PGRG2) and 95.93% (PGRG5).',\n",
       "  [(56, 64, 'SAMtools', 'samtools')],\n",
       "  'Mapping, Sequence analysis, Sequencing, Rare diseases'],\n",
       " ['PMC11385718',\n",
       "  'For the variant calling, SAMtools was used to identify the SNPs and InDels from the sorted BAM file of the mapping.',\n",
       "  [(25, 33, 'SAMtools', 'samtools')],\n",
       "  'Mapping, Sequence analysis, Sequencing, Rare diseases'],\n",
       " ['PMC10868310',\n",
       "  'This work presents vcfpp, a C++ API of HTSlib in a single file, providing an intuitive interface to manipulate VCF/BCF files rapidly and safely, in addition to being portable.',\n",
       "  [(39, 45, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  'The C API of HTSlib (Bonfield et al.',\n",
       "  [(13, 19, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  'The motivation behind vcfpp is to offer full functionalities as HTSlib, and provide a simple and safe API in a single header file, which can be easily integrated for programming in C++ as well as other languages that can call C/C++ codes, such as R (https://www.r-project.org/) and Python (https://www.python.org/).',\n",
       "  [(64, 70, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  '2009) but were later separated into HTSlib (Bonfield et al.',\n",
       "  [(36, 42, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  'Therefore, it would be more advantageous to use a specialized library, namely HTSlib, to process the VCF rather than a customized parser.',\n",
       "  [(78, 84, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  'HTSlib offers full functionalities to interact with the VCF, including support for BCF, format validation, compression, random access, identification of variant types, and support for URL links as filenames, among others.',\n",
       "  [(0, 6, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  'Furthermore, HTSlib has demonstrated the best performance among many competitors (Bonfield et al.',\n",
       "  [(13, 19, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  'However, HTSlib was written in C language, which can be challenging for beginner programmers to use, especially when it comes to memory management.',\n",
       "  [(9, 15, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  'As a C++ API of HTSlib, vcfpp inherits all functionalities from HTSlib and is implemented in a single header file that can be easily integrated and used safely.',\n",
       "  [(16, 22, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC10868310',\n",
       "  'Importantly, vcfpp and vcfppR offer users the full functionalities of HTSlib, including support for compressed VCF/BCF, selection of samples, regions, and variant types.',\n",
       "  [(70, 76, 'HTSlib', 'htslib')],\n",
       "  'Data management'],\n",
       " ['PMC11512631',\n",
       "  'Initially, PCLS were fixed in formalin for 3 days at room temperature, then paraffin-embedded and sectioned into 7\\xa0μm slices for hematoxylin and eosin (H&E) and Sirius red staining with standard protocols.3233 However, we found that PCLS from cirrhotic livers were fragile and easily degraded during sectioning and Sirius red staining (Supplemental Figure S3A, left panel, http://links.lww.com/HC9/B65).',\n",
       "  [(161, 167, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'The amount of collagen in Sirius red–stained sections was scored according to the modified Ishak method34 as described in Supplemental Table S1, http://links.lww.com/HC9/B65.',\n",
       "  [(26, 32, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'CDAHFD-induced cirrhosis32 was confirmed with Sirius red staining (Supplemental Figure S4A, http://links.lww.com/HC9/B65).',\n",
       "  [(46, 52, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'This short-term exposure of PCLS slices to erlotinib did not significantly reduce the amount of collagen measured with Sirius red staining (Figures 2C, D).',\n",
       "  [(119, 125, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'Amount of collagen (C) measured with Sirius red staining and (D) quantified with collagen proportionate area analysis in PCLS from CDAHFD-induced mouse cirrhosis after 5\\xa0μM erlotinib treatment for 72 hours.',\n",
       "  [(37, 43, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'Amount of collagen (G) measured with Sirius red staining and (H) quantified in PCLS from TAA-induced rat cirrhosis after 5\\xa0μM erlotinib treatment for 72 hours.',\n",
       "  [(37, 43, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'The images of Sirius red staining for both CDAHFD and TAA models were presented at ×20 magnification.',\n",
       "  [(14, 20, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'Cirrhosis was induced by TAA38 and confirmed with Sirius staining (Supplemental Figure S5A, http://links.lww.com/HC9/B65).',\n",
       "  [(50, 56, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'DEN-induced cirrhosis2339404142 was confirmed with Sirius staining (Supplemental Figure S6A, http://links.lww.com/HC9/B65).',\n",
       "  [(51, 57, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'Amount of collagen (B) measured with Sirius red staining and (C) quantified with collagen proportionate area analysis in PCLS after 5\\xa0μM erlotinib treatment for 72 hours.',\n",
       "  [(37, 43, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'Amount of collagen (E) measured with Sirius red staining and (F) quantified after daily i.p.',\n",
       "  [(37, 43, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'CCl4-induced cirrhosis23 was confirmed with Sirius red staining (Supplemental Figure S7A, http://links.lww.com/HC9/B65).',\n",
       "  [(44, 50, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  '(E) Amount of collagen measured with Sirius red staining and quantified with collagen proportionate area analysis in PCLS from normal rats.',\n",
       "  [(37, 43, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC11512631',\n",
       "  'Amount of collagen (B) measured with Sirius red staining and (C) quantified in PCLS from human cirrhosis treated with 5\\xa0μM erlotinib for 72 hours.',\n",
       "  [(37, 43, 'Sirius', 'sirius')],\n",
       "  'Proteomics, Proteomics experiment, Metabolomics'],\n",
       " ['PMC10765786',\n",
       "  'Here, we describe cytoviewer, an R/Bioconductor package for interactive visualization and exploration of multi-channel images and segmentation masks.',\n",
       "  [(18, 28, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The cytoviewer package supports flexible generation of image composites, allows side-by-side visualization of single channels, and facilitates the spatial visualization of single-cell data in the form of segmentation masks.',\n",
       "  [(4, 14, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'As such, cytoviewer improves image and segmentation quality control, the visualization of cell phenotyping results and qualitative validation of hypothesis at any step of data analysis.',\n",
       "  [(9, 19, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'We showcase the functionality and biological application of cytoviewer by analysis of an imaging mass cytometry dataset acquired from cancer samples.',\n",
       "  [(60, 70, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The cytoviewer package offers a rich set of features for highly multiplexed imaging data visualization in R that seamlessly integrates with the workflow for image and single-cell data analysis.',\n",
       "  [(4, 14, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'It can be installed from Bioconductor via https://www.bioconductor.org/packages/release/bioc/html/cytoviewer.html.',\n",
       "  [(98, 108, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The development version and further instructions can be found on GitHub at https://github.com/BodenmillerGroup/cytoviewer.',\n",
       "  [(111, 121, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'Here, we present cytoviewer, an R/Bioconductor package for interactive multi-channel image and segmentation mask visualization in R.',\n",
       "  [(17, 27, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The cytoviewer package builds on the cytomapper R/Bioconductor package [17] and extends its static visualization abilities via an interactive and user-friendly shiny application.',\n",
       "  [(4, 14, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The cytoviewer package integrates into the Bioconductor framework [19] for single-cell and image analysis leveraging the image handling and analysis strategies from the EBImage Bioconductor package [20] and building on commonly used Bioconductor classes such as SingleCellExperiment, SpatialExperiment [21, 22], and CytoImageList [17].',\n",
       "  [(4, 14, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'We showcase the functionality and biological application of cytoviewer by demonstrating visual exploration of an IMC dataset of cancer patients.',\n",
       "  [(60, 70, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The R/Bioconductor cytoviewer package leverages the reactive programming framework of the popular R shiny and shinydashboard packages [23], is cross-platform compatible, and launches an interactive web application.',\n",
       "  [(19, 29, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The graphical user interface (GUI) of the cytoviewer package can be opened directly from R and the function call takes up to five arguments (Fig.',\n",
       "  [(42, 52, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The full functionality of cytoviewer is leveraged when images, segmentation masks, and a metadata object are provided (Fig.',\n",
       "  [(26, 36, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  '1cytoviewer interface and functionality.',\n",
       "  [(1, 11, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  '(A) The supported functionality (right) of cytoviewer depends on the data inputs (left).',\n",
       "  [(43, 53, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  '(B) The graphical user interface of cytoviewer is divided into a body, header, and sidebar.',\n",
       "  [(36, 46, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The body of cytoviewer includes the image viewer, which has three tabs: Composite (Image-level), Channels (Image-level), and Mask (Cell-level).',\n",
       "  [(12, 22, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'Scale bar: 150\\xa0µm (C) cytoviewer supports different viewing modes.',\n",
       "  [(22, 32, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'cytoviewer interface and functionality.',\n",
       "  [(0, 10, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The cytoviewer GUI has three main parts: body, sidebar, and header (Fig.',\n",
       "  [(4, 14, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The body of cytoviewer features the image viewer.',\n",
       "  [(12, 22, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'To demonstrate the functionality and potential biological applications of the cytoviewer package, we explored an example IMC dataset from the Integrated iMMUnoprofiling of large adaptive CANcer patient cohort project (immucan.eu) (Additional file 5: Supplementary Notes—S1.4).',\n",
       "  [(78, 88, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'Here, we showcase the different viewing modes of cytoviewer by analyses of images from a breast cancer patient (Patient2_003) and outline the rich biological information content for each step (Fig.',\n",
       "  [(49, 59, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The R/Bioconductor cytoviewer package provides a rich set of visualization features including (1) visualization of composite and single-channel images, (2) outlining of cells on images, (3) setting cell outline colors based on cellular metadata (e.g., cell phenotype), (4) coloring segmentation masks based on cellular metadata (e.g., cell area) and (5) easy access and download of generated plots for publication, among others.',\n",
       "  [(19, 29, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The cytoviewer package operates on standard data containers of the Bioconductor project [19] such as SingleCellExperiment/SpatialExperiment [21, 22] and CytoImageList [17] and can thus be readily implemented into existing R pipelines for image and single-cell analysis.',\n",
       "  [(4, 14, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The versatile and easy-to-use graphical user interface of cytoviewer further allows accessibility to users with little bioinformatics training and coding experience.',\n",
       "  [(58, 68, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'Here, we demonstrated the use of cytoviewer by exploring IMC data (Additional file 1, 2).',\n",
       "  [(33, 43, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'However, data from other multiplexed imaging technologies including t-CyCIF [6], CODEX [7], or MIBI [8], which produce pixel-level intensities and (optionally) segmentation masks, can be interactively visualized with cytoviewer as long as the input format is appropriate.',\n",
       "  [(217, 227, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'We have integrated cytoviewer into our widely adopted data analysis pipeline for multiplexed imaging data [10] (https://bodenmillergroup.github.io/IMCDataAnalysis/) and envision that the package will meet the needs of the fast-growing community of highly multiplexed imaging users [11].',\n",
       "  [(19, 29, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The newly developed R/Bioconductor cytoviewer package, written in the statistical programming language R, allows interactive visualization and exploration of multi-channel images and segmentation masks.',\n",
       "  [(35, 45, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  ' Project name: cytoviewerProject home page: https://github.com/BodenmillerGroup/cytoviewerOperating system(s): Platform independentProgramming language: ROther requirements: R\\u2009>\\u2009=\\u20094.0License: NoneAny restrictions to use by non-academics: None',\n",
       "  [(15, 25, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'Project name: cytoviewer',\n",
       "  [(14, 24, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'Project home page: https://github.com/BodenmillerGroup/cytoviewer',\n",
       "  [(55, 65, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'A video demonstrating the functionality of cytoviewer.Additional file 3: Fig.',\n",
       "  [(43, 53, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'cytoviewer graphical user interface overview.',\n",
       "  [(0, 10, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'The graphical user interface of cytoviewer for the three different viewer modes.',\n",
       "  [(32, 42, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'cytoviewer image filters.',\n",
       "  [(0, 10, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC10765786',\n",
       "  'A video demonstrating the functionality of cytoviewer.',\n",
       "  [(43, 53, 'cytoviewer', 'cytoviewer')],\n",
       "  'Cytometry, Imaging, Oncology, Genotype and phenotype, Software engineering'],\n",
       " ['PMC11202377',\n",
       "  'Two of the most popular ones are RepeatModeler [9] and REPET [8].',\n",
       "  [(55, 60, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC11202377',\n",
       "  'For each of the species we compared the results of pantera to those of a reference library and those obtained with a denovo method, RepeatModeler for Drosophila melanogaster and Danio rerio and REPET for Oryza sativa.Fig.',\n",
       "  [(194, 199, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC11202377',\n",
       "  'In this case we compared the results of pantera to the uncurated library obtained with REPET [8, 34, 35] downloaded directly from REPETDB [1] composed of 2,479 families.',\n",
       "  [(87, 92, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC11202377',\n",
       "  'Pantera found more near-full length (>\\u200990%) members of the reference libraries than RepeatModeler (fruit fly and zebrafish) or REPET (rice) except for LTR elements for rice and zebrafish.',\n",
       "  [(127, 132, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC11202377',\n",
       "  'As an example, of 48 CMC-EnSpm families identified by pantera, only 5 lack the expected TIR elements, compared to 29 families missing the TIR element out of 70 in the REPET results.',\n",
       "  [(167, 172, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC11202377',\n",
       "  'pantera (N\\u2009=\\u2009525), rice6.9.5 (N\\u2009=\\u20092431), REPET (N\\u2009=\\u20092471) c Danio rerio.',\n",
       "  [(41, 46, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC11202377',\n",
       "  '8) except for the REPET library for rice for which we used a library previously generated.',\n",
       "  [(18, 23, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC11202377',\n",
       "  'Because pantera tends to generate full length consensus sequences more frequently than tools that start from a repeat-first approach, we suggest that it might be used first, then the genome be masked for sequences found by pantera, then a method such as RepeatModeler or REPET be used on the masked genome.',\n",
       "  [(271, 276, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC11202377',\n",
       "  'The authors want to acknowledge the support of Simon Orozco (Institut de Biologia Evolutiva—CSIC UPF) for his feedback on TE library benchmarking methods, Johann Confais (INRAE) for providing the REPET library for Oryza sativa and Jessica Storer (Dfam—University of Connecticut) for her insights on TE curation.',\n",
       "  [(196, 201, 'REPET', 'repet')],\n",
       "  'Sequence composition, complexity and repeats, DNA, Nucleic acid structure analysis'],\n",
       " ['PMC10620421',\n",
       "  'PASTEClassifier19 was used to categorize different types of repetitive sequences.',\n",
       "  [(0, 15, 'PASTEClassifier', 'PASTEClassifier')],\n",
       "  'Sequence composition, complexity and repeats'],\n",
       " ['PMC10620421',\n",
       "  '(1) BWA v0.7.10-r789: aln, default parameters; (2) LACHESIS: CLUSTER_MIN_RE_SITES\\u2009=\\u2009547; CLUSTER_MAX_LINK_DENSITY\\u2009=\\u20092; CLUSTER_NONINFORMATIVE_RATIO\\u2009=\\u20092; ORDER_MIN_N_RES_IN_TRUN\\u2009=\\u20091094; ORDER_MIN_N_RES_IN_SHREDS\\u2009=\\u20091076; (3) BUSCO v3.0: --evalue 1e-03 (E-value cutoff for BLAST searches), -sp human; (4) LTR_FINDER: default parameters; (5) RepeatScout: default parameters; (6) PASTEClassifier: default parameters; (7) RepeatMasker: -nolow -no_is -norna -engine wublast; (8) Genscan: default parameters; (9) Augustus v2.4: default parameters; (10) GlimmerHMM v3.0.4: default parameters; (11) GeneID v1.4: default parameters; (12) SNAP: version 2006-07-28, default parameters; (13) GeMoMa v1.3.1: default parameters; (14) Hisat v2.0.4: --max-intronlen 20000, --min-intronlen 20; (15) Stringtie v1.2.3: default parameters; (16) TransDecoder v2.0: default parameters; (17) GeneMark-ST v5.1: default parameters; (18) PASA v2.0.2: -align_tools gmap, -maxIntronLen 20000; (19) EVM v1.1.1: default parameters; (20) BLAST V2.2.31: -evalue 1e-5; (21) tRNAscan-SE: default parameters.',\n",
       "  [(375, 390, 'PASTEClassifier', 'PASTEClassifier')],\n",
       "  'Sequence composition, complexity and repeats'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer is a public web-based tool relying on a relational database available at https://urgi.versailles.inrae.fr/synteny delivering comparative genomics data and associated reservoir of conserved genes between angiosperm species for both fundamental (evolutionary studies) and applied (translational research) applications.',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer is made available for (i) providing comparative genomics data for seven major botanical families of flowering plants, (ii) delivering a robust catalog of 103\\u2009465 conserved genes between 44 species and inferred ancestral genomes, (iii) allowing us to investigate the evolutionary fate of ancestral genes and genomic regions in modern species through duplications, inversions, deletions, fusions, fissions and translocations, (iv) use as a tool to conduct translational research of key trait-related genes from model species to crops and (v) offering to host any comparative genomics data following simplified procedures and formats',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer is a web resource to perform comparative genomics in plants;',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer allows access to expertised data and to download novel analyses;',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer provides methods, scripts, documents and procedures to generate comparative genomics data.',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'This article presents SyntenyViewer, a web-based tool hosting expertised synteny relationships between angiosperm genomes through the reconstruction of ancestral genomes (17), and discusses potential uses of the delivered catalog of conserved genes for evolutionary studies as well translational research investigation.',\n",
       "  [(22, 35, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer is a tool relying on a relational database (DB), aiming at displaying and making publicly available the previously described comparative genomics data at https://urgi.versailles.inrae.fr/synteny.',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer data processing and database description.',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  '(a) Illustration of theSyntenyViewer architecture including the data integration step into a PostgreSQL instance, and the data visualization based on GWT powered by Apache Tomcat and Apache HTTP server.',\n",
       "  [(23, 36, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  '(b) Illustration of the SyntenyViewer database model with for each box a table named with its primary key term (referenced below).',\n",
       "  [(24, 37, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'Following this methodology, SyntenyViewer delivers published comparative genomics data (listed in\\xa0Table\\xa01) obtained for the two major angiosperm families with the grasses within the monocots [ancestral grass karyotype (AGK) with 12 protochromosomes and 16\\u2009560 PGs (21, 22)] and the eudicots [ancestral eudicot karyotype (AEK) with 21 chromosomes and 10\\u2009286 PGs (23)].',\n",
       "  [(28, 41, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer also provides published comparative genomics data for angiosperm lineages of agronomical interest such as Rosaceae [ancestral Rosaceae karyotype (ARK) with nine protochromosomes and 8861 PGs (24, 25)], Brassicaceae [ancestral Brassicaceae karyotype (ABK) with eight protochromosomes and 20\\u2009037 PGs (26, 27)], Cucurbitaceae [ancestral Cucurbitaceae karyotype (ACuK) with 22 protochromosomes and 17\\u2009969 PGs (28, 29)], legumes [ancestral legume karyotype (ALK) with 16 protochromosomes and 13\\u2009181 PGs (30–32)] and Solanaceae [ancestral Solanaceae karyotype (ASK) with 17 protochromosomes and 17\\u2009879 PGs (33)].',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'For example, see the Terms tab of ‘PlantSyntenyViewer Solanaceae submission file’ where the license is also prompted when a user attempts to download the associated file (Table\\xa01).',\n",
       "  [(40, 53, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'Previous synteny data are integrated into the SyntenyViewer tool relying on a DB with a Java web application for graphical dynamic web content processing, a web server (Apache HTTP and Apache Tomcat) and a PostgreSQL 9.6 instance to store the data\\xa0(Figure\\xa02).',\n",
       "  [(46, 59, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'A spreadsheet-based data exchange format allows synteny data submission to SyntenyViewer, available at https://urgi.versailles.inrae.fr/Data/Synteny/Data-submission.',\n",
       "  [(75, 88, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'Synteny data is then made publicly available when the format described on our website (https://urgi.versailles.inrae.fr/Data/Synteny/Data-submission), and the aims of SyntenyViewer are met.',\n",
       "  [(167, 180, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'There are two main entry points to visualize SyntenyViewer data (cf Supplementary Video).',\n",
       "  [(45, 58, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'At any stage, the SyntenyViewer’s Uniform Resource Locator is dynamically updated, and it can be bookmarked in the browser for sharing the visualization link, which makes users able to go back to previous work and pursue exploration.',\n",
       "  [(18, 31, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'The delivered SyntenyViewer tool gives public access to validated and reviewed comparative genomics data either between angiosperm species or within major botanical families that can be used as a backbone to investigate evolutionary trends of genes, perform translational research of traits and conduct evolutionary developmental biology (for Evo-Devo) investigation of traits.',\n",
       "  [(14, 27, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer allows deep investigation of evolutionary fates of ancestral genes/genomes, through precise identification of the changes involved (gains and losses of genes and associated gene ontologies) and their assignment to specific species or botanical families\\xa0(Figure\\xa03a).',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer, a comparative genomics-driven translational research tool.',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  '(b) ‘SyntenyViewer screen capture’.',\n",
       "  [(5, 18, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer tool with the setting parameters (search by gene name and ancestral or modern chromosomes) illustrated at the left and the derived comparative genomics data visualization, as detailed in the text, at the right (here for cereals).',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'FZP gene characterization in grasses with orthologs from SyntenyViewer (Panel b) and functional validation in wheat and Brachypodium (in mutants compared to wild type) in deriving similar SS phenotypes (adapted from (47)).',\n",
       "  [(57, 70, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer can also be used as a useful tool for translational research on genes driving key agronomical traits, particularly from model species (such as Arabidopsis thaliana) to crops (41).',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'From the gene conservation information delivered from SyntenyViewer, further validation needs to be conducted to establish the conservation of the phenotype or trait between the investigated species.',\n",
       "  [(54, 67, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'Beside translational research of genes, SyntenyViewer allows us to conduct ‘Evo-Devo dissection of traits’.',\n",
       "  [(40, 53, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'SyntenyViewer allows us to compare a group of angiosperm species that acquired new phenotypes (or traits in the broad sense) in the course of evolution, compared to a group of species that did not acquire this trait.',\n",
       "  [(0, 13, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'Overall, SyntenyViewer is a web-based tool delivering comparative genomics data either between angiosperm species or within major botanical families (including the Rosaceae, Brassicaceae, Cucurbitaceae, legume and Solanaceae) for evolutionary and translational research purposes.',\n",
       "  [(9, 22, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC10167986',\n",
       "  'The ‘Région Auvergne-Rhône-Alpes’ and ‘Fonds Européen de Développement Régional’ (#23000816 project ‘SRESRI-2015’), the Institut Carnot Plant2Pro (project ‘SyntenyViewer’), the Initiative-Science-Innovation-Territoires-Economie (ISITE) CAP2025 (#00002146 SRESRI 2015, Pack Ambition Recherche Project ‘TransBlé’), the Agence Nationale de la Recherche (ANR) projects ‘BREEDWHEAT’ (ANR-10-BTBR-03-06), the Biomass For the Future project (ANR-11-BTBR-0006), the ANR project, for Plant and Animal Genome Evolution (PAGE) (ANR-11-BSV6-0008) and the Plant Bioinformatics Facility (https://doi.org/10.15454/1.5572414581735654E12).',\n",
       "  [(156, 169, 'SyntenyViewer', 'urgi')],\n",
       "  'Mapping, Plant biology, Structural variation, Paleogenomics, Protein expression, Comparative genomics'],\n",
       " ['PMC4870745',\n",
       "  'We developed a simple Web application called Phenotator for the data providers to submit and annotate their phenotypes with an EQ.',\n",
       "  [(45, 55, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC4870745',\n",
       "  'The Phenotator is built using services from the NCBO BioPortal [15] to generate simple drop down menus and autocomplete search functionality to guide the users in generating EQs with appropriate terms (Fig.',\n",
       "  [(4, 14, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC4870745',\n",
       "  'Phenotator has a feature to export the collected EQ annotations as an OWL file containing new terms that are logically defined according to the SUBQ pattern,10 which can be expressed in Manchester OWL syntax as “(has_part some (<Quality> and inheres_in some <Entity>))”.',\n",
       "  [(0, 10, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC4870745',\n",
       "  'One hundred twenty-seven phenotype descriptions from the original 11 datasets were entered into Phenotator, together with 41 phenotypes collected from cell migration assays (Z.',\n",
       "  [(96, 106, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC4870745',\n",
       "  'The EQs were exported from Phenotator as an OWL file and loaded into the Protege OWL ontology editor.',\n",
       "  [(27, 37, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC4870745',\n",
       "  'To increase the expressivity of the annotation in Phenotator a third column was added to capture additional modifiers to the EQ resulting in annotations emerging like EQE2 (‘actin filament’, ‘localised’, ‘cytosol’).',\n",
       "  [(50, 60, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC4870745',\n",
       "  'Pattern-based tooling to rapidly generate new terms are emerging and these could nicely complement existing tooling that are primarily aimed at annotating phenotypes with EQs such as Phenotator and Phenote.',\n",
       "  [(183, 193, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC4870745',\n",
       "  'Phenotator and Phenote do little to guide the annotator to make a correct EQ annotation and the translation of these annotations to OWL typically only allows for a basic SUBQ pattern.',\n",
       "  [(0, 10, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC4870745',\n",
       "  'The Phenotator and Webulous software was developed by SJ and TB.',\n",
       "  [(4, 14, 'Phenotator', 'phenotator')],\n",
       "  'Ontology and terminology, Data management'],\n",
       " ['PMC5753341',\n",
       "  'We introduce the marine databases; MarRef, MarDB and MarCat (https://mmp.sfb.uit.no/databases/), which are publicly available resources that promote marine research and innovation.',\n",
       "  [(53, 59, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'The last database, MarCat, represents a gene (protein) catalog of uncultivable (and cultivable) marine genes and proteins derived from marine metagenomics samples.',\n",
       "  [(19, 25, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'Currently, MarCat contains 1227 records with 55 metadata fields.',\n",
       "  [(11, 17, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'Here, we introduced the contextual and sequence MAR databases: MarRef, MarDB and MarCat, with manually curated metadata including attributes for sampling, sequencing, assembly and annotation in addition to the organism and taxonomic information and their corresponding nucleotide and protein sequences.',\n",
       "  [(81, 87, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'Only metagenomic sequences in relation with MarCat has been processed using META-pipe for the first release.',\n",
       "  [(44, 50, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'The MarRef, MarDB and MarCat sequence databases are based on the non-redundant genome and metagenome datasets obtained from ENA (European Nucleotide Archive, http://www.ebi.ac.uk/ena) and NCBI (https://www.ncbi.nlm.nih.gov/).',\n",
       "  [(22, 28, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'MarCat represents a gene (protein) catalog of predicted marine genes derived from marine metagenomic samples.',\n",
       "  [(0, 6, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'The MarRef, MarDB and MarCat contextual databases are built by compiling data from a number of publicly available sequence, taxonomy and literature databases in a semi-automatic fashion.',\n",
       "  [(22, 28, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'The gene (protein) catalog database derived from marine metagenomic samples, MarCat, contains 1227 records, including samples from the Tara Ocean expedition (248 records) and Ocean Sampling Day (150 records).',\n",
       "  [(77, 83, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'While MarRef is thoroughly curated, MarDB and MarCat are only partly curated.',\n",
       "  [(46, 52, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'Records in the marine databases, MarRef, MarDB and MarCat follow the MlxS standard guidelines developed by the Genomic Standard Consortium, in addition to ontologies such as ENVO and GAZ.',\n",
       "  [(51, 57, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'The MarRef, MarDB and MarCat sequence databases are based on the non-redundant genome and metagenome datasets obtained from ENA and NCBI and by manually inspection assigned as belonging to the ‘marine microbial biome’ according to our definition.',\n",
       "  [(22, 28, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  ' MarCat represents a catalog of uncultivable (and cultivable) full-length genes (proteins) derived from marine metagenomic samples based on the Marine projects in EBI metagenomics (https://www.ebi.ac.uk/metagenomics/).',\n",
       "  [(1, 7, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'Using META-pipe for gene prediction and functional assignment allowed us to generate a consistent catalog across the datasets in MarCat (See https://f1000research.com/articles/6–70/v1 for a more detailed description of functional assignment).',\n",
       "  [(129, 135, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'The index of MarCat (not shown) is less comprehensive, thus have fever filtering options.',\n",
       "  [(13, 19, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'For MarCat, the metagenome databases, the 55 metadata fields have been divided into four categories: isolate info, sampling info, host and pathogenicity info and assembly info, in addition to Summary.',\n",
       "  [(4, 10, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'The BLAST (35) sequence databases provide similarity search against all nucleotide and protein sequences of records included in MarRef, MarDB and MarCat.',\n",
       "  [(146, 152, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'For MarCat, the marine metagenomics gene catalogue, target information can be obtained from other databases such as UniProt, InterPro and Brenda.',\n",
       "  [(4, 10, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'For samples in the MarCat database, all predicted 16 S sequences and assembled contigs in FASTA format can be downloaded.',\n",
       "  [(19, 25, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'In addition, we aim to include metatranscriptomics data to enhance the quality of the MarCat.',\n",
       "  [(86, 92, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC5753341',\n",
       "  'For MarCat we intend to include metadata fields for the provenance of analysis according to the recommendation by Hoopen et\\xa0al.',\n",
       "  [(4, 10, 'MarCat', 'marcat')],\n",
       "  'Marine biology, Microbiology, Metagenomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'We trained both models using metagenomic reads simulated from marine microbial genomes in the MarRef database.',\n",
       "  [(94, 100, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'The performance of both models (accuracy, precision, and percent microbe predicted) was compared with the standard taxonomic classification tool Kraken2 using 10 complex metagenomic data sets simulated from MarRef.',\n",
       "  [(207, 213, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Taxonomic and sequence data from version 1.6 of the MarRef database (24, 25) was used both to simulate metagenomic reads for taxonomic classification and as a source of artificial marine metagenomes for model training, testing, and validation.',\n",
       "  [(52, 58, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'MarRef is a manually curated marine microbial reference database containing fully sequenced genomes (24, 25).',\n",
       "  [(0, 6, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'A total of 1,271 genomes of marine prokaryotic species were retrieved from MarRef to develop the training, testing, and validation data sets.',\n",
       "  [(75, 81, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'For each genome, we extracted the Genome Taxonomy Database (GTDB) (26) genus and species classification from the MarRef database’s metadata file.',\n",
       "  [(113, 119, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Of the 1,271 MarRef genomes, four records had undefined GTDB taxonomy due to failed quality checks and were dropped from the data set.',\n",
       "  [(13, 19, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'For performance comparison, the default parameters of CAMISIM v1.3 (28) were used to perform 10 separate read simulations from the 1,267 MarRef genomes, with random seeds different from those used in the training and testing sets.',\n",
       "  [(137, 143, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  '(1) Full-length genome records and their metadata were downloaded from MarRef v1.6.',\n",
       "  [(71, 77, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  '(4) Blind data sets were created using CAMISIM, which randomly generates metagenomic data sets from the MarRef genomes.',\n",
       "  [(104, 110, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Kraken2 v2.1.2 was used to assign taxonomy to metagenomic reads in the blind data sets using MarRef v1.6 as the reference database.',\n",
       "  [(93, 99, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Bowtie2 v2.4.5’s bowtie-build command was used to generate a reference database containing all 1,267 MarRef genomes from the training set.',\n",
       "  [(101, 107, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Our shell script can be found at https://github.com/helloftroy/MarRef_DeepMicrobes.',\n",
       "  [(63, 69, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'We benchmarked the performance of our best-performing species and genus classification models (S-2E, S-3E, G-2E, G-3E; species 2 and 3 epoch, genus 2 and 3 epoch) against Kraken2 using 10 simulated blind metagenomes that contain random proportions of reads from the MarRef genomes.',\n",
       "  [(266, 272, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Our results suggest that the DeepMicrobes marine (MarRef)-trained model showed lower accuracy than the prior human gut model (0.65 vs 0.96).',\n",
       "  [(50, 56, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  '(e) Genomic data comparison between HGR and MarRef genomes.',\n",
       "  [(44, 50, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Notably, all genomes mispredicted for the genus models have only one representative genome in MarRef for the model to train on; many also have long genome length and high GC content.',\n",
       "  [(94, 100, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'To evaluate whether performance differences between the DeepMicrobes gut and marine models was affected by the input genomes, we compared the HGR and MarRef data sets used for training.',\n",
       "  [(150, 156, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'We noticed average genome size in the MarRef data set was larger than the average genome size in the HGR data set (4.1 vs 2.5 Mbp).',\n",
       "  [(38, 44, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'However, the average number of genomes represented in each genus was lower than the MarRef data set compared to the HGR data set (3.1 vs 20.7) (Fig.',\n",
       "  [(84, 90, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Lastly, we observed that though the average GC content was similar between both database genomes (0.48), many MarRef genus with <0.5 MCC had high GC content (0.52–0.72).',\n",
       "  [(110, 116, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'ResNet architecture results for all models using sample of MarRef genomes.',\n",
       "  [(59, 65, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC10580933',\n",
       "  'Jupyter Notebooks used to create figures and analyze datasetsdata sets, along with shell scripts to run DeepMicrobes, Bowtie2, CAMISIM, and other program code can be found at MarRef_DeepMicrobes.',\n",
       "  [(175, 181, 'MarRef', 'marref')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC11124354',\n",
       "  'A dataset of cold-adapted bacterial genome sequences was generated based on the five databases MarRef v.1.7, MarDB v.1.6, Ocean Microbiomics Database (OMD) v.1.1, BacDive (https://bacdive.dsmz.de/ (accessed on 23 January 2023)), and TEMPURA (http://togodb.org/db/tempura (accessed on 4 April 2023)).',\n",
       "  [(109, 114, 'MarDB', 'mardb')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC11124354',\n",
       "  'Filtering from MarDB, MarRef, and OMD databases was primarily focused on isolation location.',\n",
       "  [(15, 20, 'MarDB', 'mardb')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC11124354',\n",
       "  'All genomes from MarRef and MarDB bacteria classified as mesophilic, thermotolerant, and thermophilic were excluded, as well as those bacteria isolated from homoiotherms and hydrothermal vents.',\n",
       "  [(28, 33, 'MarDB', 'mardb')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC11124354',\n",
       "  'All remaining genomes, including the metagenome-assembled genomes (MAGs) from MarDB and OMD databases, were quality assessed using CheckM (v1.2.2) [41].',\n",
       "  [(78, 83, 'MarDB', 'mardb')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC11124354',\n",
       "  'The genomes of interest from MarRef (N = 86), MarDB (N = 225), OMD (N = 299), BacDive (N = 305), and TEMPURA (N = 25) were downloaded from the respective databases using their provided application programming interfaces (APIs) or download portals.',\n",
       "  [(46, 51, 'MarDB', 'mardb')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC11124354',\n",
       "  'To map antiphage defense systems in cold-adapted bacteria, we generated a dataset of assembled genomes of bacteria that were considered cold-adapted either based on isolation location (MarRef, MarDB, and OMD databases) or experimental temperature growth data (BacDive and TEMPURA databases).',\n",
       "  [(193, 198, 'MarDB', 'mardb')],\n",
       "  'Marine biology, Microbiology, Genomics, Data mining'],\n",
       " ['PMC4847265',\n",
       "  'Here, we present a different approach, relying on tightly integrated method rather than “Wiki-based” method, to support community annotation and user collaboration in the Integrated Microbial Genomes (IMG) system.',\n",
       "  [(171, 205, 'Integrated Microbial Genomes (IMG)', 'img')],\n",
       "  'Virology, Microbial ecology, Genomics, Metagenomics, Comparative genomics'],\n",
       " ['PMC4718670',\n",
       "  'Here we present an automated tool, called iMet-Q (intelligent Metabolomic Quantitation), for label-free metabolomics quantitation from high-throughput MS1 data.',\n",
       "  [(42, 48, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'By performing peak detection and peak alignment, iMet-Q provides a summary of quantitation results and reports ion abundance at both replicate level and sample level.',\n",
       "  [(49, 55, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'An in-house standard mixture and a public Arabidopsis metabolome data set were analyzed by iMet-Q.',\n",
       "  [(91, 97, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'From the mixture data set, seven standard metabolites were detected by the four quantitation tools, for which iMet-Q had a smaller quantitation error of 12% in both profile and centroid data sets.',\n",
       "  [(110, 116, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'With the isotope ratios calculated by iMet-Q, 49% (89 out of 183) metabolite candidates were filtered out.',\n",
       "  [(38, 44, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'From the public Arabidopsis data set reported with two internal standards and 167 elucidated metabolites, iMet-Q detected all of the peaks corresponding to the internal standards and 167 metabolites.',\n",
       "  [(106, 112, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'iMet-Q provides user-friendly interfaces and is publicly available for download at http://ms.iis.sinica.edu.tw/comics/Software_iMet-Q.html.',\n",
       "  [(0, 6, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'To rectify the above concerns, we develop a quantitation tool, called iMet-Q (intelligent Metabolomic Quantitation), written in C# programming language, which provides highly accurate quantitation and user-friendly graphical interfaces.',\n",
       "  [(70, 76, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'For peak detection, iMet-Q requires one input parameter, i.e., mzWidth (defined as the width of a peak in m/z dimension), and calculates the full width at half maximum (FWHM) of a peak to dynamically determine its chromatographic width so that the peak boundary can be automatically determined (i.e., without requiring parameter input).',\n",
       "  [(20, 26, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Once the charge state is determined, iMet-Q calculates its isotope ratio (i.e., the ratio of the second isotope abundance to the monoisotope abundance) to facilitate metabolite identification.',\n",
       "  [(37, 43, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Since metabolites usually have smaller retention time drifts across LC-MS technical replicates, iMet-Q first aligns peaks across different LC-MS replicates of a sample and then across different samples.',\n",
       "  [(96, 102, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'We used two metabolite data sets, an in-house standard mixture and a public Arabidopsis metabolome data set [28], to evaluate the performance of iMet-Q and compare it with three public quantitation tools, including XCMS, MetAlign, and MZmine2.',\n",
       "  [(145, 151, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'As a result, iMet-Q had small quantitation error of 12% in both profile and centroid data sets.',\n",
       "  [(13, 19, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'For the public Arabidopsis metabolome data set reported with two internal standards and 167 elucidated metabolites, iMet-Q detected the internal standards from all replicates and samples.',\n",
       "  [(116, 122, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In addition, all of the 167 metabolites were also detected by iMet-Q, whereas 103 (62%), 145 (87%), and 129 (77%) elucidated metabolites were detected by XCMS, MetAlign, and MZmine 2, respectively.',\n",
       "  [(62, 68, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Comparing with the three quantitation tools, iMet-Q had relatively small abundance variations (≤0.19) when quantifying the internal standards, and had higher abundance correlation (≥0.92) when quantifying the 167 metabolites across replicates.',\n",
       "  [(45, 51, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'The source code of iMet-Q is freely available on SourceForge (http://sourceforge.net/projects/imet-q/?source=navbar) under the license of GPL2.',\n",
       "  [(19, 25, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'iMet-Q automatically skipped centroiding step when processing the centroid data set and used the following parameter setting: mzWidth = 0.02, mzTol = 0.02, and rtTol = 0.2 for both data sets.',\n",
       "  [(0, 6, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Public Arabidopsis data set was processed by iMet-Q with mzWidth, mzTol, and rtTol of 0.03, 0.03, and 0.16 min, respectively.',\n",
       "  [(45, 51, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In order to process data from various mass spectrometers, iMet-Q accepts input files in netCDF, mzXML, and mzML formats, which can be conveniently converted from raw data by existing converters.',\n",
       "  [(58, 64, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'The workflow of iMet-Q contains two main tasks, peak detection and peak alignment, as depicted in Fig 1.',\n",
       "  [(16, 22, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'To compare compounds across different samples, iMet-Q aligns the detected peaks in the peak lists according to their m/z and retention time.',\n",
       "  [(47, 53, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'If the input files have already been centroided, iMet-Q automatically skips this step.',\n",
       "  [(49, 55, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Instead of determining a fixed baseline for each spectrum, iMet-Q equally divides a scan into s segments (s = 10 by default) with an overlap of 20% to avoid borderline issues and defines the average intensity in a segment as its noise level.',\n",
       "  [(59, 65, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'After processing signals in each scan, iMet-Q uses a two-stage algorithm to construct the extracted ion chromatogram (EIC) of a compound.',\n",
       "  [(39, 45, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In signal clustering, starting from the most intensive signal, say sij, among all scans in S, iMet-Q clusters signals in S with sij if the m/z differences between the signals and sij are within the given tolerance d, i.e., mzWidth in iMet-Q.',\n",
       "  [(94, 100, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In the step of EIC boundary determination, iMet-Q automatically calculates FWHM and then dynamically determines the boundaries using FWHM.',\n",
       "  [(43, 49, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In order to determine the charge state and isotope ratio, iMet-Q first constructs an isotope envelope for a peak.',\n",
       "  [(58, 64, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Then, to validate the correctness of the isotope envelope, iMet-Q calculates the peak shape similarity between the monoisotope peak and the second isotope peak by using the dot product as shown below: Similarity=∑i=1nxiyi∑i=1nxi2∑i=1nyi2(1) where n is the number of scans that both the monoisotope peak and the second isotope peak co-occur, and xi and yi are their respective signal intensities in the ith scan.',\n",
       "  [(59, 65, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'By performing the above peak detection procedures, iMet-Q generates a peak list with charge state and isotope ratio information for each input replicate file.',\n",
       "  [(51, 57, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'For an input data set with multiple samples (each probably with multiple technical replicates), iMet-Q performs the same algorithm for pair-wise alignment within a sample (i.e., aligning replicates) and then across samples.',\n",
       "  [(96, 102, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'iMet-Q first selects the peak list that contains the largest number of detected peaks as the reference list and aligns other peak lists to it.',\n",
       "  [(0, 6, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'For a pair of reference list and aligning peak list, iMet-Q uses peaks detected in both lists with their m/z and retention time differences within user-defined tolerances (i.e., mzTol and rtTol in iMet-Q) as landmarks [34].',\n",
       "  [(53, 59, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Given the retention time of landmarks in the reference list and in the aligning peak list as vectors A and B, respectively, iMet-Q uses LOESS regression [26, 27] to generate a mapping between A and B with span of 20% and weight of 1 since LOESS regression combines multiple linear least square regression models which fit localized subsets of the data.',\n",
       "  [(124, 130, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'iMet-Q then aligns the peaks in both peak lists that satisfy user-defined m/z tolerance and retention time tolerance based on adjusted retention time.',\n",
       "  [(0, 6, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In addition to viewing quantitation results in the main interface, iMet-Q reports the medians of m/z, retention time and abundance of the peaks in all replicates of all samples.',\n",
       "  [(67, 73, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'For the charge state of an aligned peak, iMet-Q displays the charge state that is detected in most replicates.',\n",
       "  [(41, 47, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'For the isotope ratio, iMet-Q collects the isotope ratios corresponding to the displayed charge state and reports the median as the isotope ratio.',\n",
       "  [(23, 29, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'We evaluated the performance of iMet-Q in metabolite quantitation by comparing the accuracy of detecting metabolites and the abundance consistency, calculated by Pearson product-moment correlation coefficient (PPMCC), of detected metabolites among replicates with XCMS, MetAlign, and MZmine 2 using samples ranging from a small standard data set (seven standard metabolites) to a complex biological data set (public Arabidopsis data set).',\n",
       "  [(32, 38, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'We also demonstrated the ability of iMet-Q on charge state determination and isotope ratio calculation.',\n",
       "  [(36, 42, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'To demonstrate the ability of iMet-Q, we used a standard metabolite mixture in which seven standard metabolites with different sample concentration in two samples were included for evaluation.',\n",
       "  [(30, 36, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Both profile and centroid data sets generated from the raw data were processed by iMet-Q, XCMS, MetAlign, and MZmine 2, respectively.',\n",
       "  [(82, 88, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'As shown in Table 1, iMet-Q, XCMS, and MZmine 2 had the average quantitation error of 12%, 12% and 13% on the profile data set, and 12%, 32% and 14% on the centroid data set.',\n",
       "  [(21, 27, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'The results showed that iMet-Q achieved the smallest quantitation errors and also had the same average quantitation error of 12% in both profile and centroid data set, suggesting the robustness of iMet-Q on processing input files in both data type.',\n",
       "  [(24, 30, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In addition to accurate quantitation, iMet-Q determined the charge state and isotope ratio of detected peaks.',\n",
       "  [(38, 44, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In order to evaluate the charge state determined by iMet-Q, we used CAMERA to annotate charge states and isotope peaks of the standard metabolites.',\n",
       "  [(52, 58, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Charge states and isotope peaks of the seven standard metabolites in both profile and centroid data sets annotated by iMet-Q were overall identical with CAMERA’s annotation.',\n",
       "  [(118, 124, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Using iMet-Q’s calculated isotope ratios (the isotope ratio tolerance is ±0.02) to filter out unnecessary metabolite candidates, we reduced 49% (89 out of 183) metabolite candidates (S6 Table).',\n",
       "  [(6, 12, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'The public Arabidopsis data of 36 distinct samples collected from eight different plant classes (classified by Matsuda et al.) was used to demonstrate iMet-Q’s performance on quantifying complex biological samples.',\n",
       "  [(151, 157, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Table 2 shows the reproducibility and normalized abundances of two internal standards detected by the four quantitation tools, where iMet-Q, XCMS, and MetAlign detected all the internal standards from each replicate, achieving 100% reproducibility, slightly better than MZmine 2.',\n",
       "  [(133, 139, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Furthermore, iMet-Q reported normalized abundances of 1±0.19 and 1±0.18 for the two standard metabolites, comparable to the best 1±0.18 reported by XCMS and MZmine 2 for Lidocaine and the best 1±0.17 reported by XCMS for Camphor-10-sulfonic acid.',\n",
       "  [(13, 19, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'As a result, iMet-Q detected all of the elucidated metabolites, whereas XCMS, MetAlign, and MZmine 2 detected 103 (62%), 145 (87%), and 129 (77%) of the elucidated metabolites, respectively.',\n",
       "  [(13, 19, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'As a result, iMet-Q achieved an average replicate abundance correlation of 0.92, whereas XCMS, MetAlign, and MZmine 2 achieved average abundance correlations of 0.87, 0.78, and 0.81, respectively (as shown in Fig 3).',\n",
       "  [(13, 19, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'The result revealed that iMet-Q not only achieved high accuracy on detecting the 167 elucidated metabolites but also provided very consistent quantitation result.',\n",
       "  [(25, 31, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'The numbers of detected peaks in the combined quantitation results of iMet-Q, XCMS, MetAlign, and MZmine 2 were 13079, 7487, 37364, and 19394, respectively.',\n",
       "  [(70, 76, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'The average abundance correlations of the eight plant classes using the combined quantitation results of iMet-Q, XCMS, MetAlign, and MZmine 2 were 0.71, 0.75, 0.65, and 0.59, respectively.',\n",
       "  [(105, 111, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Note that all the replicates of flower class, the best clustered plant class (blue color in Fig 4), were clustered using iMet-Q’s combined quantitation result, and the average abundance correlation of the flower class was 0.88.',\n",
       "  [(121, 127, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'This evaluation implies that, although the numbers of peaks detected by MetAlign and MZmine 2 were much larger than those detected by iMet-Q and XCMS, peaks detected by iMet-Q and XCMS might be more essential for distinguishing plant classes.',\n",
       "  [(134, 140, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'We thus used the 19 confirmed metabolites to demonstrate the ability of iMet-Q.',\n",
       "  [(72, 78, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'By calculating the abundance correlations between 19 metabolites and their in-source fragments, iMet-Q reported an abundance correlation of 0.94, whereas XCMS, MetAlign, and MZmine 2 reported abundance correlations of 0.61, 0.76, and 0.66, respectively (as shown in Fig 5).',\n",
       "  [(96, 102, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'This result revealed that the abundances of the 19 metabolites and their in-source fragments were accurately detected and calculated by iMet-Q.',\n",
       "  [(136, 142, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Furthermore, isotope ratios of 89.47% (17 out of 19) metabolites were accurately calculated by iMet-Q, with differences within 0.05 to the theoretical ratio (S12 Table).',\n",
       "  [(95, 101, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'We particularly implemented iMet-Q as a project-oriented quantitation platform, allowing users to conveniently quantify, manage and compare a large number of data sets or quantitation results in Microsoft Windows series platform (including Windows 7, 8, and Windows Server 2008, 2012).',\n",
       "  [(28, 34, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'To be specific, iMet-Q applies a tree structure to organize quantitation results, where each node represents a project and sub-nodes of a project represent the quantitation results (as shown in Fig 6).',\n",
       "  [(16, 22, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'iMet-Q provides three wizards to guide users easily executing the program as follows.',\n",
       "  [(0, 6, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Since multiple technical replicates may be conducted on a sample, iMet-Q pops up a table to guide users to group technical replicates by the sample.',\n",
       "  [(66, 72, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'During quantitation, iMet-Q sequentially detects peaks from each replicate and then aligns those detected peaks across replicates/samples.',\n",
       "  [(21, 27, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'To compare abundances across samples, iMet-Q hierarchically displays a quantitation result in the main interface, where the quantitation result is summarized in a summary table.',\n",
       "  [(38, 44, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In order to efficiently select peaks of interest in the summary table, iMet-Q provides a search function using the range of m/z, retention time, charge state, or isotope ratio to narrow down the peaks listed in the summary table.',\n",
       "  [(71, 77, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Therefore, in this paper, we introduced an intelligent quantitation tool, iMet-Q, which is capable of dynamically determining the peak widths in liquid chromatogram dimension without input parameter, and automatically performing isotope pattern assembly.',\n",
       "  [(74, 80, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In our evaluation, iMet-Q has the quantitation performance better than XCMS, MetAlign, and MZmine 2 on standard and large-scale metabolome data sets.',\n",
       "  [(19, 25, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Besides, iMet-Q provides both the charge states and isotope ratios of detected peaks.',\n",
       "  [(9, 15, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'Although our evaluation demonstrates that the metabolite isotope ratios calculated by iMet-Q are close to the theoretical values, it is important to note that isotope ratio can be affected by noise interference, saturation effects and different experimental conditions [36, 37].',\n",
       "  [(86, 92, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'In addition to providing accurate quantitation, charge state, and isotope ratios, iMet-Q is equipped with friendly user interfaces so that quantifying metabolites become an easy task.',\n",
       "  [(82, 88, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC4718670',\n",
       "  'The software program is now publicly available for download at http://ms.iis.sinica.edu.tw/comics/Software_iMet-Q.html.',\n",
       "  [(107, 113, 'iMet-Q', 'imet-q')],\n",
       "  'Metabolomics'],\n",
       " ['PMC8205298',\n",
       "  'We provide an R script for post-processing MSI data in the imzML community format (https://bitbucket.org/lababi/msi.r) and implemented the TrIQ in our open-source imaging software RmsiGUI (https://bitbucket.org/lababi/rmsigui/).',\n",
       "  [(180, 187, 'RmsiGUI', 'rmsigui')],\n",
       "  'Proteomics experiment, Metabolomics, Imaging, Small molecules'],\n",
       " ['PMC8205298',\n",
       "  'Default values for \\\\documentclass[12pt]{minimal} \\\\usepackage{amsmath} \\\\usepackage{wasysym}  \\\\usepackage{amsfonts}  \\\\usepackage{amssymb}  \\\\usepackage{amsbsy} \\\\usepackage{upgreek} \\\\usepackage{mathrsfs} \\\\setlength{\\\\oddsidemargin}{-69pt} \\\\begin{document} }{}k\\\\end{document}k and \\\\documentclass[12pt]{minimal} \\\\usepackage{amsmath} \\\\usepackage{wasysym}  \\\\usepackage{amsfonts}  \\\\usepackage{amssymb}  \\\\usepackage{amsbsy} \\\\usepackage{upgreek} \\\\usepackage{mathrsfs} \\\\setlength{\\\\oddsidemargin}{-69pt} \\\\begin{document} }{}q\\\\end{document}q in RmsiGUI are 100 and 98%, respectively.',\n",
       "  [(530, 537, 'RmsiGUI', 'rmsigui')],\n",
       "  'Proteomics experiment, Metabolomics, Imaging, Small molecules'],\n",
       " ['PMC8205298',\n",
       "  'Recently, we published an R-based platform for MSI data processing with a graphical user interface, RmsiGUI, which provides modules for the control of an open hardware imaging robot (Open LabBot), the processing of raw data, and the analysis of MSI data (Rosas-Román et al., 2020).',\n",
       "  [(100, 107, 'RmsiGUI', 'rmsigui')],\n",
       "  'Proteomics experiment, Metabolomics, Imaging, Small molecules'],\n",
       " ['PMC8205298',\n",
       "  'We integrated the TrIQ algorithm into RmsiGUI and provide the R code snippets for facilitating its adoption into other programs.',\n",
       "  [(38, 45, 'RmsiGUI', 'rmsigui')],\n",
       "  'Proteomics experiment, Metabolomics, Imaging, Small molecules'],\n",
       " ['PMC8205298',\n",
       "  'Figure 1 shows the graphical user interface of RmsiGUI with the TrIQ option selector.',\n",
       "  [(47, 54, 'RmsiGUI', 'rmsigui')],\n",
       "  'Proteomics experiment, Metabolomics, Imaging, Small molecules'],\n",
       " ['PMC8205298',\n",
       "  'RmsiGUI is freely available from https://bitbucket.org/lababi/rmsigui/.',\n",
       "  [(0, 7, 'RmsiGUI', 'rmsigui')],\n",
       "  'Proteomics experiment, Metabolomics, Imaging, Small molecules'],\n",
       " ['PMC8205298',\n",
       "  'We released TrIQ R scripts and the R package RmsiGUI under the terms of the GNU General Public License, GPL V3 (http://gplv3.fsf.org/).',\n",
       "  [(45, 52, 'RmsiGUI', 'rmsigui')],\n",
       "  'Proteomics experiment, Metabolomics, Imaging, Small molecules'],\n",
       " ['PMC8205298',\n",
       "  'The software RmsiGUI is freely available from https://bitbucket.org/lababi/rmsigui/.',\n",
       "  [(13, 20, 'RmsiGUI', 'rmsigui')],\n",
       "  'Proteomics experiment, Metabolomics, Imaging, Small molecules'],\n",
       " ['PMC11305627',\n",
       "  'A dual-chamber leadless pacemaker system has been designed for atrioventricular synchronous pacing using wireless, beat-to-beat, implant-to-implant (i2i) communication between distinct atrial and ventricular leadless pacemakers.',\n",
       "  [(149, 152, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Atrium-to-ventricle and ventricle-to-atrium i2i communication success rates were also assessed.',\n",
       "  [(44, 47, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Post hoc summary statistics describing the relationships between atrioventricular synchrony and i2i success, posture/activity, implantation indication, atrioventricular event, and heart rate were calculated.',\n",
       "  [(96, 99, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'In the evaluable population (n=384 of 464 enrolled [83%]; 61% male; age, 70 years; weight, 82 kg; 60% ejection fraction; 95% of beats evaluable), the mean atrioventricular synchrony of 98% of beats observed across all postures using the standard 300-millisecond limit was greater than both atrial-to-ventricular i2i (94%) and ventricular-to-atrial i2i (94%; P<0.001), exceeding both i2i values in 95% of patients.',\n",
       "  [(312, 315, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'In addition, the relationship between atrioventricular synchrony and the implant-to-implant (i2i) communication needed to maintain it was evaluated.',\n",
       "  [(93, 96, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'The dual-chamber LP consists of an atrial LP (ALP) and a ventricular LP (VLP; Figure 1A) that communicate bidirectionally and on a beat-to-beat basis to continually deliver atrioventricular synchronous pacing with a novel i2i communication modality.8 Transmissions are sent from ALP to VLP (A-to-V) and from VLP to ALP (V-to-A) immediately before each paced event and after each sensed event to avoid interference with noncommunication (ie, pacing) operations.',\n",
       "  [(222, 225, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Examples of i2i communication are shown in Figure 1B.',\n",
       "  [(12, 15, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'It is important to note that the i2i signal characteristics can be tuned to enhance transmission success using the i2i setting level, which incorporates the transmitting signal pulse amplitude, transmitting signal pulse duration, and receiving sensing threshold (range, 1–7; nominal, 4).',\n",
       "  [(33, 36, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'B, Example of implant-to-implant, or i2i, communication messages for As-Vp and Ap-Vs event sequences.',\n",
       "  [(37, 40, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'During successful bidirectional i2i communication, the system can function in DDD(R) mode.',\n",
       "  [(32, 35, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Persistent i2i communication interruption, however, could disrupt this harmony and result in delayed ventricular pacing or withheld atrial pacing.',\n",
       "  [(11, 14, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'The system mitigates such asynchrony by monitoring the receipt of each transmission and automatically transitioning to 1 of 3 “safeguard modes.” If A-to-V i2i communication fails, the system effectively functions in DDI mode.',\n",
       "  [(155, 158, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'In contrast, if V-to-A i2i communication fails, the system effectively functions in VDD mode.',\n",
       "  [(23, 26, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'If i2i communication fails in both directions simultaneously, the system effectively functions in VDI mode.8 Collectively, these 3 safeguard modes guarantee ventricular pacing at the programmed rate during i2i interruption while providing atrial tracking and pacing whenever possible, despite changes in intrinsic rhythm or atrioventricular conduction.',\n",
       "  [(3, 6, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'The atrioventricular synchrony that ultimately results from i2i, however, can be definitively assessed only by manually verifying atrial and ventricular activation timing through external electrocardiographic adjudication.',\n",
       "  [(60, 63, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Although such a manual assessment may not readily be feasible in a clinical implantation setting, the A-to-V and V-to-A i2i success rates (ie, i2i throughput, percent of transmissions sent that were successfully received) are device-based diagnostics that can be interrogated by the programmer.',\n",
       "  [(120, 123, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'These i2i success rates can serve as feedback metrics, upstream from atrioventricular synchrony, to guide implantation site selection and provide LP communication monitoring.9',\n",
       "  [(6, 9, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'LP diagnostics were cleared immediately before each posture period with the programmer (Merlin Patient Care System, Abbott), and LPs were interrogated after each posture period to determine interim i2i success rates.',\n",
       "  [(198, 201, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'In addition to the acute posture/activity recordings collected in clinic at 3 months, out-of-clinic durations of all distinct instances of i2i communication interruption from 1 to 3 months were collected from device interrogations at 3 months.',\n",
       "  [(139, 142, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'The model was fit with 3-month atrioventricular synchrony as the outcome and the following linear predictors with no associations: i2i success (A-to-V and V-to-A), posture/activity, pacing indication, mean heart rate, RA implantation site, RV implantation site, and atrioventricular event (percent Ap and percent Vp).',\n",
       "  [(131, 134, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'To evaluate the relationship between i2i communication success and the resulting atrioventricular synchrony, each metric was calculated as the mean across all completed postures for each patient.',\n",
       "  [(37, 40, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Over the analyzed patient population (N=384), the resulting mean success rates for atrioventricular synchrony, A-to-V i2i, and V-to-A i2i were 98.1% (95% CI, 96.7–99.4) of beats, 93.6% (95% CI, 91.2–96.1) of transmissions, and 94.1% (95% CI, 91.7–96.4) of transmissions, respectively, as shown in Figure 2A.',\n",
       "  [(118, 121, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'These i2i rates were achieved with mean programmed A-to-V and V-to-A i2i setting levels of 4.8±1.4 and 5.2±1.3 of 7, respectively.',\n",
       "  [(6, 9, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'V-to-A i2i was statistically greater than A-to-V i2i (P<0.005), and it is important to note that atrioventricular synchrony was greater than both V-to-A i2i and A-to-V i2i (P<0.001 for both).',\n",
       "  [(7, 10, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Relationship between atrioventricular (AV) synchrony and i2i transmission success.',\n",
       "  [(57, 60, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'A, atrioventricular synchrony, Atrial-to-ventricular (A-to-V) implant-to-implant (i2i) success, and ventricular-to-atrial (V-to-A) i2i success, calculated from the mean percent of beats across all postures for each patient.',\n",
       "  [(82, 85, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'B, Color-coded atrioventricular synchrony vs A-to-V i2i and V-to-A i2i.',\n",
       "  [(52, 55, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'The impact of i2i communication interruption on atrioventricular synchrony can be visualized in Figure 2B, which shows that most outlier patients with atrioventricular synchrony of 70% to 90% (orange) or <70% (red) were associated with A-to-V or V-to-A i2i success rates <80%.',\n",
       "  [(14, 17, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'The ability of the device-based i2i values to ultimately serve as the lower bound of the resulting atrioventricular synchrony was revealed using the difference between atrioventricular synchrony and the lower of the V-to-A and A-to-V i2i values.',\n",
       "  [(32, 35, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Atrioventricular synchrony was 7.1 (95% CI, 4.6–9.7) mean absolute percentage points greater than the lower of the 2 i2i success rates for each patient.',\n",
       "  [(117, 120, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Overall, atrioventricular synchrony exceeded both i2i values in 94.5% of patients (363/384) and was no lower than 5% below the lower i2i value in 99.0% of patients (380/384).',\n",
       "  [(50, 53, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'The durations of distinct instances of i2i interruption were quantified for the out-of-clinic period before the 3-month atrioventricular synchrony assessment.',\n",
       "  [(39, 42, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'From 1 to 3 months after implantation across all patients, 86.6% (95% CI, 82.5–90.7) of all instances of A-to-V i2i interruption and 96.8% (95% CI, 95.0–98.7) of all instances of V-to-A i2i interruption were <6 seconds in duration; moreover, 98.9% (95% CI, 97.7–100.0) and 98.9% (95% CI, 97.8–100.0) of all instances of A-to-V and V-to-A interruption, respectively, were <30 seconds in duration.',\n",
       "  [(112, 115, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Specifically, posture/activity, A-to-V i2i success, V-to-A i2i success, percent Ap, and percent Vp demonstrated a significant impact on atrioventricular synchrony (P<0.001), but pacing indication, heart rate, and implantation site did not.',\n",
       "  [(39, 42, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'This first dual-chamber LP system maintains atrioventricular synchrony through a novel beat-to-beat, i2i communication channel.',\n",
       "  [(101, 104, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'If 100% of such i2i transmissions are successful, atrioventricular synchrony would be observed in 100% of beats.',\n",
       "  [(16, 19, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'During bouts of i2i communication interruption, novel safeguard algorithms provide dynamic shifts in pacing mode to minimize any resulting asynchrony.',\n",
       "  [(16, 19, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'An important finding was that ≈99% of i2i interruption instances resolved within 30 seconds, when the system returns to DDD behavior before changes in atrial rate or PR interval are likely given an opportunity to disrupt atrioventricular synchrony.',\n",
       "  [(38, 41, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'As a consequence of the safeguards, the fleeting nature of most i2i interruptions, and potential for intrinsic ventricular activation, atrioventricular synchrony should always meet or exceed i2i success in either direction, absent any confounding sensing or capture issues.',\n",
       "  [(64, 67, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Across a broad spectrum of patient implantation indications, postures, activities, heart rates, implantation sites, and atrioventricular event combinations in this patient population, A-to-V and V-to-A i2i transmissions were successful in 93.6% and 94.1% of beats, respectively, yet atrioventricular synchrony was achieved in 98.1% of beats.',\n",
       "  [(202, 205, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Moreover, atrioventricular synchrony exceeded both i2i success rates in 94.5% of patients and was no lower than 5% below both i2i values in 99.0% of patients.',\n",
       "  [(51, 54, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Note that the apparent patient anomalies in which the mean atrioventricular synchrony did not exceed mean i2i success rates (eg, 3 orange points in the upper-right corner of Figure 2B) were associated with atrioventricular synchrony values just under the 90% cutoff (ie, 88.8%, 87.0%, and 83.3%).',\n",
       "  [(106, 109, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Although atrioventricular synchrony can be definitively verified only by manual electrocardiographic adjudication, i2i success rates are device-based diagnostics that can be readily interrogated by the programmer and serve as a lower bound for atrioventricular synchrony estimation.',\n",
       "  [(115, 118, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Therefore, implantation site selection and subsequent device monitoring can be facilitated by using i2i success as a conservative surrogate for atrioventricular synchrony.',\n",
       "  [(100, 103, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Two previous preclinical ovine studies of this system’s i2i communication reported long-term A-to-V and V-to-A i2i transmission success rates exceeding 99%8 and atrioventricular synchrony exceeding 99% across a variety of postures/activities, similar to this clinical study.9 The slightly higher i2i success rates observed in those 2 preclinical studies may be attributed to differences in ovine compared with human intra-LP distances and orientations, which are anatomically constrained and may affect i2i signal reception.',\n",
       "  [(56, 59, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Successful i2i communication, which is verified intraprocedurally during implantation, could theoretically be hindered by the ambulatory postures, activities, and cardiovascular conduction scenarios associated with daily living.',\n",
       "  [(11, 14, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'However, these anatomical LP locations may not fully capture the factors that can affect i2i communication and atrioventricular synchrony (eg, relative distance and orientation of the LPs) due to patient-to-patient differences in cardiac anatomy.',\n",
       "  [(89, 92, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'It is important to note that the i2i setting level can be manually adjusted beyond the automatic recommendations to achieve a patient-specific balance of i2i success and battery longevity.',\n",
       "  [(33, 36, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'During atrial pacing, greater atrioventricular synchrony could result from the VLP maintaining both the V-A and V-V delays, despite transient A-to-V i2i interruption.',\n",
       "  [(149, 152, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'In contrast, greater atrioventricular synchrony during ventricular sensing may be explained by its lack of dependence on A-to-V i2i communication to initiate the atrioventricular delay for timely ventricular activity when intact atrioventricular conduction was present.',\n",
       "  [(128, 131, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'This additional and more accurate quantification of true atrioventricular synchrony further illustrated the functionality of beat-to-beat i2i communication for true dual-chamber leadless pacing.',\n",
       "  [(138, 141, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'However, the reasons for exclusion (eg, withdrawn consent, LP fixation, improper device programming, ECG signals with persistent noise or low amplitudes) were independent of i2i communication and thus not expected to bias the reported communication success or downstream atrioventricular synchrony.',\n",
       "  [(174, 177, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'In addition, such out-of-clinic settings may also include environmental electrical noise that could potentially interfere with i2i communication.',\n",
       "  [(127, 130, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'An evaluation of atrioventricular synchrony during ambulatory 24-hour Holter periods without programming restrictions or prescribed postures/activities should be the subject of future analyses, as should longer-term i2i communication performance.',\n",
       "  [(216, 219, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  'Device-based i2i communication success diagnostics were within 5% of the assessed atrioventricular synchrony in 99% of patients and thus may serve as a lower-bound surrogate to an otherwise cumbersome manual atrioventricular synchrony measurement.',\n",
       "  [(13, 16, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC11305627',\n",
       "  ' Nonstandard Abbreviations and AcronymsA-to-Vatrial to ventricular deviceALPatrial leadless pacemakeri2iimplant-to-implantLPleadless pacemakerRAright atrialRVright ventricleV-to-Aventricular to atrial deviceVLPventricular leadless pacemaker ',\n",
       "  [(101, 104, 'i2i', 'ion-to-image')],\n",
       "  'Proteomics experiment, Imaging, Analytical chemistry'],\n",
       " ['PMC10146746',\n",
       "  'Alignment of the spectra was performed after the preprocessing step with MSIWarp, a Python package provided with C++ implementation.',\n",
       "  [(73, 80, 'MSIWarp', 'msiwarp')],\n",
       "  'Proteomics experiment, Imaging, Structure analysis'],\n",
       " ['PMC10146746',\n",
       "  'MSIWarp is a flexible tool compatible with multiple instrument types to perform mass alignment of mass spectrometry imaging spectra [22].',\n",
       "  [(0, 7, 'MSIWarp', 'msiwarp')],\n",
       "  'Proteomics experiment, Imaging, Structure analysis'],\n",
       " ['PMC10146746',\n",
       "  'MSIWarp alignment significantly improved the performance of the CNN model, especially for the two machines that showed lower performances without alignment.',\n",
       "  [(0, 7, 'MSIWarp', 'msiwarp')],\n",
       "  'Proteomics experiment, Imaging, Structure analysis'],\n",
       " ['PMC10146746',\n",
       "  'Keeping the alignment with MSIWarp, we compared the results obtained on the three culture media per machine (Table 2).',\n",
       "  [(27, 34, 'MSIWarp', 'msiwarp')],\n",
       "  'Proteomics experiment, Imaging, Structure analysis'],\n",
       " ['PMC10146746',\n",
       "  'Keeping the alignment with MSIWarp, we compared the performances obtained after 24 h and 48 h of growth on two machines (MYCO-PSL and SAINT-ANTOINE).',\n",
       "  [(27, 34, 'MSIWarp', 'msiwarp')],\n",
       "  'Proteomics experiment, Imaging, Structure analysis'],\n",
       " ['PMC10146746',\n",
       "  'Impact of the machine and of the alignment with MSIWarp.',\n",
       "  [(48, 55, 'MSIWarp', 'msiwarp')],\n",
       "  'Proteomics experiment, Imaging, Structure analysis'],\n",
       " ['PMC5608769',\n",
       "  'We have developed massPix—an R package for analysing and interpreting data from MSI of lipids in tissue.',\n",
       "  [(18, 25, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  ' massPix produces single ion images, performs multivariate statistics and provides putative lipid annotations based on accurate mass matching against generated lipid libraries.',\n",
       "  [(1, 8, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  ' massPix is an open-source tool for the analysis and statistical interpretation of MSI data, and is particularly useful for lipidomics applications.',\n",
       "  [(1, 8, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'Here, we have developed massPix, an R-based package which processes MSI data, plots single ion distributions and performs multivariate statistics [principal components analysis (PCA) and clustering].',\n",
       "  [(24, 31, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  ' massPix supports data in imzML format (Race et al.',\n",
       "  [(1, 8, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'Whilst massPix has been developed for high resolution matrix assisted laser desorption ionisation (MALDI) data acquired with Thermo Scientific instrumentation, the software is vendor agnostic and can be applied to any data in imzML format independent of mass spectrometry platform.',\n",
       "  [(7, 14, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'massPix is compatible with Windows, Mac and Linux operating systems, and requires at least sufficient RAM to load the entire experimental dataset into memory (for instance to process 3\\xa0GB image file, ~3.2\\xa0GB memory is used).',\n",
       "  [(0, 7, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'massPix is run from the R scripting interface, however a detailed knowledge of R is not required to install and use the software.',\n",
       "  [(0, 7, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'massPix outputs high quality images, a data frame of the final normalised and annotated image which can be further manipulated in R, and csv files for spectra corresponding to cluster centers, PCA loadings, and lipid annotations.',\n",
       "  [(0, 7, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'The massPix R package, all R scripts, library files and the imzML Converter are available on GitHub (https://github.com/hallz/massPix).',\n",
       "  [(4, 11, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'MALDI–MSI is currently more widely used within the field and these datasets have been used to developed massPix.',\n",
       "  [(104, 111, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'The overall data processing workflow followed by massPix is shown (b) ',\n",
       "  [(49, 56, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'The overall data processing workflow followed by massPix is shown (b)',\n",
       "  [(49, 56, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  ' massPix has the further capability to perform difference matching on deisotoped features to search for mass differences associated with measurement-introduced alternation (e.g.',\n",
       "  [(1, 8, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'massPix uses an unsupervised approach to classify pixels of high spectral similarity using PCA (Fig.',\n",
       "  [(0, 7, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'The use of massPix software can thus aid interpretation of region-specific molecular changes.',\n",
       "  [(11, 18, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'Single ion distributions produced by massPix for [PC(36:1)+K]+, [PC(38:6)+K]+ , [PC(40:6)+K]+, in a sub-section of cerebellum (b).',\n",
       "  [(37, 44, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'Availability and implementation: The source code, R package, documentation and test data are freely available to download from https://github.com/hallz/massPix.',\n",
       "  [(152, 159, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC5608769',\n",
       "  'massPix is distributed under the GNU General Public Licence (version 3).',\n",
       "  [(0, 7, 'massPix', 'masspix')],\n",
       "  'Proteomics experiment, Statistics and probability, Imaging, Metabolomics'],\n",
       " ['PMC7953914',\n",
       "  'The International Wheat Information System (WheatIS) Expert Working Group (EWG) was initiated in 2012 under the Wheat Initiative with a broad range of contributing organizations.',\n",
       "  [(44, 51, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The mission of the WheatIS EWG was to create an informational infrastructure, establish data standards, and build a single portal that allows search, retrieval, and display of globally distributed wheat data sets that are indexed in standard data formats at servers around the world.',\n",
       "  [(19, 26, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The web portal at WheatIS.org was released publicly in 2015, and by 2020, it expanded to 8 geographically-distributed nodes and around 20 organizations under its umbrella.',\n",
       "  [(18, 25, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Realizing the importance of findability and accessibility of wheat data sets distributed around the world, the Wheat Information System (WheatIS) Expert Working Group was established in 2012 to develop data standards for the wheat community and enable data query and access to globally-distributed data sets in standardized formats.',\n",
       "  [(137, 144, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Three years after the inception of the WheatIS EWG, the WheatIS portal was made publicly available in 2015 through wheatis.org.',\n",
       "  [(39, 46, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Currently, the web portal is maintained at the University of Western Australia, Australia, and the portal WheatIS servers are located at the “Plant Bioinformatics Facility” hosted by URGI France.',\n",
       "  [(106, 113, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Only after 7 years, WheatIS “nodes,” i.e., servers that contain indexed and formatted data sets, proliferated and are currently distributed in 3 continents in 5 countries, demonstrating the buy-in from the wheat research communities ( Alaux et al., 2018; Blake et al., 2019; Scheben et al., 2019; Wilkinson et al., 2016b; Yuan et al., 2017).',\n",
       "  [(20, 27, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  ' Goals of WheatIS Expert Working Group.',\n",
       "  [(10, 17, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The Wheat Initiative tasked the WheatIS EWG’s to provide the international wheat research community with easy access to wheat genetics, phenotype with environmental information, genomic data and bioinformatics tools, and to support and promote the diverse wheat databases internationally.',\n",
       "  [(32, 39, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'To help, the Wheat Data Interoperability Working Group (WDI-WG) was created as one of the Research Data Alliance working groups, under the umbrella of the WheatIS Expert Working Group.',\n",
       "  [(155, 162, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Some members belong to the WheatIS EWG, other have a more fundamental or transversal interest.',\n",
       "  [(27, 34, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'To this date, no salary is provided to WheatIS EWG members or the members of their research groups to create or contribute to wheatis.org.',\n",
       "  [(39, 46, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'This meager funding from the Wheat Initiative means that many people that are involved in the WheatIS EWG activities, such as curating data, building indexed data sets, configuring and maintaining servers, are doing these tasks on a volunteer basis in addition to their regular daily tasks.',\n",
       "  [(94, 101, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Fortunately, both computational and experimental research groups that are part of the WheatIS community recognize the primary importance of data availability, access, and sharing through wheatis.org, and because it is beneficial to the larger scientific community, they consider their service a crucial part of their scientific responsibility.',\n",
       "  [(86, 93, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The most significant accomplishment of the WheatIS EWG is the creation of a central hub, called WheatIS that provides a publicly available single-entry point.',\n",
       "  [(43, 50, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The WheatIS core server have access to resources at the globally-distributed nodes and enables data query and extraction through the web portal, unifying data discovery for the wheat research community.',\n",
       "  [(4, 11, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Specifically, the WheatIS portal was created to: 1) provide access to a data file repository storing files with their associated metadata; 2) allow queries to find data available in the WheatIS core and its nodes using keywords through a google-like search engine; 3) Data standards recommendations ( Dzale Yeumo et al., 2017); and 4) catalog several dedicated integrative databases that manage data types such as genomic, genetic, phenotypic, and functional genomic.',\n",
       "  [(18, 25, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  ' Current WheatIS searchable nodes.',\n",
       "  [(9, 16, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The following are the current organizations that manage a WheatIS server node: 1) the International Maize and Wheat Improvement Center CIMMYT (Mexico), 2) the European Bioinformatics Institute (EMBL-EBI) (UK), 3) the GrainGenes database (USA), 4) the Gramene database (USA), 5) the Triticeae Toolbox database (USA), 6) Transplant-IPGPAS (Poland), 7) l’Unité de Recherche Génomique Info (URGI) (France), and 8) wheatgenome.info at the University of Western Australia (Australia).',\n",
       "  [(58, 65, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  ' Rules of how to become a part of the WheatIS community.',\n",
       "  [(38, 45, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The WheatIS community is always expanding, adding new data sets and nodes from groups that never contributed data to wheatis.org.',\n",
       "  [(4, 11, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'WheatIS contributing members provides know-how and support to those who would like to create and maintain their own WheatIS nodes at their locations or contribute data to WheatIS, a simple request to wheatis-contact@wheatis.org will provide help and support.',\n",
       "  [(0, 7, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Our primary goal for the WheatIS Expert Working Group is to create a single portal that can query indexed data sets distributed worldwide, extract information, and provide access to these data sets to the wheat research community.',\n",
       "  [(25, 32, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The size and types of data sets accessible at WheatIS are growing daily with more nodes being added.',\n",
       "  [(46, 53, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'In all apparent measures, WheatIS is successful in building up a highly collaborative community of wheat research groups and creating a valuable product that is useful in connecting heterogenous data sets.',\n",
       "  [(26, 33, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'In the case of WheatIS, the remarkable need to search, reach, and extract wheat data sets that are generated across the globe was a given among wheat scientists, and still energizes the community as more data sets are being generated with continually cheapening experimental technologies and computational power.',\n",
       "  [(15, 22, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'In the case of WheatIS EWG, the formation of the Wheat Initiative by G20 ministers of agriculture instantly created such a supra-national umbrella organization.',\n",
       "  [(15, 22, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'It is also important to mention that some members of the WheatIS EWG are also members of the maize, Brassica and rice communities and their contributions played a significant role in WheatIS’s success and in turn their experience in the WheatIS initiative are making an impact in their communities.',\n",
       "  [(57, 64, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'It needed technical expertise to build and maintain a strong computational infrastructure and create data formats to make data sets readable; scientific expertise to understand different types of wheat data sets (including genetic, genomic, phenotypic, and metabolic); outreach capability to help build relationships to add new nodes with new data sets; and leaders who not only motivate and manage personnel, but also work with the Wheat Initiative and the broader wheat community to promote and support WheatIS.',\n",
       "  [(505, 512, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'For WheatIS, or for any scientific community for that matter, the critical question is the type of the expertise needed and how much time the experts can devote to a fledgling community.',\n",
       "  [(4, 11, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'When the WheatIS EWG was formed to create a single portal to make wheat data sets findable, accessible, and shareable ( Wilkinson et al., 2016a), the initial focus was primarily on the data sets.',\n",
       "  [(9, 16, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'However, sharing data sets also lead to strengthening wheat communities as well, which happened for WheatIS working under the Wheat Initiative.',\n",
       "  [(100, 107, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Through WheatIS and through sharing data sets, WheatIS has evolved into a fledgling nexus for the other EWGs, a few in the beginning, and more later, to contribute to a single portal where any data points generated would be made accessible.',\n",
       "  [(8, 15, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Through this communication network, WheatIS is helping the Wheat Initiative to become a more cohesive group and facilitates future collaborations.',\n",
       "  [(36, 43, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Although the current graphical user interface for WheatIS is functional, it needs improvement in several areas.',\n",
       "  [(50, 57, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'In the new interface, which we colloquially name WheatIS 2.0, we plan not only to work with the cosmetic issues, but also functional issues such as providing a more advanced and easy-to-use search capabilities.',\n",
       "  [(49, 56, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The WheatIS 2.0 will be shaped in these and other specific areas that were identified through personal discussions in the EWG meetings and the feedback we received from actual users.',\n",
       "  [(4, 11, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'We first thank the “Wheat Information System Expert Working Group” (WheatIS EWG) members for their valuable input to build this infrastructure.',\n",
       "  [(68, 75, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'We thank the Wheat Initiative led by Hélène Lucas, Peter Langridge and Frank Ordon that created a remarkable umbrella organization for WheatIS to thrive.',\n",
       "  [(135, 142, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  ' I wonder if they are planning to provide some data analytics or AI type of data prediction in WheatIS in the future.',\n",
       "  [(95, 102, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The manuscript described the process and successful experiences of forming an international research community, namely the international Wheat Information System (WheatIS) Expert working group.',\n",
       "  [(163, 170, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'At the lowest level of quality check, when a new dataset is indexed at one of the nodes, the WheatIS system automatically checks its formatting, and if the dataset is not formatted correctly, it is not displayed through the web interface, and flagged for further quality check.',\n",
       "  [(93, 100, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  ' Although correct formatting and accurate display of the datasets are important, the most important criterion for indexing a dataset at WheatIS is primarily its scientific value for the WheatIS users.',\n",
       "  [(136, 143, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'The process of selecting and indexing datasets at the WheatIS nodes is crucial; every dataset available through the WheatIS framework needs to be high-quality.',\n",
       "  [(54, 61, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Therefore, all indexed datasets at WheatIS are required to be either peer-reviewed, manually curated by professional curators, or both.',\n",
       "  [(35, 42, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'Most datasets at WheatIS are the products of peer-reviewed scientific research with associated peer-reviewed publications.',\n",
       "  [(17, 24, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC7953914',\n",
       "  'In addition to our ongoing volunteer efforts, we continuously look for new funding resources for WheatIS.',\n",
       "  [(97, 104, 'WheatIS', 'WheatIS')],\n",
       "  'Plant biology, Ecology, Biological databases, Bioinformatics, Agricultural science'],\n",
       " ['PMC11223827',\n",
       "  'The PSIPRED Workbench is available at http://bioinf.cs.ucl.ac.uk/psipred.',\n",
       "  [(4, 11, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'The PSIPRED Workbench is part of a worldwide ecosystem of Bioscience data repositories and web services.',\n",
       "  [(4, 11, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'We have been developing the PSIPRED Workbench for nearly 25 years.',\n",
       "  [(28, 35, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'Since 2019, we have published a number of new methods in the UCL Bioinformatics Group and have made some of these available online via the PSIPRED Workbench.',\n",
       "  [(139, 146, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'Similarly to PSIPRED, S4PRED prediction results comprise a confidence score, a cartoon representation, 3-state prediction assignment, and the original amino acid sequence.',\n",
       "  [(13, 20, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'This is a significant improvement over our cutting edge PSIPRED method, which achieves a Q3 accuracy of 70.6% when tested on single sequences without any provided homology information.',\n",
       "  [(56, 63, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'The PSIPRED Workbench offers a number of analysis methods.',\n",
       "  [(4, 11, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'Methods available via the PSIPRED workbench',\n",
       "  [(26, 33, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'Figure 4 shows the trends in usage of the PSIPRED web server since 2018, when our faster and more user-friendly web site was first launched.',\n",
       "  [(42, 49, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'Total number of predictive analysis tasks run by the PSIPRED Workbench in the years 2019 to 2023.',\n",
       "  [(53, 60, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'The PSIPRED workbench remains a popular and well-used bioinformatics resource for researchers across the globe.',\n",
       "  [(4, 11, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11223827',\n",
       "  'Funding for the PSIPRED Workbench is provided by the UK’s Biotechnology and Biological Sciences Research Council\\xa0under BBSRC\\xa0[BB/T019379/1].',\n",
       "  [(16, 23, 'PSIPRED', 'psipred')],\n",
       "  'Structure prediction, Protein folds and structural domains, Protein secondary structure, Protein folding, stability and design, Protein structure analysis'],\n",
       " ['PMC11508656',\n",
       "  'Thus, APE could be applied in the future as an accelerator of endothelial cell proliferation after, e.g., stent placement or atherosclerosis.',\n",
       "  [(6, 9, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'In a first preliminary pilot study, it was shown that the supplementation of human umbilical vein endothelial cells (HUVECs) with an aqueous extract of AP (APE) did not harm the cells but dose-dependently enhanced the development of an endothelial cell monolayer in vitro four days after seeding [17].',\n",
       "  [(156, 159, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'These data led us to hypothesise that APE can influence the proliferation of HUVECs and the formation of thrombi.',\n",
       "  [(38, 41, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'HUVECs were seeded in well plates in a cell culture medium supplemented with APE at three different concentrations (50, 100, and 200 µg/mL) [17].',\n",
       "  [(77, 80, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The HUVECs were evaluated at least 12 h after APE and cell culture medium supplementation to avoid a misinterpretation of the morphological state of the endothelial cells [21].',\n",
       "  [(46, 49, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The effect of APE on the proliferation of HUVECs was assessed in real-time by the xCELLigence system E-plate 16 (ACEA Biosciences, Inc., San Diego, CA, USA) for 85 h at 10 min intervals.',\n",
       "  [(14, 17, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The effect of APE in increasing concentrations (0 µg/mL, 50 µg/mL, 100 µg/mL, and 200 µg/mL) on the CI was measured.',\n",
       "  [(14, 17, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The endothelial cells reacted very differently to the addition of APE depending on the concentration of the extract.',\n",
       "  [(66, 69, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Compared to the control cells (46,720.0 ± 6473.9), there was a significant increase in endothelial cells 85 h after the addition of APE at 50 µg/mL (62,428.3 ± 4035.8) and at 100 µg/mL (70,931.1 ± 4850.2).',\n",
       "  [(132, 135, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Figure 2 clearly shows that the addition of 100 µg/mL APE resulted in a significant increase in the HUVEC density already after 48 h (25%) and even more so after 85 h (52%) of culturing compared to that in the control cells.',\n",
       "  [(54, 57, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'As an example, the course of monolayer growth for the addition of 100 µg/mL APE is shown in Figure 4.',\n",
       "  [(76, 79, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'While the addition of APE at a concentration of 50 µg/mL caused no change in monolayer growth compared to that in the control cells (p = 0.60), the cell index was significantly increased by 22% 85 h after the addition of 100 µg/mL of APE compared to that in the control cells (p = 0.0054).',\n",
       "  [(22, 25, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Two days after cell seeding and continuous cultivation in culture medium with APE, no significant differences in viability between untreated and APE-treated HUVECs could be detected (Figure 5).',\n",
       "  [(78, 81, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Eighty-five hours after cell seeding, a higher number of viable HUVECs cultured with APE-100 and APE-200 was detected compared to that in the control cells.',\n",
       "  [(85, 88, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'There were also significantly more viable HUVECs after the addition of APE-200 than after the addition of AP 50.',\n",
       "  [(71, 74, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'To analyse the influence of the possible damage of APE on HUVECs, the integrity of the outer endothelial cell membrane was examined.',\n",
       "  [(51, 54, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Therefore, two days after cell seeding, no cell damage caused by the addition of the APE was detected.',\n",
       "  [(85, 88, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Interestingly, the lowest LDH release also occurred in HUVECs treated with APE-100.',\n",
       "  [(75, 78, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Four days after cell seeding, LDH release was significantly lower in cell cultures treated with APE-50 and APE-100 compared to that in the untreated control.',\n",
       "  [(96, 99, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Cells cultured with APE-200 showed no differences compared to the control cells.',\n",
       "  [(20, 23, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'It can therefore be concluded that the cultivation of cells with APE does not have a negative effect on the integrity of the cell membrane and does not induce cell damage in the concentration range investigated.',\n",
       "  [(65, 68, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The absorbance values obtained for the APE-treated cells were then normalised to this value.',\n",
       "  [(39, 42, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Forty-eight hours after cell seeding, the metabolic activity of all the APE-treated HUVECs was significantly higher than 100% and thus above that of the untreated control cells.',\n",
       "  [(72, 75, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'A significantly increased metabolic activity was observed in cells cultured with 100 µg/mL APE compared to 50 and 200 µg/mL APE (Figure 8).',\n",
       "  [(91, 94, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Eighty-five hours after cell seeding, the metabolic activity of all the APE-treated cells was still above 100% and thus significantly higher than that of the untreated control.',\n",
       "  [(72, 75, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The highest metabolic activity was again observed in HUVECs cultured with APE-100 (Figure 8).',\n",
       "  [(74, 77, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Forty-eight hours after cell seeding, a significantly increased prostacyclin release was observed in APE-200-treated HUVECs compared to untreated and APE-50- and APE-100-treated cells (Figure 9).',\n",
       "  [(101, 104, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The significantly highest release was observed in APE-50-treated HUVECs compared to control and APE-100-treated cells.',\n",
       "  [(50, 53, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'HUVECs cultured with APE-100 showed the significantly lowest prostacyclin release compared to APE-50- and APE-200-treated cells (Figure 9).',\n",
       "  [(21, 24, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'To investigate the influence of cultivating HUVECs with APE on a possible inflammatory activation of the cells, the release of the pro-inflammatory interleukin-6 (IL-6) was analysed.',\n",
       "  [(56, 59, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Forty-eight hours after seeding, the significantly highest IL-6 release was found for HUVECs treated with APE-200 compared to those of the untreated and APE-100-treated cells.',\n",
       "  [(106, 109, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The lowest release was found in APE-100-treated HUVECs (Figure 10).',\n",
       "  [(32, 35, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The lowest IL-6 release occurred in untreated control cells compared to those in APE-50- and APE-200-treated cells.',\n",
       "  [(81, 84, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The highest release was observed in APE-200-treated HUVECs compared to that in APE-100-treated and untreated cells.',\n",
       "  [(36, 39, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Compared to APE-200- and APE-50-treated cells, the release of IL-6 by APE-100-treated cells was significantly reduced (Figure 10).',\n",
       "  [(12, 15, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'It can therefore be concluded that the release of IL-6 was significantly increased by culturing the cells in APE-50 and APE-200, whereas culturing the cells in APE-100 was comparable to the release of IL-6 from untreated HUVECs.',\n",
       "  [(109, 112, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Figure 11 shows adherent HUVECs after supplementation of the culture medium with 100 µg/mL APE.',\n",
       "  [(91, 94, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Forty-eight or eighty-five hours after cell seeding, all HUVECs (with or without APE) showed uniform vinculin expression.',\n",
       "  [(81, 84, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Thus, the formation of cell–cell and cell–substrate contacts was unaffected by the addition of APE.',\n",
       "  [(95, 98, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The study revealed that APE at a concentration of 100 µg/mL resulted in a significant increase in adherent HUVECs compared to untreated control cells after 48 h and especially 85 h days of cultivation.',\n",
       "  [(24, 27, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Figure 2 clearly shows that the addition of 100 µg/mL APE led to a significant increase in the density of HUVECs already after 48 h (+25%) and even more significantly after 85 h (+52%) of culturing.',\n",
       "  [(54, 57, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'However, the exact molecular mechanism by which APE achieves the increased HUVEC density remains unclear.',\n",
       "  [(48, 51, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'One possible reason could be the supplementation of cells with iron from APE.',\n",
       "  [(73, 76, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'There are studies analysing the iron content in APE [38,39,40].',\n",
       "  [(48, 51, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'After supplementation of the culture medium with APE, significantly more viable HUVECs adhered compared to control cells (see Figure 2, nearly no death cells were visible), the damage to the outer cell membrane was prevented and less LDH was detected in the culture medium compared to the untreated control, while at the same time the metabolic activity was increased, hinting at an increased proliferation of the HUVECs.',\n",
       "  [(49, 52, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The activation or perturbation of the endothelial cells, as previously described as a reaction to altered conditions in the culture milieu [41,42,43,44,45], also occurred here in the first two days after the addition of APE at concentrations of 50 and 200 µg/mL, but was completely prevented at a concentration of 100 µg/mL.',\n",
       "  [(220, 223, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The pattern of the IL-6 release of HUVECs with APE cultivation was very similar to that of prostacyclin.',\n",
       "  [(47, 50, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Here also, no increase in IL-6 concentration compared to that in control cells occurred for the APE concentration of 100 µg/mL (see Figure 10).',\n",
       "  [(96, 99, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Interestingly, no TXA2 was detected in the cell culture supernatant of HUVECs cultured with APE.',\n",
       "  [(92, 95, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'These release responses consistently indicated that there was no inflammatory response of the endothelial cells after adding 100 µg/mL APE to the culture medium.',\n",
       "  [(135, 138, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Thus, in the present study, the reduced synthesis of, e.g., prostacyclin and IL-6 (after 100 µg/mL APE in comparison to 50 or 200 µ/mL APE) could be associated with increased proliferation and thus stronger growth of the endothelial cell monolayer.',\n",
       "  [(99, 102, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The anti-inflammatory effect of APE may also contribute to the improvement of endothelial proliferation.',\n",
       "  [(32, 35, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'APE has been shown to reduce both TnFα and TGFβ [53,54].',\n",
       "  [(0, 3, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'showed that APE significantly reduced apoptosis [55].',\n",
       "  [(12, 15, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'In addition, an APE-induced increase in endothelial nitric oxide synthase (eNOS) expression has been repeatedly reported [59,60,61,62].',\n",
       "  [(16, 19, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Two or four days after cell seeding, HUVECs (with or without APE) showed uniform vinculin expression, so that the formation of cell–cell and cell–substrate contacts appeared to be unaffected by the addition of APE.',\n",
       "  [(61, 64, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'To date, in vivo toxicological studies of APE have not revealed any toxic effects on kidney, liver, the reproductive system, or body physiology during or after the administration of acute or chronic doses [68,69,70,71,72].',\n",
       "  [(42, 45, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'The highest proliferation was achieved after the addition of 100 µg/mL APE.',\n",
       "  [(71, 74, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Since neither cell–cell (VE-cadherin) nor cell–substrate (vinculin) binding was affected by APE, it is reasonable to assume that the increase in endothelial density was due to the proliferation of HUVECs.',\n",
       "  [(92, 95, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Representative examples of the HUVEC monolayers of the control culture (without APE) and the culture wit APE in the concentration of 100 µg/mL (85 h after seeding).',\n",
       "  [(80, 83, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Continuous detection of the HUVEC monolayer for the addition of 100 µg/mL APE to the culture medium using the xCelligence system, n = 8 each.',\n",
       "  [(74, 77, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'APE 50: HUVECs in culture medium with 50 µg/mL AP extract.',\n",
       "  [(0, 3, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'APE-100: HUVECs in culture medium with 100 µg/mL AP extract.',\n",
       "  [(0, 3, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'APE-200: HUVECs in culture medium with 200 µg/mL APE.',\n",
       "  [(0, 3, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  '(A) Phase contrast image; (B) dead HUVECs stained with PI in red, (C) FDA-stained HUVEC monolayer showing viable cells in green; (D) HUVEC monolayer supplemented with 50 µg/mL APE; (E) HUVEC monolayer supplemented with 100 µg/mL APE; (F) HUVEC monolayer supplemented with 200 µg/mL APE.',\n",
       "  [(176, 179, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'APE-50: HUVECs in culture medium with 50 µg/mL APE.',\n",
       "  [(0, 3, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'AP-100: HUVECs in culture medium with 100 µg/mL APE extract.',\n",
       "  [(48, 51, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'AP200: HUVECs in culture medium with 200 µg/mL APE extract.',\n",
       "  [(47, 50, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'APE-100: HUVECs in culture medium with 100 µg/mL APE.',\n",
       "  [(0, 3, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Prostacyclin (PGl2)—secretion of HUVECs two and four days after seeding (untreated control: HUVECs in culture medium; 50 µg/mL APE: HUVECs in culture medium with 50 µg/mL APE; 100 µg/mL APE: HUVECs in culture medium with 100 µg/mL APE; 200 µg/mL APE200: HUVECs in culture medium with 200 µg/mL APE.',\n",
       "  [(127, 130, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Representative image of paraformaldehyde-fixated HUVECs in culture medium supplemented with 100 µg/mL APE 85 h after seeding (HUVECs were fluorescently stained threefold, the actin cytoskeleton in green, von Willebrand factor in red, and genomic DNA in blue; scale bar: 50 µm).',\n",
       "  [(102, 105, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC11508656',\n",
       "  'Representative immunofluorescence images of paraformaldehyde-fixated HUVECs in culture medium, HUVECs with 50 µg/mL APE, HUVECs with 100 µg/mL APE, and HUVECs with 200 µg/mL APE 85 h after seeding (40× primary magnification).',\n",
       "  [(116, 119, 'APE', 'ape')],\n",
       "  'Workflows, Ontology and terminology, Bioinformatics'],\n",
       " ['PMC9929198',\n",
       "  'In Phase III the data was engineered using Agilent’s proprietary software and open source software: OpenChrom 1.4 [2], Julia 1.7 [3] and Linux (Ubuntu 21.04) [4].Data formatRaw, Analysed, FilteredParameters for data collectionGCMS standard operating procedures were adhered to for sample analysis, including temperature, run times, carrier gas and internal standards.',\n",
       "  [(100, 109, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'These files are required to be readable by OpenChrom in order that the analytical output can be accessed independently of the GCMS instrument.',\n",
       "  [(43, 52, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'OpenChrom was used to produce a range of files for each sample which were saved into the sample folder.',\n",
       "  [(0, 9, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'The method for producing these files using OpenChrom are described in the experimental design.',\n",
       "  [(43, 52, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'The retention times are recorded in intervals on an average of 600 milliseconds that is exported from OpenChrom.3.“Chromatogram-MS_”.',\n",
       "  [(102, 111, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'The other columns present are automatically produced by OpenChrom and are not relevant for sample identification.6.“Report_”.',\n",
       "  [(56, 65, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'The retention times are recorded in intervals on an average of 600 milliseconds that is exported from OpenChrom.',\n",
       "  [(102, 111, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'The other columns present are automatically produced by OpenChrom and are not relevant for sample identification.',\n",
       "  [(56, 65, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'In phase III, OpenChrom [7] was used to process the Agilent proprietary data into open source formats.',\n",
       "  [(14, 23, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'OpenChrom is an open source program developed to view and analyse chromatographic as well as other types of data.',\n",
       "  [(0, 9, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'Specifically OpenChrom can open GCMS data acquired from most proprietary vendors and runs on macOS, Windows and Linux [2].',\n",
       "  [(13, 22, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'DrugSamples_Drug-MDMA_Pathway-AlHg_Batch-75C.D.2.OpenChrom can read the sample data folder generated by ChemStation without any modification.3.OpenChrom detect peaks using the sample data and the AMDIS database, which must be installed.',\n",
       "  [(49, 58, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'OpenChrom detects peaks as follows:(i)right-click on the chromatogram,(ii)move the mouse pointer over “Peak Detector” and(iii)click “AMDIS (extern)” from the menu.This produces a number of windows.',\n",
       "  [(0, 9, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'The peaks are stored in memory and an inverted triangle is displayed on top of the detected peaks.4.Peak areas were obtained in OpenChrom following the steps:(i)right-click on the chromatogram,(ii)hover over “Peak Integrator”,(iii)select “Peak Integrator Trapezoid”; a window with title “Edit Processor Options” appears,(iv)keep the default button highlighted “Use System Options”, and(v)select “Finish” at the bottom right corner of the window.5.The calculated retention times and areas were recorded in a comma separated value (CSV) file following the naming convention in Fig.',\n",
       "  [(128, 137, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'with prefix “Peaks_”,(vi)click “Finish” at the bottom right of the pop-up window.6.OpenChrom created a report that contains the sample information and the peak areas.',\n",
       "  [(83, 92, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'The report was saved in the file with prefix “Report_” in text format (with extension “*.txt”); the report is obtained following the steps,(i)right-click on the chromatogram,(ii)select “Chromatogram Reports”,(iii)select “OpenChrom Report (*.txt)”,(iv)save the report by following the same process as in item 5, and by replacing prefix “Peaks_” with “Report_”.7.To record the TIS as a CSV file,(i)right-click on the chromatogram,(ii)move the cursor over “Chromatogram Export”,(iii)click on “CSV Chromatogram (*.csv)”(iv)and follow the same process as in item\\xa05, replacing the prefix with “Chromatogram-MS_”.8.At this point the sample file names are of the form DrugSamples_Drug-MDMA_Pathway-AlHg_Batch-75C.D, where the last block after the “_” represents the batch number of the drug production.',\n",
       "  [(221, 230, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'OpenChrom can read the sample data folder generated by ChemStation without any modification.',\n",
       "  [(0, 9, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'OpenChrom detect peaks using the sample data and the AMDIS database, which must be installed.',\n",
       "  [(0, 9, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'OpenChrom detects peaks as follows:(i)right-click on the chromatogram,(ii)move the mouse pointer over “Peak Detector” and(iii)click “AMDIS (extern)” from the menu.',\n",
       "  [(0, 9, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'Peak areas were obtained in OpenChrom following the steps:(i)right-click on the chromatogram,(ii)hover over “Peak Integrator”,(iii)select “Peak Integrator Trapezoid”; a window with title “Edit Processor Options” appears,(iv)keep the default button highlighted “Use System Options”, and(v)select “Finish” at the bottom right corner of the window.',\n",
       "  [(28, 37, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'OpenChrom created a report that contains the sample information and the peak areas.',\n",
       "  [(0, 9, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'The report was saved in the file with prefix “Report_” in text format (with extension “*.txt”); the report is obtained following the steps,(i)right-click on the chromatogram,(ii)select “Chromatogram Reports”,(iii)select “OpenChrom Report (*.txt)”,(iv)save the report by following the same process as in item 5, and by replacing prefix “Peaks_” with “Report_”.',\n",
       "  [(221, 230, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC9929198',\n",
       "  'select “OpenChrom Report (*.txt)”,',\n",
       "  [(8, 17, 'OpenChrom', 'openchrom')],\n",
       "  'Proteomics experiment, Proteomics, Metabolomics, Data visualisation'],\n",
       " ['PMC11480384',\n",
       "  'In the current study, the raw data for metabolomics can be obtained in Metabolights, with the ID MTBLS10002 (MetaboLights MTBLS)31.',\n",
       "  [(109, 121, 'MetaboLights', 'metabolights')],\n",
       "  'Metabolomics'],\n",
       " ['PMC11188463',\n",
       "  'The Human Splicing Finder, SpliceAI, and varSEAK database were used to predict the effect of mutations on splicing function.',\n",
       "  [(4, 25, 'Human Splicing Finder', 'human_splicing_finder')],\n",
       "  'Gene transcripts, RNA splicing, Genetic variation, Data governance, Rare diseases, Medical informatics'],\n",
       " ['PMC11188463',\n",
       "  'Human Splicing Finder Version 2.4.1 suggested that the c.5633A>T of DYSF mutation caused alteration of auxiliary sequences and significant alteration of the ESE/ESS motif ratio.',\n",
       "  [(0, 21, 'Human Splicing Finder', 'human_splicing_finder')],\n",
       "  'Gene transcripts, RNA splicing, Genetic variation, Data governance, Rare diseases, Medical informatics'],\n",
       " ['PMC11188463',\n",
       "  'Different complementary online software tools were employed for splicing prediction: Human Splicing Finder (HSF) Version 2.4.1 (http://www.umd.be/HSF/), SpliceAI (https://spliceailookup.broadinstitute.org/), and varSEAK (https://varseak.bio/).',\n",
       "  [(85, 106, 'Human Splicing Finder', 'human_splicing_finder')],\n",
       "  'Gene transcripts, RNA splicing, Genetic variation, Data governance, Rare diseases, Medical informatics'],\n",
       " ['PMC10696839',\n",
       "  'The prediction procedure for mass values was written in part of the GPMsDB-tk (GPMsDB-dbtk) scripts developed in this study.',\n",
       "  [(79, 90, 'GPMsDB-dbtk', 'gpmsdb-dbtk')],\n",
       "  'Database management, Microbiology, Proteomics'],\n",
       " ['PMC10696839',\n",
       "  'Matching of protein mass peaks was performed using GPMsDB-tk (v1.0.1, including GPMsDB-dbtk v1.0.1 for building custom databases for user-provided genomes), developed as part of this study.',\n",
       "  [(80, 91, 'GPMsDB-dbtk', 'gpmsdb-dbtk')],\n",
       "  'Database management, Microbiology, Proteomics'],\n",
       " ['PMC10696839',\n",
       "  'The source code and documentation are available at https://github.com/ysekig/GPMsDB-tk [59] and https://github.com/ysekig/GPMsDB-dbtk [60].',\n",
       "  [(122, 133, 'GPMsDB-dbtk', 'gpmsdb-dbtk')],\n",
       "  'Database management, Microbiology, Proteomics'],\n",
       " ['PMC10696839',\n",
       "  'For identification purposes, a toolkit (GPMsDB-tk) was developed in this study based on these observations with the current GPMsDB.',\n",
       "  [(40, 49, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'The best-matched genome entries were identified using GPMsDB-tk.',\n",
       "  [(54, 63, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'This confirmed that microbial identification by GPMsDB-tk is robust to differences in cultivation conditions.',\n",
       "  [(48, 57, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'In the cultivation of anaerobes from the same sample,\\u2009>\\u2009100 colonies were selected and anaerobically subcultured in 96-well plates, and then the resultant cells were used for identification with GPMsDB-tk.',\n",
       "  [(195, 204, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'Overall, 76 isolates (72%), including those matched with MAGs, were correctly identified at the species level; i.e., in these cases, MALDI-TOF MS identification with GPMsDB-tk gave the same GTDB taxonomy string down to the species level with those of 16S rRNA gene sequencing.',\n",
       "  [(166, 175, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  '4Identification of the 103 isolates obtained from mouse fecal samples based on MALDI-TOF MS profiles with GPMsDB-tk.',\n",
       "  [(106, 115, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'Identification of the 103 isolates obtained from mouse fecal samples based on MALDI-TOF MS profiles with GPMsDB-tk.',\n",
       "  [(105, 114, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'Our toolkit (GPMsDB-tk) and database (GPMsDB) thus substantially expand its utility for not only clinical discipline but also the screening of microbes in other microbial fields, including a search for uncultured lineages of descent in the domains Bacteria and Archaea.',\n",
       "  [(13, 22, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'We found that MALDI-TOF MS measurements with GPMsDB-tk successfully identified the isolates in most of the determinations (96 correct calls out of 103 isolates at the genus to strain [MAG] levels, 93%), as validated by full-length 16S rRNA gene sequencing (Additional file 7: Table S6).',\n",
       "  [(45, 54, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'The prediction procedure for mass values was written in part of the GPMsDB-tk (GPMsDB-dbtk) scripts developed in this study.',\n",
       "  [(68, 77, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'Matching of protein mass peaks was performed using GPMsDB-tk (v1.0.1, including GPMsDB-dbtk v1.0.1 for building custom databases for user-provided genomes), developed as part of this study.',\n",
       "  [(51, 60, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'GPMsDB-tk is a software toolkit for assigning taxonomic identification to user-provided MALDI-TOF mass spectrometry profiles.',\n",
       "  [(0, 9, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'Details of the functionality available in the GPMsDB-tk are shown in Additional file 1: Fig.',\n",
       "  [(46, 55, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'GPMsDB-tk is a Python package that can be run on a regular laptop using\\u2009~\\u20095\\xa0Gb of RAM for a single thread.',\n",
       "  [(0, 9, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'Unless specified, for taxonomic identification based on MALDI-TOF MS measurements, self-mass calibration in the GPMsDB-tk (option -aa) and -m option (-m 0) were used with the option “reps” (reference data with representative genome entries at the species level), “all” (reference data with all genome entries), or “custom” (reference data with all genome entries plus user-provided custom genome data) with the option “peak_bwf” with schoring theme III (“–score_type weighted”).',\n",
       "  [(112, 121, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'GPMsDB-tk provides not only score values for hit genomes but also probability value for each score.',\n",
       "  [(0, 9, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'Schematic of the GPMsDB-tk.',\n",
       "  [(17, 26, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'GPMsDB-tk results for identification of 81 bacterial and archaeal reference strains.Additional file 5: Table S4.',\n",
       "  [(0, 9, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'GPMsDB-tk results for identification of 13 acidobacterial strains.Additional file 6: Table S5.',\n",
       "  [(0, 9, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'GPMsDB-tk results for identification of 81 bacterial and archaeal reference strains.',\n",
       "  [(0, 9, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'GPMsDB-tk results for identification of 13 acidobacterial strains.',\n",
       "  [(0, 9, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'GPMsDB-tk, which consists of Python scripts and documentation, is licensed under the GNU General Public License (Version 3).',\n",
       "  [(0, 9, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC10696839',\n",
       "  'The source code and documentation are available at https://github.com/ysekig/GPMsDB-tk [59] and https://github.com/ysekig/GPMsDB-dbtk [60].',\n",
       "  [(77, 86, 'GPMsDB-tk', 'gpmsdb-tk')],\n",
       "  'Taxonomy, Proteomics, Microbiology'],\n",
       " ['PMC11351099',\n",
       "  'MacSyFinder with the TXSScan models is used to identify genes and gene clusters encoding for diverse secretion systems (24, 56).',\n",
       "  [(0, 11, 'MacSyFinder', 'macsyfinder')],\n",
       "  'Functional genomics'],\n",
       " ['PMC11351099',\n",
       "  'For example, DefenseFinder, MacSyFinder, and GapMind took around one and a half minutes or less to run for each test genome.',\n",
       "  [(28, 39, 'MacSyFinder', 'macsyfinder')],\n",
       "  'Functional genomics'],\n",
       " ['PMC10320092',\n",
       "  'In this scenario, we present KVFinder-web, an open-source web-based application of parKVFinder software for cavity detection and characterization of biomolecular structures.',\n",
       "  [(29, 37, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web has two independent components: a RESTful web service and a web graphical portal.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Our web service, KVFinder-web service, handles client requests, manages accepted jobs, and performs cavity detection and characterization on accepted jobs.',\n",
       "  [(17, 25, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Our graphical web portal, KVFinder-web portal, provides a simple and straightforward page for cavity analysis, which customizes detection parameters, submits jobs to the web service component, and displays cavities and characterizations.',\n",
       "  [(26, 34, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'We provide a publicly available KVFinder-web at https://kvfinder-web.cnpem.br, running in a cloud environment as docker containers.',\n",
       "  [(32, 40, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Further, this deployment type allows KVFinder-web components to be configured locally and customized according to user demand.',\n",
       "  [(37, 45, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Hence, users may run jobs on a locally configured service or our public KVFinder-web.',\n",
       "  [(72, 80, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  ' Graphical AbstractKVFinder-web: A web-based application for cavity detection and characterization of any type of biomolecular structure.',\n",
       "  [(19, 27, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'KVFinder-web: A web-based application for cavity detection and characterization of any type of biomolecular structure.',\n",
       "  [(0, 8, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Recently, we have developed parKVFinder (13), a robust method for detecting and characterizing binding sites in biomolecules, using a geometric\\xa0grid-and-sphere method and thread-level parallelization.',\n",
       "  [(31, 39, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Compared to other software for cavity detection, parKVFinder has an intuitive set of parameters and has been extensively benchmarked in the literature for detection and computational capabilities, delivering accurate and robust performance with any type of protein cavity (1,2,4,13).',\n",
       "  [(52, 60, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'parKVFinder stands out by combining geometric\\xa0and physicochemical characterizations of binding sites, effectively aiding users to identify functionally relevant cavities and to study the molecular recognition process.',\n",
       "  [(3, 11, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  \"Aside from parKVFinder's advances in performance and usability, the installation and configuration procedures of our software, as well as other standalone cavity detection software, still pose a major barrier to users who lack the technical knowledge ideally required to perform them.\",\n",
       "  [(14, 22, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'In this scenario, we introduce KVFinder-web, an open-source web-based application of the parKVFinder software, consisting of a RESTful web service for cavity detection and characterization and a graphical web portal.',\n",
       "  [(31, 39, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Our web application aims to democratize and expand even further the parKVFinder user base in the structural biology community.',\n",
       "  [(71, 79, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'These users will be able to perform cavity detection on third-party computing platforms, such as institutional servers or cloud infrastructures, using parKVFinder as a service (SaaS).',\n",
       "  [(154, 162, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'KVFinder-web is a powerful web-based tool designed to detect cavities in a wide range of biomolecular structures, including but not limited to proteins and nucleic acids.',\n",
       "  [(0, 8, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The interactive web interface (KVFinder-web portal) provides users an easy-to-use platform to execute the parKVFinder software and analyze the results through any web browser.',\n",
       "  [(31, 39, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Meanwhile, the web service (KVFinder-web service) itself executes parKVFinder (13) with its geometric\\xa0grid-and-sphere-based method, as detailed in (2,4,13).',\n",
       "  [(28, 36, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'To further enhance user experience, we also provide alternative client-side applications, including a Python HTTP client and a graphical PyMOL plugin (PyMOL KVFinder-web Tools).',\n",
       "  [(157, 165, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'To ensure more comprehensive characterization, KVFinder-web has been integrated with additional features, such as depth calculation and Eisenberg & Weiss hydropathy characterization, which are implemented in pyKVFinder (4).',\n",
       "  [(47, 55, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'These features are now part of the latest version of parKVFinder (v1.2.0) and enable users to obtain more robust and detailed results.',\n",
       "  [(56, 64, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web portal (Figure 1; https://github.com/LBC-LNBio/KVFinder-web-portal), developed using the Shiny R package (14) (https://shiny.rstudio.com), provides a simple and straightforward website for cavity analysis and visualization, requiring only a biomolecule in Protein Data Bank (PDB) format or PDB ID.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'KVFinder-web portal.',\n",
       "  [(0, 8, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  '(A) Screenshot of the KVFinder-web portal main page showing the main tabs and the sections of target biomolecule input and choice of KVFinder-web run mode.',\n",
       "  [(22, 30, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'After uploading the PDB, the KVFinder-web portal checks the PDB and informs non-standard residues detected [3].',\n",
       "  [(29, 37, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The ‘Run Cavity Analysis’ tab features the major functionalities of KVFinder-web, in which users can load a target biomolecule, customize the cavity detection parameters (‘Probe In’, ‘Probe Out’, ‘Removal Distance’, and ‘Volume Cutoff’), and download and visualize results.',\n",
       "  [(68, 76, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'To execute KVFinder-web (Figure 1B), users must first choose a biomolecule and then select an appropriate run mode.',\n",
       "  [(11, 19, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'KVFinder-web allows users to submit any kind of target biomolecule for cavity analysis\\xa0(Figure 1B).',\n",
       "  [(0, 8, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'After completing these two steps, users submit the job to the KVFinder-web service by clicking the ‘Submit the job’ button.',\n",
       "  [(62, 70, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web portal provides an easy and interactive way to download and visualize cavity detection results.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Results and visualization in KVFinder-web portal.',\n",
       "  [(29, 37, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  '(A) Screenshot of the results section of the KVFinder-web portal.',\n",
       "  [(45, 53, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web service (https://github.com/LBC-LNBio/KVFinder-web-service) is a RESTful service with a Web-Queue-Worker architecture.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The received data must contain the molecular structures and parKVFinder detection parameters.',\n",
       "  [(63, 71, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The ‘Worker’ module communicates with the ‘Queue’ module and requests the next job to be processed by the parKVFinder software.',\n",
       "  [(109, 117, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Each KVFinder-web service module is packaged in a Docker container (21), making it available to run in local or cloud computing environments.',\n",
       "  [(5, 13, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Besides our publicly available KVFinder-web service, a local web service can also be configured on third-party platforms, such as institutional servers or cloud services, using parKVFinder as a service (SaaS).',\n",
       "  [(31, 39, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web service has some limitations compared to a standalone version of parKVFinder, which are designed to prevent very demanding jobs from exhausting the web service.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Therefore, some parKVFinder parameters are constrained or even predefined.',\n",
       "  [(19, 27, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web application components (KVFinder-web service and KVFinder-web portal) will be continuously improved and updated to meet the needs of the scientific community, including new characterizations and performance enhancements.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'In addition to the primary interaction mode of KVFinder-web, we also provide additional client-side applications, such as PyMOL KVFinder-web and an example of a Python HTTP client, which broaden the range of possibilities for user interaction.',\n",
       "  [(47, 55, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'For users familiarized with PyMOL (16), PyMOL KVFinder-web Tools (Supplementary Figure S1; https://github.com/LBC-LNBio/PyMOL-KVFinder-web-Tools), developed in Python3 and Qt, integrates the KVFinder-web service with the molecular visualization software.',\n",
       "  [(46, 54, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'This user-friendly graphical user interface (GUI) enables customization of detection parameters for a target biomolecular structure and submits jobs to a configured KVFinder-web service (Supplementary Figure S1A).',\n",
       "  [(165, 173, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'As in parKVFinder PyMOL plugin (13), the search space can also be adjusted to a custom box (box adjustment mode) and/or a radius within a target ligand or molecule (ligand adjustment mode), instead of detecting and characterizing cavities throughout the biomolecular surface (whole structure mode).',\n",
       "  [(9, 17, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'After successful submission, accepted jobs are routinely and asynchronously requested to the KVFinder-web service.',\n",
       "  [(93, 101, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'This graphical plugin operates in a similar way to KVFinder-web portal, the characterizations are shown in lists (Supplementary Figure S1B) and the cavities customized by their properties in PyMOL viewer (Supplementary Figure S1C and D).',\n",
       "  [(51, 59, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Nevertheless, submitted jobs in the KVFinder-web portal can be loaded in PyMOL KVFinder-web Tools and vice versa.',\n",
       "  [(36, 44, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web portal and PyMOL KVFinder-web Tools may not fully fulfill the needs of advanced users.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'This client operates using standard parKVFinder input syntax and provides tools to interact with a configured KVFinder-web service.',\n",
       "  [(39, 47, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web service is publicly available at https://kvfinder-web.cnpem.br, which is free and open to all users without login requirement.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'As an illustrative example of KVFinder-web (Figure 3), we detected and characterized the catalytic site of the Human Immunodeficiency Virus type 1 (HIV-1) protease bound to cyclic urea carbonyl oxygen (22), using KVFinder-web portal.',\n",
       "  [(30, 38, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Besides cavity detection and shape definition, KVFinder-web also provides the volume, area, depth\\xa0and hydropathy of the detected cavities (Figure 2; cavity KAG).',\n",
       "  [(47, 55, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The cavity detection implemented in the KVFinder-web application has been extensively benchmarked against other well-known cavity detection methods in the literature (2,4,13), presenting robust detection and computational performance.',\n",
       "  [(40, 48, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'In addition, we evaluated KVFinder-web computational performance with a protein structure dataset from (13), varying two major parameters related to cavity detection: ‘Probe Out’ and ‘Removal Distance’ (Figure 4).',\n",
       "  [(26, 34, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Thus, greater ‘Probe Out’ sizes tend to reduce the degree of accessibility of the molecular surface created and ultimately, increase the elapsed time to perform calculations in KVFinder-web service (Figure 4A).',\n",
       "  [(177, 185, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Effects of detection parameters on KVFinder-web service performance.',\n",
       "  [(35, 43, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  \"KVFinder-web provides simple access to a rapid and accurate method of cavity detection and characterization in any type of biomolecular structure, regardless of the molecule's nature.\",\n",
       "  [(0, 8, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web portal provides a simple, easy and straightforward pipeline on any web browser with interactive visualization capabilities powered by NGLVieweR and DataTable.',\n",
       "  [(4, 12, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The graphical PyMOL plugin replicates the main features of parKVFinder PyMOL plugin with few limitations to avoid exhausting KVFinder-web service computational resources and to ensure a smooth and efficient operation for all users.',\n",
       "  [(62, 70, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'Together with the KVFinder-web service, the KVFinder-web portal and PyMOL KVFinder-web Tools aim to democratize parKVFinder software and remove barriers for users who do not have the technical knowledge to install and configure a cavity detection software, who have restricted computational resources, or who just want to perform a simple and quick analysis.',\n",
       "  [(18, 26, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The publicly available KVFinder-web service is accessible at https://kvfinder-web.cnpem.br.',\n",
       "  [(23, 31, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The corresponding open-source code of the KVFinder-web service, KVFinder-web portal, and PyMOL KVFinder-web Tools are available at https://github.com/LBC-LNBio/KVFinder-web-service\\xa0(https://doi.org/10.5281/zenodo.7825790), https://github.com/LBC-LNBio/KVFinder-web-portal\\xa0(https://doi.org/10.5281/zenodo.7825788), and https://github.com/LBC-LNBio/PyMOL-KVFinder-web-Tools\\xa0(https://doi.org/10.5281/zenodo.7825798), respectively.',\n",
       "  [(42, 50, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'The complete documentation of the modules of the KVFinder-web application is available at https://lbc-lnbio.github.io/KVFinder-web/.',\n",
       "  [(49, 57, 'KVFinder', 'kvfinder')],\n",
       "  'Structural biology, Computational chemistry, Structure analysis, Protein structure analysis'],\n",
       " ['PMC10320092',\n",
       "  'In this scenario, we present KVFinder-web, an open-source web-based application of parKVFinder software for cavity detection and characterization of biomolecular structures.',\n",
       "  [(29, 41, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web has two independent components: a RESTful web service and a web graphical portal.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Our web service, KVFinder-web service, handles client requests, manages accepted jobs, and performs cavity detection and characterization on accepted jobs.',\n",
       "  [(17, 29, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Our graphical web portal, KVFinder-web portal, provides a simple and straightforward page for cavity analysis, which customizes detection parameters, submits jobs to the web service component, and displays cavities and characterizations.',\n",
       "  [(26, 38, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'We provide a publicly available KVFinder-web at https://kvfinder-web.cnpem.br, running in a cloud environment as docker containers.',\n",
       "  [(32, 44, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Further, this deployment type allows KVFinder-web components to be configured locally and customized according to user demand.',\n",
       "  [(37, 49, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Hence, users may run jobs on a locally configured service or our public KVFinder-web.',\n",
       "  [(72, 84, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  ' Graphical AbstractKVFinder-web: A web-based application for cavity detection and characterization of any type of biomolecular structure.',\n",
       "  [(19, 31, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'KVFinder-web: A web-based application for cavity detection and characterization of any type of biomolecular structure.',\n",
       "  [(0, 12, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'In this scenario, we introduce KVFinder-web, an open-source web-based application of the parKVFinder software, consisting of a RESTful web service for cavity detection and characterization and a graphical web portal.',\n",
       "  [(31, 43, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'KVFinder-web is a powerful web-based tool designed to detect cavities in a wide range of biomolecular structures, including but not limited to proteins and nucleic acids.',\n",
       "  [(0, 12, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The interactive web interface (KVFinder-web portal) provides users an easy-to-use platform to execute the parKVFinder software and analyze the results through any web browser.',\n",
       "  [(31, 43, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Meanwhile, the web service (KVFinder-web service) itself executes parKVFinder (13) with its geometric\\xa0grid-and-sphere-based method, as detailed in (2,4,13).',\n",
       "  [(28, 40, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'To further enhance user experience, we also provide alternative client-side applications, including a Python HTTP client and a graphical PyMOL plugin (PyMOL KVFinder-web Tools).',\n",
       "  [(157, 169, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'To ensure more comprehensive characterization, KVFinder-web has been integrated with additional features, such as depth calculation and Eisenberg & Weiss hydropathy characterization, which are implemented in pyKVFinder (4).',\n",
       "  [(47, 59, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web portal (Figure 1; https://github.com/LBC-LNBio/KVFinder-web-portal), developed using the Shiny R package (14) (https://shiny.rstudio.com), provides a simple and straightforward website for cavity analysis and visualization, requiring only a biomolecule in Protein Data Bank (PDB) format or PDB ID.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'KVFinder-web portal.',\n",
       "  [(0, 12, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  '(A) Screenshot of the KVFinder-web portal main page showing the main tabs and the sections of target biomolecule input and choice of KVFinder-web run mode.',\n",
       "  [(22, 34, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'After uploading the PDB, the KVFinder-web portal checks the PDB and informs non-standard residues detected [3].',\n",
       "  [(29, 41, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The ‘Run Cavity Analysis’ tab features the major functionalities of KVFinder-web, in which users can load a target biomolecule, customize the cavity detection parameters (‘Probe In’, ‘Probe Out’, ‘Removal Distance’, and ‘Volume Cutoff’), and download and visualize results.',\n",
       "  [(68, 80, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'To execute KVFinder-web (Figure 1B), users must first choose a biomolecule and then select an appropriate run mode.',\n",
       "  [(11, 23, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'KVFinder-web allows users to submit any kind of target biomolecule for cavity analysis\\xa0(Figure 1B).',\n",
       "  [(0, 12, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'After completing these two steps, users submit the job to the KVFinder-web service by clicking the ‘Submit the job’ button.',\n",
       "  [(62, 74, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web portal provides an easy and interactive way to download and visualize cavity detection results.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Results and visualization in KVFinder-web portal.',\n",
       "  [(29, 41, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  '(A) Screenshot of the results section of the KVFinder-web portal.',\n",
       "  [(45, 57, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web service (https://github.com/LBC-LNBio/KVFinder-web-service) is a RESTful service with a Web-Queue-Worker architecture.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Each KVFinder-web service module is packaged in a Docker container (21), making it available to run in local or cloud computing environments.',\n",
       "  [(5, 17, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Besides our publicly available KVFinder-web service, a local web service can also be configured on third-party platforms, such as institutional servers or cloud services, using parKVFinder as a service (SaaS).',\n",
       "  [(31, 43, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web service has some limitations compared to a standalone version of parKVFinder, which are designed to prevent very demanding jobs from exhausting the web service.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web application components (KVFinder-web service and KVFinder-web portal) will be continuously improved and updated to meet the needs of the scientific community, including new characterizations and performance enhancements.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'In addition to the primary interaction mode of KVFinder-web, we also provide additional client-side applications, such as PyMOL KVFinder-web and an example of a Python HTTP client, which broaden the range of possibilities for user interaction.',\n",
       "  [(47, 59, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'For users familiarized with PyMOL (16), PyMOL KVFinder-web Tools (Supplementary Figure S1; https://github.com/LBC-LNBio/PyMOL-KVFinder-web-Tools), developed in Python3 and Qt, integrates the KVFinder-web service with the molecular visualization software.',\n",
       "  [(46, 58, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'This user-friendly graphical user interface (GUI) enables customization of detection parameters for a target biomolecular structure and submits jobs to a configured KVFinder-web service (Supplementary Figure S1A).',\n",
       "  [(165, 177, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'After successful submission, accepted jobs are routinely and asynchronously requested to the KVFinder-web service.',\n",
       "  [(93, 105, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'This graphical plugin operates in a similar way to KVFinder-web portal, the characterizations are shown in lists (Supplementary Figure S1B) and the cavities customized by their properties in PyMOL viewer (Supplementary Figure S1C and D).',\n",
       "  [(51, 63, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Nevertheless, submitted jobs in the KVFinder-web portal can be loaded in PyMOL KVFinder-web Tools and vice versa.',\n",
       "  [(36, 48, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web portal and PyMOL KVFinder-web Tools may not fully fulfill the needs of advanced users.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'This client operates using standard parKVFinder input syntax and provides tools to interact with a configured KVFinder-web service.',\n",
       "  [(110, 122, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web service is publicly available at https://kvfinder-web.cnpem.br, which is free and open to all users without login requirement.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'As an illustrative example of KVFinder-web (Figure 3), we detected and characterized the catalytic site of the Human Immunodeficiency Virus type 1 (HIV-1) protease bound to cyclic urea carbonyl oxygen (22), using KVFinder-web portal.',\n",
       "  [(30, 42, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Besides cavity detection and shape definition, KVFinder-web also provides the volume, area, depth\\xa0and hydropathy of the detected cavities (Figure 2; cavity KAG).',\n",
       "  [(47, 59, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The cavity detection implemented in the KVFinder-web application has been extensively benchmarked against other well-known cavity detection methods in the literature (2,4,13), presenting robust detection and computational performance.',\n",
       "  [(40, 52, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'In addition, we evaluated KVFinder-web computational performance with a protein structure dataset from (13), varying two major parameters related to cavity detection: ‘Probe Out’ and ‘Removal Distance’ (Figure 4).',\n",
       "  [(26, 38, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Thus, greater ‘Probe Out’ sizes tend to reduce the degree of accessibility of the molecular surface created and ultimately, increase the elapsed time to perform calculations in KVFinder-web service (Figure 4A).',\n",
       "  [(177, 189, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Effects of detection parameters on KVFinder-web service performance.',\n",
       "  [(35, 47, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  \"KVFinder-web provides simple access to a rapid and accurate method of cavity detection and characterization in any type of biomolecular structure, regardless of the molecule's nature.\",\n",
       "  [(0, 12, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The KVFinder-web portal provides a simple, easy and straightforward pipeline on any web browser with interactive visualization capabilities powered by NGLVieweR and DataTable.',\n",
       "  [(4, 16, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The graphical PyMOL plugin replicates the main features of parKVFinder PyMOL plugin with few limitations to avoid exhausting KVFinder-web service computational resources and to ensure a smooth and efficient operation for all users.',\n",
       "  [(125, 137, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'Together with the KVFinder-web service, the KVFinder-web portal and PyMOL KVFinder-web Tools aim to democratize parKVFinder software and remove barriers for users who do not have the technical knowledge to install and configure a cavity detection software, who have restricted computational resources, or who just want to perform a simple and quick analysis.',\n",
       "  [(18, 30, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The publicly available KVFinder-web service is accessible at https://kvfinder-web.cnpem.br.',\n",
       "  [(23, 35, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The corresponding open-source code of the KVFinder-web service, KVFinder-web portal, and PyMOL KVFinder-web Tools are available at https://github.com/LBC-LNBio/KVFinder-web-service\\xa0(https://doi.org/10.5281/zenodo.7825790), https://github.com/LBC-LNBio/KVFinder-web-portal\\xa0(https://doi.org/10.5281/zenodo.7825788), and https://github.com/LBC-LNBio/PyMOL-KVFinder-web-Tools\\xa0(https://doi.org/10.5281/zenodo.7825798), respectively.',\n",
       "  [(42, 54, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC10320092',\n",
       "  'The complete documentation of the modules of the KVFinder-web application is available at https://lbc-lnbio.github.io/KVFinder-web/.',\n",
       "  [(49, 61, 'KVFinder-web', 'kvfinder-web')],\n",
       "  'Structural biology, Computational biology, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'To fulfill this need, we developed pyKVFinder, a Python package to detect and characterize cavities in biomolecular structures for data science and automated pipelines.',\n",
       "  [(35, 45, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder efficiently detects cavities in biomolecular structures and computes their volume, area, depth and hydropathy, storing these cavity properties in NumPy arrays.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Benefited from Python ecosystem interoperability and data structures, pyKVFinder can be integrated with third-party scientific packages and libraries for mathematical calculations, machine learning and 3D visualization in automated workflows.',\n",
       "  [(70, 80, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'As proof of pyKVFinder’s capabilities, we successfully identified and compared ADRP substrate-binding site of SARS-CoV-2 and a set of homologous proteins with pyKVFinder, showing its integrability with data science packages such as matplotlib, NGL Viewer, SciPy and Jupyter notebook.',\n",
       "  [(12, 22, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder facilitates biostructural data analysis with scripting routines in the Python ecosystem and can be building blocks for data science and drug design applications.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'In this scenario, we developed pyKVFinder, an open-source python package for cavity detection and characterization abstracted into multidimensional arrays.',\n",
       "  [(31, 41, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder adopts the original geometrical grid-and-sphere-based detection method as implemented in KVFinder [7], which has been improved in the latest parallel version, parKVFinder [8].',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'To fulfill this need, pyKVFinder is wrapped into Python and, using Python’s well-established data structure (e.',\n",
       "  [(22, 32, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder can be integrated with third-party scientific packages and libraries for mathematical calculations, statistical analysis, and tridimensional visualization.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Moreover, users can explore the functionality of pyKVFinder step-by-step using interactive interfaces, such as IPython/Jupyter notebooks.',\n",
       "  [(49, 59, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'As mentioned above, one of pyKVFinder’s main contributions in data science workflows is to translate the detected cavities from tridimensional coordinates of cavity points to NumPy arrays, a data structure that allows for a wide diversity of scientific computation and efficient storage and access to N-dimensional arrays (ndarrays), also called tensors [19].',\n",
       "  [(27, 37, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Besides conventional cavity properties such as volume and area, which are stored as Python dictionaries, pyKVFinder computes cavity depth and hydropathy.',\n",
       "  [(105, 115, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Thus, pyKVFinder provides a versatile way to detect and characterize biomolecular cavities and integrate this information into data science or automated workflows.',\n",
       "  [(6, 16, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Python-C parallel KVFinder (pyKVFinder) applies a Simplified Wrapper and Interface Generator (SWIG; http://www.swig.org/) to extend grid operations written in C to Python, a high-level programming language.',\n",
       "  [(28, 38, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder can be easily installed with the pip package management system.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'In pyKVFinder, the target biomolecule is inserted into a regular 3D grid, which is stored as an ndarray, considering the van der Waals radii of the atoms.',\n",
       "  [(3, 13, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'To detect cavities, pyKVFinder uses a dual-probe algorithm that scans the biomolecular structure, as described in [7, 8].',\n",
       "  [(20, 30, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'On each detected cavity, pyKVFinder may perform spatial, depth, hydropathy and constitutional characterizations.',\n",
       "  [(25, 35, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder can be imported as a package in the Python environment and users can decide to run the full cavity detection and characterization workflow through the run_workflow function or run pyKVFinder functions in a stepwise fashion.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'At the latter, users can integrate pyKVFinder functions into third-party Python packages and benefit from interactive IPython/Jupyter notebooks.',\n",
       "  [(35, 45, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'By running pyKVFinder in Python environment, users can visualize the detected cavities through the Python NGL Viewer widget [21].',\n",
       "  [(11, 21, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  '1Diagram of cavity detection and characterization workflow using pyKVFinder package.',\n",
       "  [(65, 75, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'The flowchart illustrates function calls and their dependencies for performing cavity detection and characterization with pyKVFinder package',\n",
       "  [(122, 132, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Diagram of cavity detection and characterization workflow using pyKVFinder package.',\n",
       "  [(64, 74, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  '2Representative view of detected cavities in the pyKVFinder data structure.',\n",
       "  [(49, 59, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Based on a 3D grid (left figure), pyKVFinder detects cavities in biomolecules and returns an ndarray with dimensions (m, n, o) (right figure).',\n",
       "  [(34, 44, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Representative view of detected cavities in the pyKVFinder data structure.',\n",
       "  [(48, 58, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'With the ndarray of detected cavities, pyKVFinder package may perform four characterization procedures, i.e., spatial, constitutional, depth and hydropathy characterizations.',\n",
       "  [(39, 49, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'a Three different characterizations of the apo ADRP substrate-binding cavity of SARS-CoV-2 (PDB ID: 6WEN) using pyKVFinder.',\n",
       "  [(112, 122, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'c Hydropathy profile of the same compared cavities collected from pyKVFinder ndarrays.',\n",
       "  [(66, 76, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'The cavity detection and characterization objects, from detect, spatial, constitutional, calculate_frequencies, depth, hydropathy functions, can be stored into a pyKVFinderResults class that accumulates them in its attributes.',\n",
       "  [(162, 172, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'In this scenario, the export, write and plot_frequencies methods have their counterparts in the export, write_results and plot_frequencies functions of pyKVFinder package, respectively.',\n",
       "  [(152, 162, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Additionally, the read_cavity function reads a cavity file (.pdb extension), written by pyKVFinder, parKVFinder or KVFinder, and a target PDB or XYZ file (.pdb or.xyz extension), and returns an ndarray with each element corresponding to the cavity space (>\\u20091), biomolecule space (0), or bulk or empty space (−\\u20091), similar to the output of the detect function.',\n",
       "  [(88, 98, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'To demonstrate the use of pyKVFinder and how it benefits from the Python ecosystem, we identified the substrate-binding pocket of the ADP-ribose phosphatase (ADRP) domain of SARS-CoV-2 nsp3 protein in the apo form (PDB ID: 6WEN).',\n",
       "  [(26, 36, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder successfully detected the ADRP substrate-binding cavity and determined traditional cavity properties such as volume, area and residues surrounding the ADP-ribose cavity (Fig.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'For instance, we used pyKVFinder calculate_frequencies and plot_frequencies functions to determine the composition of the type of residues surrounding the cavity and plotted this composition as a bar chart (Additional file 1: Fig.',\n",
       "  [(22, 32, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'In pyKVFinder, this step is performed using matplotlib library [38], but users are free to analyze data and present results on their favorite graphing library.',\n",
       "  [(3, 13, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'The standard workflow of pyKVFinder, as in parKVFinder, detects cavities and applies spatial and constitutional characterizations.',\n",
       "  [(25, 35, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'The full workflow of pyKVFinder comprises standard workflow with depth and hydropathy characterizations',\n",
       "  [(21, 31, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Since pyKVFinder stores the properties to be colored in cavities in the B-factor column of a PDB-formatted file, users can easily change the style and color scheme in most of molecular visualization programs.',\n",
       "  [(6, 16, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'For this reason, we used pyKVFinder to detect the ADRP substrate-binding site in aligned ADRP domains from different species and compare their properties.',\n",
       "  [(25, 35, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Finally, since pyKVFinder uses native Python dictionaries to store the residues surrounding the detected cavity, we can easily tabulate the residue frequency.',\n",
       "  [(15, 25, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Thus, we used pyKVFinder with its box adjustment mode to detect and estimate the volume of the ADP-ribose binding site throughout 600 frames of the ADRP domain’s trajectory.',\n",
       "  [(14, 24, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'The shape of the detected cavities defined by pyKVFinder and parKVFinder finely adjust to the original ligand in the binding site, as well as MSPocket (Fig.',\n",
       "  [(46, 56, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  '4a) Besides that, the volume calculated by pyKVFinder (346.8\\u2009±\\u200978.7 Å3) and parKVFinder (346.5\\u2009±\\u200979.3 Å3) is closely related to the volume of ADP-ribose (351.1 Å3; molecular surface volume estimated by YASARA program [40]), the ligand that originally occupied the binding site in the crystallographic structure used in the molecular dynamics simulations (Fig.',\n",
       "  [(43, 53, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'For instance, pyKVFinder, parKVFinder and POVME can segment the search space, which trims points outside this custom space, while the other methods only explore the whole structure, which includes neighboring regions at the binding site.',\n",
       "  [(14, 24, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder outperformed all analyzed methods.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  \"Even when applying the newly available characterization, depth and hydropathy, pyKVFinder's elapsed time only increased 36%, still outperforming other benchmarking methods.\",\n",
       "  [(79, 89, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Further, compared to its counterpart, parKVFinder, pyKVFinder was 3.3 times faster in detecting ADRP binding site.',\n",
       "  [(51, 61, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Hence, experienced users requiring scripting routines are encouraged to use pyKVFinder due to its improved performance, while newcomers should prioritize parKVFinder due to its simplicity of installation and execution.',\n",
       "  [(76, 86, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Further, the scalability of pyKVFinder, upon increasing number of threads, follows the same behavior presented by parKVFinder [8].',\n",
       "  [(28, 38, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'However, the cavities data structure is only accessible inside the Python ecosystem in pyKVFinder, which provides ndarrays and Python dictionaries.',\n",
       "  [(87, 97, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder will undergo continuous improvements and updates, according to its applications by the scientific community.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Additionally, pyKVFinder aims to offload its routines to the GPU for performance enhancement in data-intensive applications.',\n",
       "  [(14, 24, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder provides an efficient and integrable Python package for cavity detection and characterization in biomolecular structures for data science and automated pipelines.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'In addition to fast, accurate and efficient cavity detection and characterization, pyKVFinder stores spatial and physicochemical properties in Python ndarrays, that ease scripting and data analysis.',\n",
       "  [(83, 93, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Further, pyKVFinder performance was benchmarked against well-known geometrical methods for cavity detection and characterization.',\n",
       "  [(9, 19, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Finally, we have successfully shown an application of pyKVFinder integration with matplotlib, NGL Viewer, SciPy and Jupyter notebook, that compared the ADRP substrate-binding site of SARS-CoV-2 in homologous proteins.',\n",
       "  [(54, 64, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  ' Project name: pyKVFinderProject home page: https://github.com/LBC-LNBio/pyKVFinderOperating system(s): any supporting Python\\u2009>\\u2009\\u2009=\\u20093.7 (tested on Linux and macOS)Programming language: Python, COther requirements: swig\\u2009>\\u2009\\u2009=\\u20094.0.1, toml\\u2009>\\u2009\\u2009=\\u20090.10.2, numpy\\u2009>\\u2009\\u2009=\\u20091.20.3, matplotlib\\u2009>\\u2009\\u2009=\\u20093.3.3License: GNU General Public License v3.0Any restrictions to use by non-academics: None.',\n",
       "  [(15, 25, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Project name: pyKVFinder',\n",
       "  [(14, 24, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Project home page: https://github.com/LBC-LNBio/pyKVFinder',\n",
       "  [(48, 58, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'pyKVFinder source code, documentation and tutorials are available in the Python Package Index (PyPI) repository, https://pypi.org/project/pyKVFinder, and the GitHub repository, https://github.com/LBC-LNBio/pyKVFinder.',\n",
       "  [(0, 10, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC8685811',\n",
       "  'Documentation and tutorials are available at pyKVFinder webpage, https://lbc-lnbio.github.io/pyKVFinder.',\n",
       "  [(45, 55, 'pyKVFinder', 'pykvfinder')],\n",
       "  'Structure analysis, Workflows, Mathematics, Structural biology, Machine learning, Computational chemistry, Protein structural motifs and surfaces'],\n",
       " ['PMC9341511',\n",
       "  'Tests on both simulated and real data sets show that TransMeta consistently outperforms PsiCLASS, StringTie2 plus its merge mode, and Scallop plus TACO, the most popular tools, in terms of precision and recall under a wide range of coverage thresholds at the meta-assembly level.',\n",
       "  [(88, 96, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'The second paradigm, such as the state-of-the-art meta-assembly tool PsiCLASS (Song et al.',\n",
       "  [(69, 77, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'PsiCLASS uses statistical methods to build the global subexon graphs and generates the assembly by dynamic programming optimization and voting algorithms.',\n",
       "  [(0, 8, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'We evaluated the performance of TransMeta at both the meta-assembly level and the individual sample level by comparing it with the state-of-the-art multisample assembler PsiCLASS, which can produce a meta-assembly for multiple samples and a certain set of transcripts for each individual sample as well.',\n",
       "  [(170, 178, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  '(2019) for their evaluation of PsiCLASS, we benchmarked the TransMeta under two widely used aligners, HISAT2 and STAR, in terms of the standard criteria that have been widely used in evaluation of transcriptome assemblers.',\n",
       "  [(31, 39, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'On the three simulated data sets under the HISAT2 and STAR alignments, we ran TransMeta and PsiCLASS with different filtering thresholds to get the sets of meta-assemblies, then ran StringTie2 and Scallop with their default settings for each individual sample, and finally merged them by using StringTie2-Merge and TACO, respectively, with different filtering thresholds.',\n",
       "  [(92, 100, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'Specifically, on the simulated data set S1 under the HISAT2 alignments, when we adjusted the coverage thresholds of all the assemblers to set their assemblies all to the precision of 0.7 (or any other value), the recall of TransMeta reached about 0.38, which is ∼20% higher than that of PsiCLASS, ∼41% higher than StringTie2 system, and ∼90% higher than Scallop + TACO (Fig.',\n",
       "  [(287, 295, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'Evaluated based on the areas under the precision-recall curve (AUC) scores, we found that the AUC score of TransMeta was ∼13.7% higher than that of PsiCLASS, 22.3% higher than that of StringTie2 system, and 49.8% higher than that of Scallop + TACO.',\n",
       "  [(148, 156, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'TransMeta and PsiCLASS are the so-called meta-assemblers, but they may also produce a certain set of transcripts for each sample.',\n",
       "  [(14, 22, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'Specifically, averaged on the 20 samples of data set S1 under the HISAT2 alignments, the recall of TransMeta reached about 0.38, which was ∼5.6% higher than that of PsiCLASS and >11.4% higher than those of StringTie2 and Scallop.',\n",
       "  [(165, 173, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  '(2019) in their evaluation of PsiCLASS.',\n",
       "  [(30, 38, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'Then on the five tested data sets R1–R5 under the HISAT2 alignments, the recalls of TransMeta reached about 0.11, 0.15, 0.16, 0.14, and 0.17, respectively; those of the second-best assembler PsiCLASS were about 0.07, 0.11, 0.12, 0.11, and 0.14, respectively; those of StringTie2 were about 0.05, 0.09, 0.11, 0.09, and 0.12, respectively; and those of Scallop + TACO were relatively worse.',\n",
       "  [(191, 199, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'It means that under the same precision level, TransMeta correctly identified ∼21%–57% more transcripts than PsiCLASS and 42%–120% more than the StringTie2 system.',\n",
       "  [(108, 116, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'Furthermore, in terms of the AUC, TransMeta showed an improvement of ∼17%–35% over PsiCLASS, as well as 19%–62% over the StringTie2 system on the five data sets, whereas Scallop + TACO showed relatively poor performance.',\n",
       "  [(83, 91, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'For instance, averaged on the 73 samples from R1, TransMeta correctly reconstructed ∼19.6% more transcripts than PsiCLASS, as well as 62.1% and 75.2% more than the single-sample assemblers StringTie2 and Scallop, respectively.',\n",
       "  [(113, 121, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'Meanwhile, its precision was the highest on all 73 samples, which were on average ∼15% higher than those of PsiCLASS and over twofold higher than those of StringTie2 and Scallop (Fig.',\n",
       "  [(108, 116, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'TransMeta and PsiCLASS are the so-called meta-assemblers that simultaneously analyze multiple samples.',\n",
       "  [(14, 22, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'Based on the alignments, we ran TransMeta and PsiCLASS with different filtering thresholds.',\n",
       "  [(46, 54, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'Comparisons of TransMeta and PsiCLASS on the large-scale data sets under the HISAT2 alignments at the meta-assembly level.',\n",
       "  [(29, 37, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  '(A–E) Precision-recall curves of TransMeta and PsiCLASS on 100, 200, 300, 400, and 500 subsets of GEUVADIS samples.',\n",
       "  [(47, 55, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'The meta-assemblers TransMeta and PsiCLASS were run with 25 threads, and the single-sample assemblers StringTie2 and Scallop were run sequentially to assemble individual samples.',\n",
       "  [(34, 42, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC9341511',\n",
       "  'For instance, on data set R1 with 73 samples under the HISAT2 alignments, TransMeta took ∼133 min with a peak memory of 11 GB, whereas PsiCLASS took ∼228 min with a peak memory of 11.7 GB, StringTie2 took 188 min with <1 GB memory, and Scallop took 251 min with a peak memory of ∼4.9 GB.',\n",
       "  [(135, 143, 'PsiCLASS', 'psiclass')],\n",
       "  'Sequence assembly'],\n",
       " ['PMC11530086',\n",
       "  'Information about the best performing definitions of IOH regarding 30-day mortality, hospital length of stay (hLOS), and postanesthesia care unit length of stay (PACU-LOS) is missing.',\n",
       "  [(162, 166, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'First, we used one subset to choose the best fitting definitions of IOH for the outcomes 30-day mortality, hLOS, and PACU-LOS.',\n",
       "  [(117, 121, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'The best fitting definitions were relative time with a MAP (mean arterial pressure) of <80 mmHg for 30-day mortality, lowest MAP for one minute for hLOS, and lowest MAP for one cumulative minute for PACU-LOS.',\n",
       "  [(199, 203, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'Testing these three definitions of IOH in the independent second subset confirmed the associations of IOH with 30-day mortality, hLOS, and PACU-LOS.',\n",
       "  [(139, 143, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'Using a data-driven approach, we identified the best fitting definitions of IOH for 30-day mortality, hLOS, and PACU-LOS.',\n",
       "  [(112, 116, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'The main objective of this trial was to find the best fitting definitions of IOH and their associations with 30-day mortality, hospital LOS (hLOS), and postanesthesia care unit (PACU) LOS in a general surgical population at a large academic center.',\n",
       "  [(178, 182, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'PACU-LOS was defined as the difference between the date of discharge and the date of admission to the PACU.',\n",
       "  [(0, 4, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'The LOS and PACU-LOS datasets were subset of the mortality dataset.',\n",
       "  [(12, 16, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'The statistical framework of this study was described similarly before in [21]: The aims of the study were structured into two sequential steps for each of the three outcomes (30-day mortality, hLOS and PACU LOS): 1) to determine which definition of hypotension has the strongest association with 30-day mortality, hLOS, and PACU-LOS and 2) to estimate the association between the best fitting hypotension definition identified in Aim 1 and the outcomes: 30-day mortality, hLOS and PACU-LOS.',\n",
       "  [(203, 207, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'This was done using a univariable, a multivariable logistic regression model (for binary outcome 30-day mortality) and two linear regression models (for the metric outcomes hLOS and PACU-LOS).',\n",
       "  [(182, 186, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'The 1% trimmed hLOS and PACU-LOS were log-transformed for use as dependent variables in the models in order to stabilize the residual distribution; the results were back-transformed to the original scale for graphical presentation.',\n",
       "  [(24, 28, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'By further excluding all patients not admitted to a PACU, we finally included 56,990 patients in the analysis of PACU LOS (Fig 1).',\n",
       "  [(52, 56, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'Abbreviations: hLOS: hospital length of stay; PACU: postanesthesia care unit; PACU-LOS: postanesthesia care unit length of stay; n, number.',\n",
       "  [(46, 50, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'The median hospital LOS was 5.1 (IQR: 3.1–10.3) days, and the median PACU-LOS was 2.1 (IQR: 1.6–2.9) h.',\n",
       "  [(69, 73, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'LOS: length of stay, PACU: postoperative care unit, BMI: body mass index, ASA: American association of anesthesiologist score, min: minutes, d: days, COPD: chronic obstructive pulmonary disease, AIDS: Acquired Immunodeficiency Syndrome, n: number, ENT: ear-nose-throat, non-or: non operating room',\n",
       "  [(21, 25, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'LOS: length of stay, PACU: postoperative care unit, BMI: body mass index, ASA: American association of anesthesiologist score, min: minutes, d: days, n: number, MAP: mean arterial pressure, IOH: intraoperative hypotension',\n",
       "  [(21, 25, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'Most of the definitions were negatively associated with mortality, hLOS, and PACU-LOS in univariable models.',\n",
       "  [(77, 81, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'For PACU-LOS, we found the strongest association with “lowest MAP for one cumulative minute”, closely followed by “lowest MAP for one minute”.',\n",
       "  [(4, 8, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'LOS: Length of stay, PACU: Post anesthesia care unit, LOS: length of stay, PACU: postoperative care unit, MAP: mean arterial pressure, IOH: intraoperative hypotension, min: minutes',\n",
       "  [(21, 25, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'The PACU-LOS showed very similar behavior but for “lowest MAP for 1 cumulative minute” (Fig 5).',\n",
       "  [(4, 8, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'MAP: mean arterial pressure, PACU: postanesthesia care unit.',\n",
       "  [(29, 33, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'BMI: body mass index, MAP: mean arterial pressure, PACU: postanesthesia care unit.',\n",
       "  [(51, 55, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'In addition to evaluating the association of IOH with mortality, hLOS, and PACU-LOS, we assessed the degree of influence of IOH and adjustment variables on these outcomes using the Brier score and MSE.',\n",
       "  [(75, 79, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'While mortality and hLOS were mainly influenced by the ASA score (mortality) and duration of surgery (hLOS), respectively, PACU LOS was mostly influenced by duration of surgery and IOH.',\n",
       "  [(123, 127, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'Regarding hLOS and PACU-LOS differences could be shown.',\n",
       "  [(19, 23, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'In contrast, the 0.95 confidence set for hLOS contains only three IOH definitions (Lowest MAP for 1 minute, Lowest MAP for 1 cumulative minute, and Lowest MAP for 3 cumulative minutes) and that for PACU-LOS contains only the selected Lowest MAP for 1 cumulative minute.',\n",
       "  [(198, 202, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'Data on the association of IOH and PACU-LOS were missing entirely to the authors’ best knowledge.',\n",
       "  [(35, 39, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'Regarding our LOS analyses, we found Lowest MAP for one minute to be the best fitting IOH definition for hLOS and “Lowest MAP for one cumulative minute” to be the best fitting IOH definition for PACU-LOS.',\n",
       "  [(195, 199, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'While IOH had less influence than length of surgery and duration from hospital admission to surgery on hLOS, it had as much influence as length of surgery on PACU-LOS.',\n",
       "  [(158, 162, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'Multivariable regression analyses revealed a sigmoid relationship between IOH and both hLOS and PACU-LOS (Figs 4 and 5).',\n",
       "  [(96, 100, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'In this study, we found a three-phase association of IOH with postoperative outcomes as follows: 1) below a MAP of approximately 60 mmHg, hLOS and PACU-LOS decreased with increasing MAP, followed by 2) a plateau between 60 and 75 mmHg, and 3) above a MAP of 75 mmHg, hLOS and PACU-LOS further decreased with increasing MAP.',\n",
       "  [(147, 151, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC11530086',\n",
       "  'In this retrospective trial, “Cumulative time with a MAP <80 mmHg” was the best fitting definition of intraoperative hypotension for associations with mortality, while “Lowest MAP for one minute” performed best for hospital length of stay, and “Lowest MAP for one cumulative minute” for PACU-length of stay.',\n",
       "  [(287, 291, 'PACU', 'pacu')],\n",
       "  'Phylogenetics, Sequence analysis'],\n",
       " ['PMC10924745',\n",
       "  'We applied our findings to the BDEI model and implemented its parameter estimator PyBDEI, employing targeted numerical analysis methods for accurate and fast resolution of its equations.',\n",
       "  [(82, 88, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'We show the accuracy and speed of PyBDEI on simulated data and compare it to the gold standard Bayesian tool BEAST2\\xa0(Bouckaert et\\xa0al., 2019) and the deep-learning-based tool PhyloDeep\\xa0(Voznica et\\xa0al., 2022).',\n",
       "  [(34, 40, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'Lastly, we apply PyBDEI to infer the epidemiological parameters that shaped the Ebola epidemic in Sierra-Leone in 2014.',\n",
       "  [(17, 23, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'We applied our theoretical findings to implement a fast and efficient parameter estimator for the BDEI model (which we called PyBDEI).',\n",
       "  [(126, 132, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'To assess the performance of our maximum-likelihood estimator PyBDEI, we used the simulated data from Voznica et\\xa0al.',\n",
       "  [(62, 68, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'To evaluate PyBDEI performance on forests, we additionally generated two types of forests for the large data set.',\n",
       "  [(12, 18, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'We applied PyBDEI to these data sets, and compared the results to those reported for BEAST2 and PhyloDeep in Voznica et\\xa0al.',\n",
       "  [(11, 17, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'For the large data set, we applied PyBDEI to full trees, but also to the two types of forests.',\n",
       "  [(35, 41, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'Average relative errors for PyBDEI were ≤13% on the medium trees and ≤2% on the large trees (hence decreasing with the data set size, as expected), and well centered around zero (i.e., unbiased), as shown in Fig.',\n",
       "  [(28, 34, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'For the medium data set BEAST2 (in orange), PhyloDeep (in green) and our estimator (PyBDEI, in blue) are compared.',\n",
       "  [(84, 90, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'We show the swarmplots (colored by method) of relative errors for each test tree/forest and parameter, which are measured as the normalized distance between the median a posteriori estimate by BEAST2 or a point estimate by PhyloDeep/PyBDEI and the real value.',\n",
       "  [(233, 239, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'For the infectious time, 1/ψ, PyBDEI was at least as accurate as PhyloDeep and more accurate than BEAST2 (P-value <0.05).',\n",
       "  [(30, 36, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'For the incubation period, 1/μ, PyBDEI was more accurate than both other methods (see Fig.',\n",
       "  [(32, 38, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'On the large data set BEAST2 was inapplicable due to computation times (57 CPU hours were already required for each medium-sized tree, on average), while PyBDEI was more accurate than PhyloDeep, both using full trees and the forests of the first type (i.e., where all the trees started at the same time, P-value <0.01 for all the parameters, see Fig.',\n",
       "  [(154, 160, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'On the forests of the second type (i.e., trees starting at different times) PyBDEI’s performance was comparable to the one of PhyloDeep for all the parameters.',\n",
       "  [(76, 82, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'While the mean relative errors were low (<10%), PyBDEI performed worse (P-value <0.01) on forests of type 2 than on full trees or forests of type 1 for the infectious period and incubation time.',\n",
       "  [(48, 54, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'PyBDEI performed well on these two trees: real parameters were within estimated confidence intervals, relative errors for Re<15%, relative errors for incubation period and infectious time <10%.',\n",
       "  [(0, 6, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'PyBDEI estimates had a higher or equal likelihood than any other method and real values for all the trees of both data sets, suggesting that PyBDEI reaches the global optimum of the likelihood function.',\n",
       "  [(0, 6, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'On the medium dataset, PyBDEI estimates had an equal likelihood to the ones of BEAST2 for 85% of trees, and a higher likelihood for the other 15% of trees.',\n",
       "  [(23, 29, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'Comparing to PhyloDeep, PyBDEI estimates had an equal likelihood for 48% of trees, and a higher likelihood for the other 52% of trees.',\n",
       "  [(24, 30, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'For example, 0.847 in the row “BEAST2 =” and the column “PyBDEI medium” means that the estimates of BEAST2 had an equal likelihood to those of PyBDEI on 84.7% of trees of the medium data set.',\n",
       "  [(57, 63, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'In terms of time, on the medium data set PyBDEI needed on average 4 seconds per tree on 1 CPU, and converged in 864 iterations (including CI calculation).',\n",
       "  [(41, 47, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'These times cannot be directly compared to BEAST2 times, as BEAST2 performs a Markov Chain Monte Carlo (MCMC) parameter space exploration instead of looking for the optimum (as PyBDEI does), hence requires many more steps: for 106 MCMC steps it took on average 57 CPU hours.',\n",
       "  [(177, 183, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'The average time of PyBDEI convergence on the large data set was 2 min 28 s on 1 CPU, and required 960 iterations.',\n",
       "  [(20, 26, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'However, to assess PyBDEI performance with other parameters fixed, we estimated parameters for trees in the large data set under three additional settings: with (1) μ, (2) λ, or (3) ψ fixed to its real value.',\n",
       "  [(19, 25, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'Using PyBDEI, we analysed the 2014 Ebola epidemic in Sierra-Leone (SLE).',\n",
       "  [(6, 12, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'This application shows the advantages of PyBDEI not only in terms of calculation times, but also in terms of flexibility of input settings (extracting information from multiple trees).',\n",
       "  [(41, 47, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'Our parameter estimator, PyBDEI, drastically increases parameter optimization performance, accuracy and speed with respect to previously available estimators.',\n",
       "  [(25, 31, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'PyBDEI is applicable to very large data sets (2 min on a 10000-tip tree), making parameter and CI estimation instantaneous with respect to phylogenetic tree reconstruction times (hours or even days).',\n",
       "  [(0, 6, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'With rapidly growing genome sequence data, castor and PyBDEI open way to fast and accurate parameter estimations for ecology and epidemiology.',\n",
       "  [(54, 60, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'To facilitate the use of our estimator in Python and perform additional validation of input trees, we wrapped the core estimator into a Python 3 library PyBDEI.',\n",
       "  [(153, 159, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10924745',\n",
       "  'PyBDEI uses ETE 3 framework for tree manipulation\\xa0(Huerta-Cepas et\\xa0al., 2016) and NumPy package for array operations\\xa0(Harris et\\xa0al., 2020).',\n",
       "  [(0, 6, 'PyBDEI', 'pybdei')],\n",
       "  'Phylogenetics'],\n",
       " ['PMC10885866',\n",
       "  'Because of fewer false positives and the greatest AUC, AntiBP3 is able to attain a balanced sensitivity and specificity.',\n",
       "  [(55, 62, 'AntiBP3', 'antibp3')],\n",
       "  ''],\n",
       " ['PMC10885866',\n",
       "  'We integrated our best-performing models in AntiBP3, a user-friendly online server (https://webs.iiitd.edu.in/raghava/antibp3/ (accessed on 20 November 2023)) that predicts antibacterial peptides for different groups of bacteria based on sequence information for each entry.',\n",
       "  [(44, 51, 'AntiBP3', 'antibp3')],\n",
       "  ''],\n",
       " ['PMC10885866',\n",
       "  'The datasets constructed for this study can be retrieved from the ‘AntiBP3′ web server at https://webs.iiitd.edu.in/raghava/antibp3/downloads.php (accessed on 20 November 2023).',\n",
       "  [(67, 74, 'AntiBP3', 'antibp3')],\n",
       "  ''],\n",
       " ['PMC10885866',\n",
       "  'The source code for this study is freely accessible on GitHub and can be found at https://github.com/raghavagps/AntiBP3 (accessed on 20 November 2023).',\n",
       "  [(112, 119, 'AntiBP3', 'antibp3')],\n",
       "  ''],\n",
       " ['PMC10885866',\n",
       "  'A flow diagram showing the process of creating AntiBP3 datasets for gram-positive, gram-negative, and gram-variable bacteria.',\n",
       "  [(47, 54, 'AntiBP3', 'antibp3')],\n",
       "  ''],\n",
       " ['PMC10885866',\n",
       "  'The comparison of existing prediction tools with AntiBP3 on the validation dataset to discriminate the three groups of ABPs, i.e., gram-positive, negative and variable.',\n",
       "  [(49, 56, 'AntiBP3', 'antibp3')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'Our method—MRSLpred—outperforms the existing state-of-the-art classifier in terms of performance and computation efficiency.',\n",
       "  [(11, 19, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'This method, named MRSLpred, is based on machine learning and uses the composition features of mRNA sequences for prediction.',\n",
       "  [(19, 27, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'Importantly, MRSLpred can be implemented at a genome scale as its computational resource requirements are minimal.',\n",
       "  [(13, 21, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'Outline of the methodology followed by MRSLpred.',\n",
       "  [(39, 47, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'The performance of both DM3Loc and MRSLpred was evaluated on the validation dataset, and it was observed that DM3Loc performs better than MRSLpred in terms of AUROC and MCC.',\n",
       "  [(35, 43, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'Both MRSLpred and DM3Loc have been designed as multi-label classifiers where more than one location can be assigned to a single sequence.',\n",
       "  [(5, 13, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'On the other hand, other existing methods perform multiclass classification, assigning only one location to a single sequence, and comparing MRSLpred/DM3Loc with these tools would be unfair to them.',\n",
       "  [(141, 149, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'However, for the sake of comparison, we evaluated the performance of all these tools on the validation dataset used in MRSLpred.',\n",
       "  [(119, 127, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'The detailed comparison of MRSLpred with DM3Loc and all other tools which do not support multi-label classification is provided in Table 5.',\n",
       "  [(27, 35, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'Benchmarking of MRSLpred with existing prediction tools on the validation dataset used in MRSLpred.',\n",
       "  [(16, 24, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'Another significant advantage of MRSLpred is that it is computationally very efficient and consumes very less time.',\n",
       "  [(33, 41, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'The detailed comparison of MRSLpred and DM3Loc based on the time taken to generate predictions is provided in Table 6.',\n",
       "  [(27, 35, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'Comparison of MRSLpred with DM3Loc based on the time taken for similar datasets.',\n",
       "  [(14, 22, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'MRSLpred achieves an AUROC between 0.708 and 0.816 (average: 0.742) and an MCC between 0.073 and 0.319 (average: 0.232) on the validation dataset.',\n",
       "  [(0, 8, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'MRSLpred uses a much simpler approach, combining an XGBoost model with a motif-based module and using compositional features for the model.',\n",
       "  [(0, 8, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'The time taken by MRSLpred for predicting the location of 500 mRNA sequences is less than 1\\xa0min, whereas for the same dataset, DM3Loc takes approximately 32\\xa0min.',\n",
       "  [(18, 26, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC10877048',\n",
       "  'MRSLpred is available online at https://webs.iiitd.edu.in/raghava/mrslpred/, and its standalone can be downloaded from https://webs.iiitd.edu.in/raghava/mrslpred/standalone.php.',\n",
       "  [(0, 8, 'MRSLpred', 'mrslpred')],\n",
       "  ''],\n",
       " ['PMC11177086',\n",
       "  'The authors have developed an in-silico tool, named PhageTB, for the prediction of phage-bacteria interactions with high accuracy.',\n",
       "  [(52, 59, 'PhageTB', 'cellppd-mod')],\n",
       "  ''],\n",
       " ['PMC11177086',\n",
       "  'We believe that PhageTB will be an effective method for prediction of phage-based therapy against bacterial infections.',\n",
       "  [(16, 23, 'PhageTB', 'cellppd-mod')],\n",
       "  ''],\n",
       " ['PMC11253343',\n",
       "  'We introduce the open-source PyChelator web application and the Python-based Google Colaboratory notebook, PyChelator Colab.',\n",
       "  [(29, 39, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'No notable differences were observed compared to the results obtained from the PyChelator web application.',\n",
       "  [(79, 89, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator is a user-friendly metal and chelator calculator that provides a platform for further development.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'It is provided as an interactive web application, freely available for use at https://amrutelab.github.io/PyChelator, and as a Python-based Google Colaboratory notebook at https://colab.research.google.com/github/AmruteLab/PyChelator/blob/main/PyChelator_Colab.ipynb.',\n",
       "  [(106, 116, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'To address these needs and concerns, we developed PyChelator web application (https://amrutelab.github.io/PyChelator/), an open source program, based on the well-established and widely used Maxchelator framework.',\n",
       "  [(50, 60, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator offers enhanced user experience and customization options.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'The Python code in a Google Colaboratory notebook makes the PyChelator functionalities readily available to the Python-using scientific community for further development.',\n",
       "  [(60, 70, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator web application uses JavaScript for calculations and utilizes HTML & CSS for the frontend interface.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'The graphical user interface (GUI) of the PyChelator web application is designed for simplicity of use and comprises three panels.',\n",
       "  [(42, 52, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'An easy-to-follow tutorial of an example calculation and manual entry of constants are included inside PyChelator, and shown in Fig.',\n",
       "  [(103, 113, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  '1Example uses of PyChelator.',\n",
       "  [(17, 27, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Example uses of PyChelator.',\n",
       "  [(16, 26, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator facilitates the entry of smaller concentrations of chelators and metals through a dropdown menu, enabling entry of values from molar (M) to nanomolar range (M, mM, μM, and nM).',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'A function to calculate the ionic equivalence inside PyChelator is also included.',\n",
       "  [(53, 63, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Additionally, PyChelator includes an option for the log-transformed values of the free metal concentrations (−\\u2009log10[free]), corresponding to the pX values (for example pCa for free calcium).',\n",
       "  [(14, 24, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'A screenshot of the PyChelator GUI, accompanied by annotations, is presented in Fig.',\n",
       "  [(20, 30, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  '2Screenshot of PyChelator graphical user interface (GUI), illustrating the intuitive layout and functionality.',\n",
       "  [(15, 25, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Screenshot of PyChelator graphical user interface (GUI), illustrating the intuitive layout and functionality.',\n",
       "  [(14, 24, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator is a single page application where users can switch constants by the use of a dropdown menu.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  '3Comparison of PyChelator results obtained from different stability constants.',\n",
       "  [(15, 25, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'A PyChelator offers the selection of stability constants from four sources, and allows the input of user-defined constants.',\n",
       "  [(2, 12, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'B Calculations done in PyChelator using the four sets of stability constants, sourced from the National Institute of Standards and Technology ‘NIST’ [11], ‘SPECS’ by Fabiato [2], ‘Chelator’ by Schoenmakers et al.',\n",
       "  [(23, 33, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Notably, the high-precision calculations performed in PyChelator Colab employing the NIST constants and arbitrary precision of 50 decimals in calculations (PyNIST), were similar to those obtained from PyChelator web application using the same constants (NIST)',\n",
       "  [(54, 64, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Comparison of PyChelator results obtained from different stability constants.',\n",
       "  [(14, 24, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator further enhances user customization by allowing manual input of user-defined stability constants.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'This feature makes it possible to use PyChelator as “Any Metal Any Chelator Calculator”.',\n",
       "  [(38, 48, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Some commonly used metal-chelator pairs are included together with references under the “Stability Constants” folder in PyChelator Github repository.',\n",
       "  [(120, 130, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator Colab offers a cloud-based interactive Python environment that lets users write and execute code in their browsers.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Unlike Flask and Django, which are designed to handle server-side operations and require managing server hosting and domain registration (which can be costly), PyChelator Colab and PyChelator web applications do not require server hosting.',\n",
       "  [(160, 170, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator Colab is also modified to use the built-in Python Decimal module, introducing user-defined precision in the calculations, which are otherwise limited by double precision in the floating-point arithmetic of Python and JavaScript.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Nevertheless, the results obtained from the PyChelator web application, which utilizes JavaScript in calculations, compared to PyChelator Colab, did not show a notable difference (Fig.',\n",
       "  [(44, 54, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'A general comparison of the features offered by PyChelator compared to other calculator software is given in Table\\xa01.Fig.',\n",
       "  [(48, 58, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  '4PyChelator Colab GUI.',\n",
       "  [(1, 11, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'A To run a calculation in PyChelator Colab, the user has to run the first code cell \"Constants\" which enables the dropdown for selection/entry of constants, followed by saving the options through clicking \"Save Values\".',\n",
       "  [(26, 36, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'A field for decimal precision is included to specify the level of precision in decimal places, employing arbitrary precision provided by the Decimals module of PythonTable\\xa01Comparison of PyChelator to other softwareProgramming languageConstantsInputOutputRefsPyChelator (2024)JavaScript, PythonNIST, Schoenmakers, Fabiato, Chelator, User-defined entryM, mM, μM, nMChoose parameters, Append, ExcelThis workMaxChelator (2010)JavaScriptNIST, Schoenmakers, Editable constants fileMOne at a time[6]Calcium (1993)N/AMartell and Smith, Editable constantsmMOne at a time[5]Chelator (1992)Turbo PascalSchoenmakers, Editable constants fileMOne at a time[3]SPECS (1988)FortranFabiato and Fabiato, Editable using WordStarMOne at a time[2]',\n",
       "  [(187, 197, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator Colab GUI.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Comparison of PyChelator to other software',\n",
       "  [(14, 24, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator validates the input values for temperature (0–40\\xa0°C), ionic equivalence (0–500\\xa0mM), and pH (0–14).',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'PyChelator was prepared to facilitate improvements in these calculations.',\n",
       "  [(0, 10, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'The PyChelator delivers a significant improvement over currently available web applications by offering a user-friendly metal chelator calculator that allows, among others, selection out of four sets of stability constants, input in smaller units, the option to specify user-defined constants, and the convenient ability to download results as a single Excel file.',\n",
       "  [(4, 14, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'The Python-based PyChelator Colab offers easy customization in the modern programming language with user-defined precision in decimal calculations.',\n",
       "  [(17, 27, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Project name: PyChelator.',\n",
       "  [(14, 24, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Project home page: https://amrutelab.github.io/PyChelator/.',\n",
       "  [(47, 57, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11253343',\n",
       "  'Source code: https://github.com/AmruteLab/PyChelator.',\n",
       "  [(42, 52, 'PyChelator', 'pychelator')],\n",
       "  'Biochemistry'],\n",
       " ['PMC11223802',\n",
       "  'The RNAscape webserver produces visualizations that can preserve topological correspondence while remaining both visually intuitive and structurally insightful.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape achieves this by designing a mathematical structural mapping algorithm which prioritizes the helical segments, reflecting their tertiary organization.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape runs a plotting script that is designed to generate publication-quality images.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape natively supports non-standard nucleotides, multiple base-pairing annotation styles\\xa0and requires no programming experience.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape can also be used to analyze RNA/DNA hybrid structures and DNA topologies, including G-quadruplexes.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The RNAscape webserver allows users to customize visualizations through various settings as desired.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape addresses and overcomes the outlined issues and limitations of existing approaches at several levels.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The RNAscape algorithm includes a mapping process that conforms to the helical geometry of RNA structures.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'At the same time, RNAscape optimizes each layout to place non-helical segments of the structure without sacrificing tertiary interactions.',\n",
       "  [(18, 26, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape output for various structures from the PDB.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The 3D structure at the top of each panel is from the PDB structure, with its corresponding RNAscape visualization shown below it.',\n",
       "  [(92, 100, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The RNAscape webserver (Figure 2) offers various customization options for its visualizations.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape encourages users to iteratively refine an image.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'In addition, RNAscape allows the user to modify the calculated map.',\n",
       "  [(13, 21, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'Upon completion, RNAscape visualizations can be exported to vector format (SVG) or image format (PNG), enabling further refinements by the user.',\n",
       "  [(17, 25, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'Additionally, RNAscape can directly fetch structures (biological assembly 1) from the PDB (21) based on a given PDB ID.',\n",
       "  [(14, 22, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape overview.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  '(A) RNAscape algorithm for geometric mapping.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape builds a 2D ladder representation for helical segments that closely adheres to their principal component analysis (PCA) projection.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  '(B) RNAscape webserver for plotting flexible, publication-quality visualizations.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  '(C) RNAscape frontend for user interaction and customization.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape allows users to upload a file or input a PDB ID for processing and to adjust various settings including base-pairing annotations, residue colors, nucleotide or text-label sizes\\xa0and numbering schema.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape also provides an\\xa0associated documentation.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'After processing, RNAscape supports interaction with output layouts including zooming, rotating, panning, and modifying the map.',\n",
       "  [(18, 26, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape supports multiple base-pairing annotation conventions: Leontis-Westhof (LW) (16), Saenger (19), DSSR (Dissecting the Spatial Structure of RNA) (20) and a no-annotation option.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The RNAscape webserver is a single-page web application.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'Upon upload, the structure file is sent via Hypertext Transfer Protocol Secure (HTTPS) to the RNAscape webserver where backend processing occurs.',\n",
       "  [(94, 102, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape algorithm.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'For each helical segment, RNAscape estimates the ladder axis by connecting the centroids of the starting and ending nucleotides with a line segment.',\n",
       "  [(26, 34, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The RNAscape backend utilizes the Matplotlib (24) and NetworkX (25) packages to plot visualizations.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'We present RNAscape output for various structures (Figures 1 and\\xa04) from the PDB (21).',\n",
       "  [(11, 19, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape output preserves the L-shaped topology (as opposed to known ‘clover leaf’ shaped secondary structure (26) visualizations) and annotates non-standard bases and base-pairing geometries (critical in many RNA interactions (27)).',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape can also process unusual DNA structures, as shown by a single-stranded DNA with circular topology (PDB ID: 4NOE, Figure 1B).',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape output for large-size structures from the PDB.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNA loop modeling (28) for riboswitches is an important area of research, and RNAscape visualizations (e.g.',\n",
       "  [(78, 86, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The pistol ribozyme (PDB ID: 6R47, Figure 1D) and\\xa0the Nicotinamide Adenine Dinucleotide-II (NAD-II) riboswitch (PDB ID: 8HBA, Figure 1G) illustrate\\xa0how RNAscape places non-helical segments and can clearly depict their non-standard base pairs with helical segments.',\n",
       "  [(152, 160, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape natively supports multiple strands (e.g.',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape is also able to visualize G-quadruplexes (PDB ID: 2M18, Figure 1E).',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The RNAscape visualization (Figure 1J, Supplementary Figure S1) for the same reflects the high degree of conformational symmetry, based on structural coordinates of the PTC provided by Bose et\\xa0al.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'RNAscape can run on relatively large structures (structures of up to 50 MB are processed\\xa0by the webserver).',\n",
       "  [(0, 8, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'stacked ladders, PDB ID: 7QDU, Figure 4D, 552 nucleotides), while RNAscape is able to reflect the 3D topology of these large RNA molecules.',\n",
       "  [(66, 74, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The RNAscape webserver (Figure 2C) displays three primary items: header, file upload, and documentation panels.',\n",
       "  [(4, 12, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'Clicking the ‘Run’ button runs the RNAscape pipeline on the uploaded structure file or provided PDB ID (biological assembly 1).',\n",
       "  [(35, 43, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'After running RNAscape for a structure, a user has the option to add a second structure for side-by-side viewing.',\n",
       "  [(14, 22, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The documentation panel enables easy navigation and provides a quick start guide, tips, and examples for using RNAscape.',\n",
       "  [(111, 119, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ['PMC11223802',\n",
       "  'The user also has the option of downloading RNAscape mapped points in a numerical format (.npz) processable by the NumPy (32) library.',\n",
       "  [(44, 52, 'RNAscape', 'rnascape')],\n",
       "  'RNA, Nucleic acid structure analysis, Mathematics, Imaging'],\n",
       " ...]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "result_df = pd.DataFrame(results_list, columns=[\"PMCID\", \"Sentence\", \"NER_Tags\", \"Topics\"])\n",
    "result_df = result_df.explode(\"NER_Tags\")\n",
    "\n",
    "# p = Path(__file__).resolve().parents[2]\n",
    "# print(p)\n",
    "result_df.to_csv(\"/Users/vedran/Desktop/tmp.csv\", index=False)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2026"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(results_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PMCID</th>\n",
       "      <th>Sentence</th>\n",
       "      <th>NER_Tags</th>\n",
       "      <th>Topics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1476</th>\n",
       "      <td>PMC11031912</td>\n",
       "      <td>Next, we calculated the differences in AUC bet...</td>\n",
       "      <td>(55, 70, GenoMycAnalyzer, genomycanalyzer)</td>\n",
       "      <td>Whole genome sequencing, Public health and epi...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            PMCID                                           Sentence  \\\n",
       "1476  PMC11031912  Next, we calculated the differences in AUC bet...   \n",
       "\n",
       "                                        NER_Tags  \\\n",
       "1476  (55, 70, GenoMycAnalyzer, genomycanalyzer)   \n",
       "\n",
       "                                                 Topics  \n",
       "1476  Whole genome sequencing, Public health and epi...  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_df.sample(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "bh24-literature-mining-3mhmtudb-py3.12",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
